// @layui/layui-vue/es/utils/index.js
const e = (e2) => {
  const t2 = e2;
  return t2.install = (n2) => {
    n2.component(t2.name, e2);
  }, t2;
}, t = (e2) => Boolean(e2 && 1 & e2.shapeFlag), n = (e2, t2) => Boolean(e2 && 6 & e2.shapeFlag), l = (e2, t2) => Boolean(e2 && 16 & e2.shapeFlag);
function o(e2, t2) {
  const n2 = t2.replace(/-(\w)/g, (e3, t3) => t3 ? t3.toUpperCase() : "");
  const l2 = t2.replace(/([A-Z])/g, " $1").trim().split(" ").join("-").toLowerCase();
  return e2.slots[n2] ? n2 : e2.slots[l2] ? l2 : t2;
}
function s(e2, t2) {
  let n2 = null;
  return function e3(t3, l2) {
    if (t3) {
      let o2 = Object.prototype.toString.call(t3);
      if ("[object Object]" === o2)
        if (t3.id && t3.id === l2)
          n2 = t3;
        else {
          let n3 = t3.children || null;
          e3(n3, l2);
        }
      else if ("[object Array]" === o2) {
        let o3 = t3.find((e4) => true == !!e4 && e4.id === l2);
        o3 ? n2 = o3 : t3.length && t3.forEach((t4) => {
          if (t4 && t4.children) {
            let n3 = t4.children;
            n3 && n3.length && e3(n3, l2);
          }
        });
      }
    }
  }(e2, t2), n2;
}
function a(e2) {
  let t2 = null;
  return function(n2) {
    t2 || (t2 = setTimeout(() => {
      t2 = null, e2(n2);
    }, 30));
  };
}
function r(e2) {
  e2.preventDefault();
}
function i(e2) {
  if (0 === e2.step)
    return [];
  let t2 = 0, n2 = [0], l2 = Math.floor(100 / e2.step) - 1;
  for (let o2 = 0; o2 < l2; o2++)
    t2 += e2.step, n2.push(t2);
  return n2.push(100), n2;
}
export { n as a, l as b, o as c, s as g, r as h, t as i, i as m, a as t, e as w };
