// @layui/layui-vue/es/_chunks/dayjs/index.js
import { c as t } from "../@umijs/index.js";
var n = { exports: {} }, e = n.exports = function() {
  var t2 = 1e3, n2 = 6e4, e2 = 36e5, r = "millisecond", i = "second", s = "minute", u = "hour", a = "day", o = "week", h = "month", f = "quarter", c = "year", d = "date", $ = "Invalid Date", l = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, M = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, m = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(t3) {
    var n3 = ["th", "st", "nd", "rd"], e3 = t3 % 100;
    return "[" + t3 + (n3[(e3 - 20) % 10] || n3[e3] || n3[0]) + "]";
  } }, v = function(t3, n3, e3) {
    var r2 = String(t3);
    return !r2 || r2.length >= n3 ? t3 : "" + Array(n3 + 1 - r2.length).join(e3) + t3;
  }, y = { s: v, z: function(t3) {
    var n3 = -t3.utcOffset(), e3 = Math.abs(n3), r2 = Math.floor(e3 / 60), i2 = e3 % 60;
    return (n3 <= 0 ? "+" : "-") + v(r2, 2, "0") + ":" + v(i2, 2, "0");
  }, m: function t3(n3, e3) {
    if (n3.date() < e3.date())
      return -t3(e3, n3);
    var r2 = 12 * (e3.year() - n3.year()) + (e3.month() - n3.month()), i2 = n3.clone().add(r2, h), s2 = e3 - i2 < 0, u2 = n3.clone().add(r2 + (s2 ? -1 : 1), h);
    return +(-(r2 + (e3 - i2) / (s2 ? i2 - u2 : u2 - i2)) || 0);
  }, a: function(t3) {
    return t3 < 0 ? Math.ceil(t3) || 0 : Math.floor(t3);
  }, p: function(t3) {
    return { M: h, y: c, w: o, d: a, D: d, h: u, m: s, s: i, ms: r, Q: f }[t3] || String(t3 || "").toLowerCase().replace(/s$/, "");
  }, u: function(t3) {
    return void 0 === t3;
  } }, D = "en", g = {};
  g[D] = m;
  var S = function(t3) {
    return t3 instanceof _;
  }, p = function t3(n3, e3, r2) {
    var i2;
    if (!n3)
      return D;
    if ("string" == typeof n3) {
      var s2 = n3.toLowerCase();
      g[s2] && (i2 = s2), e3 && (g[s2] = e3, i2 = s2);
      var u2 = n3.split("-");
      if (!i2 && u2.length > 1)
        return t3(u2[0]);
    } else {
      var a2 = n3.name;
      g[a2] = n3, i2 = a2;
    }
    return !r2 && i2 && (D = i2), i2 || !r2 && D;
  }, w = function(t3, n3) {
    if (S(t3))
      return t3.clone();
    var e3 = "object" == typeof n3 ? n3 : {};
    return e3.date = t3, e3.args = arguments, new _(e3);
  }, O = y;
  O.l = p, O.i = S, O.w = function(t3, n3) {
    return w(t3, { locale: n3.$L, utc: n3.$u, x: n3.$x, $offset: n3.$offset });
  };
  var _ = function() {
    function m2(t3) {
      this.$L = p(t3.locale, null, true), this.parse(t3);
    }
    var v2 = m2.prototype;
    return v2.parse = function(t3) {
      this.$d = function(t4) {
        var n3 = t4.date, e3 = t4.utc;
        if (null === n3)
          return new Date(NaN);
        if (O.u(n3))
          return new Date();
        if (n3 instanceof Date)
          return new Date(n3);
        if ("string" == typeof n3 && !/Z$/i.test(n3)) {
          var r2 = n3.match(l);
          if (r2) {
            var i2 = r2[2] - 1 || 0, s2 = (r2[7] || "0").substring(0, 3);
            return e3 ? new Date(Date.UTC(r2[1], i2, r2[3] || 1, r2[4] || 0, r2[5] || 0, r2[6] || 0, s2)) : new Date(r2[1], i2, r2[3] || 1, r2[4] || 0, r2[5] || 0, r2[6] || 0, s2);
          }
        }
        return new Date(n3);
      }(t3), this.$x = t3.x || {}, this.init();
    }, v2.init = function() {
      var t3 = this.$d;
      this.$y = t3.getFullYear(), this.$M = t3.getMonth(), this.$D = t3.getDate(), this.$W = t3.getDay(), this.$H = t3.getHours(), this.$m = t3.getMinutes(), this.$s = t3.getSeconds(), this.$ms = t3.getMilliseconds();
    }, v2.$utils = function() {
      return O;
    }, v2.isValid = function() {
      return !(this.$d.toString() === $);
    }, v2.isSame = function(t3, n3) {
      var e3 = w(t3);
      return this.startOf(n3) <= e3 && e3 <= this.endOf(n3);
    }, v2.isAfter = function(t3, n3) {
      return w(t3) < this.startOf(n3);
    }, v2.isBefore = function(t3, n3) {
      return this.endOf(n3) < w(t3);
    }, v2.$g = function(t3, n3, e3) {
      return O.u(t3) ? this[n3] : this.set(e3, t3);
    }, v2.unix = function() {
      return Math.floor(this.valueOf() / 1e3);
    }, v2.valueOf = function() {
      return this.$d.getTime();
    }, v2.startOf = function(t3, n3) {
      var e3 = this, r2 = !!O.u(n3) || n3, f2 = O.p(t3), $2 = function(t4, n4) {
        var i2 = O.w(e3.$u ? Date.UTC(e3.$y, n4, t4) : new Date(e3.$y, n4, t4), e3);
        return r2 ? i2 : i2.endOf(a);
      }, l2 = function(t4, n4) {
        return O.w(e3.toDate()[t4].apply(e3.toDate("s"), (r2 ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(n4)), e3);
      }, M2 = this.$W, m3 = this.$M, v3 = this.$D, y2 = "set" + (this.$u ? "UTC" : "");
      switch (f2) {
        case c:
          return r2 ? $2(1, 0) : $2(31, 11);
        case h:
          return r2 ? $2(1, m3) : $2(0, m3 + 1);
        case o:
          var D2 = this.$locale().weekStart || 0, g2 = (M2 < D2 ? M2 + 7 : M2) - D2;
          return $2(r2 ? v3 - g2 : v3 + (6 - g2), m3);
        case a:
        case d:
          return l2(y2 + "Hours", 0);
        case u:
          return l2(y2 + "Minutes", 1);
        case s:
          return l2(y2 + "Seconds", 2);
        case i:
          return l2(y2 + "Milliseconds", 3);
        default:
          return this.clone();
      }
    }, v2.endOf = function(t3) {
      return this.startOf(t3, false);
    }, v2.$set = function(t3, n3) {
      var e3, o2 = O.p(t3), f2 = "set" + (this.$u ? "UTC" : ""), $2 = (e3 = {}, e3[a] = f2 + "Date", e3[d] = f2 + "Date", e3[h] = f2 + "Month", e3[c] = f2 + "FullYear", e3[u] = f2 + "Hours", e3[s] = f2 + "Minutes", e3[i] = f2 + "Seconds", e3[r] = f2 + "Milliseconds", e3)[o2], l2 = o2 === a ? this.$D + (n3 - this.$W) : n3;
      if (o2 === h || o2 === c) {
        var M2 = this.clone().set(d, 1);
        M2.$d[$2](l2), M2.init(), this.$d = M2.set(d, Math.min(this.$D, M2.daysInMonth())).$d;
      } else
        $2 && this.$d[$2](l2);
      return this.init(), this;
    }, v2.set = function(t3, n3) {
      return this.clone().$set(t3, n3);
    }, v2.get = function(t3) {
      return this[O.p(t3)]();
    }, v2.add = function(r2, f2) {
      var d2, $2 = this;
      r2 = Number(r2);
      var l2 = O.p(f2), M2 = function(t3) {
        var n3 = w($2);
        return O.w(n3.date(n3.date() + Math.round(t3 * r2)), $2);
      };
      if (l2 === h)
        return this.set(h, this.$M + r2);
      if (l2 === c)
        return this.set(c, this.$y + r2);
      if (l2 === a)
        return M2(1);
      if (l2 === o)
        return M2(7);
      var m3 = (d2 = {}, d2[s] = n2, d2[u] = e2, d2[i] = t2, d2)[l2] || 1, v3 = this.$d.getTime() + r2 * m3;
      return O.w(v3, this);
    }, v2.subtract = function(t3, n3) {
      return this.add(-1 * t3, n3);
    }, v2.format = function(t3) {
      var n3 = this, e3 = this.$locale();
      if (!this.isValid())
        return e3.invalidDate || $;
      var r2 = t3 || "YYYY-MM-DDTHH:mm:ssZ", i2 = O.z(this), s2 = this.$H, u2 = this.$m, a2 = this.$M, o2 = e3.weekdays, h2 = e3.months, f2 = function(t4, e4, i3, s3) {
        return t4 && (t4[e4] || t4(n3, r2)) || i3[e4].slice(0, s3);
      }, c2 = function(t4) {
        return O.s(s2 % 12 || 12, t4, "0");
      }, d2 = e3.meridiem || function(t4, n4, e4) {
        var r3 = t4 < 12 ? "AM" : "PM";
        return e4 ? r3.toLowerCase() : r3;
      }, l2 = { YY: String(this.$y).slice(-2), YYYY: this.$y, M: a2 + 1, MM: O.s(a2 + 1, 2, "0"), MMM: f2(e3.monthsShort, a2, h2, 3), MMMM: f2(h2, a2), D: this.$D, DD: O.s(this.$D, 2, "0"), d: String(this.$W), dd: f2(e3.weekdaysMin, this.$W, o2, 2), ddd: f2(e3.weekdaysShort, this.$W, o2, 3), dddd: o2[this.$W], H: String(s2), HH: O.s(s2, 2, "0"), h: c2(1), hh: c2(2), a: d2(s2, u2, true), A: d2(s2, u2, false), m: String(u2), mm: O.s(u2, 2, "0"), s: String(this.$s), ss: O.s(this.$s, 2, "0"), SSS: O.s(this.$ms, 3, "0"), Z: i2 };
      return r2.replace(M, function(t4, n4) {
        return n4 || l2[t4] || i2.replace(":", "");
      });
    }, v2.utcOffset = function() {
      return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
    }, v2.diff = function(r2, d2, $2) {
      var l2, M2 = O.p(d2), m3 = w(r2), v3 = (m3.utcOffset() - this.utcOffset()) * n2, y2 = this - m3, D2 = O.m(this, m3);
      return D2 = (l2 = {}, l2[c] = D2 / 12, l2[h] = D2, l2[f] = D2 / 3, l2[o] = (y2 - v3) / 6048e5, l2[a] = (y2 - v3) / 864e5, l2[u] = y2 / e2, l2[s] = y2 / n2, l2[i] = y2 / t2, l2)[M2] || y2, $2 ? D2 : O.a(D2);
    }, v2.daysInMonth = function() {
      return this.endOf(h).$D;
    }, v2.$locale = function() {
      return g[this.$L];
    }, v2.locale = function(t3, n3) {
      if (!t3)
        return this.$L;
      var e3 = this.clone(), r2 = p(t3, n3, true);
      return r2 && (e3.$L = r2), e3;
    }, v2.clone = function() {
      return O.w(this.$d, this);
    }, v2.toDate = function() {
      return new Date(this.valueOf());
    }, v2.toJSON = function() {
      return this.isValid() ? this.toISOString() : null;
    }, v2.toISOString = function() {
      return this.$d.toISOString();
    }, v2.toString = function() {
      return this.$d.toUTCString();
    }, m2;
  }(), Y = _.prototype;
  return w.prototype = Y, [["$ms", r], ["$s", i], ["$m", s], ["$H", u], ["$W", a], ["$M", h], ["$y", c], ["$D", d]].forEach(function(t3) {
    Y[t3[1]] = function(n3) {
      return this.$g(n3, t3[0], t3[1]);
    };
  }), w.extend = function(t3, n3) {
    return t3.$i || (t3(n3, _, w), t3.$i = true), w;
  }, w.locale = p, w.isDayjs = S, w.unix = function(t3) {
    return w(1e3 * t3);
  }, w.en = g[D], w.Ls = g, w.p = {}, w;
}();
export { e as d };
