// @layui/layui-vue/es/_chunks/@vueuse/index.js
import { getCurrentScope as e, onScopeDispose as n, computed as t, toRefs as u, getCurrentInstance as o, onMounted as a, nextTick as i, ref as s, isRef as r, reactive as l, unref as c, watch as d, customRef as v, onUpdated as p } from "vue";
function f(t2) {
  return !!e() && (n(t2), true);
}
function w(e2) {
  return function(e3) {
    if (!r(e3))
      return l(e3);
    const n2 = new Proxy({}, { get: (n3, t2, u2) => c(Reflect.get(e3.value, t2, u2)), set: (n3, t2, u2) => (r(e3.value[t2]) && !r(u2) ? e3.value[t2].value = u2 : e3.value[t2] = u2, true), deleteProperty: (n3, t2) => Reflect.deleteProperty(e3.value, t2), has: (n3, t2) => Reflect.has(e3.value, t2), ownKeys: () => Object.keys(e3.value), getOwnPropertyDescriptor: () => ({ enumerable: true, configurable: true }) });
    return l(n2);
  }(t(e2));
}
function m(e2, ...n2) {
  return w(() => Object.fromEntries(Object.entries(u(e2)).filter((e3) => !n2.includes(e3[0]))));
}
var O;
const h = "undefined" != typeof window, g = Object.prototype.toString, y = (e2) => "number" == typeof e2, b = (e2) => "string" == typeof e2, I = (e2) => "[object Object]" === g.call(e2), T = () => {
};
function E(e2) {
  return e2;
}
function P(e2, n2 = 200, t2 = true, u2 = true) {
  return function(e3, n3) {
    return function(...t3) {
      e3(() => n3.apply(this, t3), { fn: n3, thisArg: this, args: t3 });
    };
  }(function(e3, n3 = true, t3 = true) {
    let u3, o2 = 0, a2 = true;
    const i2 = () => {
      u3 && (clearTimeout(u3), u3 = void 0);
    };
    return (s2) => {
      const r2 = c(e3), l2 = Date.now() - o2;
      if (i2(), r2 <= 0)
        return o2 = Date.now(), s2();
      l2 > r2 && (t3 || !a2) ? (o2 = Date.now(), s2()) : n3 && (u3 = setTimeout(() => {
        o2 = Date.now(), a2 = true, i2(), s2();
      }, r2)), t3 || u3 || (u3 = setTimeout(() => a2 = true, r2)), a2 = false;
    };
  }(n2, t2, u2), e2);
}
function x(e2, n2 = true) {
  o() ? a(e2) : n2 ? e2() : i(e2);
}
function D(e2) {
  const n2 = c(e2);
  return null != n2 && "$el" in n2 ? n2.$el : n2;
}
h && (null == (O = null == window ? void 0 : window.navigator) ? void 0 : O.userAgent) && /iP(ad|hone|od)/.test(window.navigator.userAgent);
const j = h ? window : void 0;
function Q(...e2) {
  let n2, t2, u2, o2;
  if (b(e2[0]) ? ([t2, u2, o2] = e2, n2 = j) : [n2, t2, u2, o2] = e2, !n2)
    return T;
  let a2 = T;
  const i2 = d(() => D(n2), (e3) => {
    a2(), e3 && (e3.addEventListener(t2, u2, o2), a2 = () => {
      e3.removeEventListener(t2, u2, o2), a2 = T;
    });
  }, { immediate: true, flush: "post" }), s2 = () => {
    i2(), a2();
  };
  return f(s2), s2;
}
function R(e2, n2, t2 = {}) {
  const { window: u2 = j, ignore: o2, capture: a2 = true } = t2;
  if (!u2)
    return;
  const i2 = s(true);
  let r2;
  const l2 = (t3) => {
    u2.clearTimeout(r2);
    const a3 = D(e2), s2 = t3.composedPath();
    a3 && a3 !== t3.target && !s2.includes(a3) && i2.value && (o2 && o2.length > 0 && o2.some((e3) => {
      const n3 = D(e3);
      return n3 && (t3.target === n3 || s2.includes(n3));
    }) || n2(t3));
  }, c2 = [Q(u2, "click", l2, { passive: true, capture: a2 }), Q(u2, "pointerdown", (n3) => {
    const t3 = D(e2);
    i2.value = !!t3 && !n3.composedPath().includes(t3);
  }, { passive: true }), Q(u2, "pointerup", (e3) => {
    if (0 === e3.button) {
      const n3 = e3.composedPath();
      e3.composedPath = () => n3, r2 = u2.setTimeout(() => l2(e3), 50);
    }
  }, { passive: true })];
  return () => c2.forEach((e3) => e3());
}
function S(e2, n2 = null) {
  const t2 = o();
  let u2 = () => {
  };
  const a2 = v((o2, a3) => (u2 = a3, { get() {
    var u3, a4;
    return o2(), null != (a4 = null == (u3 = null == t2 ? void 0 : t2.proxy) ? void 0 : u3.$refs[e2]) ? a4 : n2;
  }, set() {
  } }));
  return x(u2), p(u2), a2;
}
const A = "undefined" != typeof globalThis ? globalThis : "undefined" != typeof window ? window : "undefined" != typeof global ? global : "undefined" != typeof self ? self : {}, B = "__vueuse_ssr_handlers__";
A[B] = A[B] || {}, A[B];
var C, H, N = Object.getOwnPropertySymbols, _ = Object.prototype.hasOwnProperty, k = Object.prototype.propertyIsEnumerable;
function F(e2, n2, t2 = {}) {
  const u2 = t2, { window: o2 = j } = u2, a2 = ((e3, n3) => {
    var t3 = {};
    for (var u3 in e3)
      _.call(e3, u3) && n3.indexOf(u3) < 0 && (t3[u3] = e3[u3]);
    if (null != e3 && N)
      for (var u3 of N(e3))
        n3.indexOf(u3) < 0 && k.call(e3, u3) && (t3[u3] = e3[u3]);
    return t3;
  })(u2, ["window"]);
  let i2;
  const s2 = o2 && "ResizeObserver" in o2, r2 = () => {
    i2 && (i2.disconnect(), i2 = void 0);
  }, l2 = d(() => D(e2), (e3) => {
    r2(), s2 && o2 && e3 && (i2 = new ResizeObserver(n2), i2.observe(e3, a2));
  }, { immediate: true, flush: "post" }), c2 = () => {
    r2(), l2();
  };
  return f(c2), { isSupported: s2, stop: c2 };
}
function G(e2 = {}) {
  const { initialValue: n2 = "" } = e2, t2 = Boolean("undefined" != typeof window && "EyeDropper" in window), u2 = s(n2);
  return { isSupported: t2, sRGBHex: u2, open: async function(e3) {
    if (!t2)
      return;
    const n3 = new window.EyeDropper(), o2 = await n3.open(e3);
    return u2.value = o2.sRGBHex, o2;
  } };
}
function L(e2 = {}) {
  const { touch: n2 = true, drag: u2 = true, initialValue: o2 = false, window: a2 = j } = e2, i2 = s(o2), r2 = s(null);
  if (!a2)
    return { pressed: i2, sourceType: r2 };
  const l2 = (e3) => () => {
    i2.value = true, r2.value = e3;
  }, c2 = () => {
    i2.value = false, r2.value = null;
  }, d2 = t(() => D(e2.target) || a2);
  return Q(d2, "mousedown", l2("mouse"), { passive: true }), Q(a2, "mouseleave", c2, { passive: true }), Q(a2, "mouseup", c2, { passive: true }), u2 && (Q(d2, "dragstart", l2("mouse"), { passive: true }), Q(a2, "drop", c2, { passive: true }), Q(a2, "dragend", c2, { passive: true })), n2 && (Q(d2, "touchstart", l2("touch"), { passive: true }), Q(a2, "touchend", c2, { passive: true }), Q(a2, "touchcancel", c2, { passive: true })), { pressed: i2, sourceType: r2 };
}
(H = C || (C = {})).UP = "UP", H.RIGHT = "RIGHT", H.DOWN = "DOWN", H.LEFT = "LEFT", H.NONE = "NONE";
const W = { linear: E, easeInSine: [0.12, 0, 0.39, 0], easeOutSine: [0.61, 1, 0.88, 1], easeInOutSine: [0.37, 0, 0.63, 1], easeInQuad: [0.11, 0, 0.5, 0], easeOutQuad: [0.5, 1, 0.89, 1], easeInOutQuad: [0.45, 0, 0.55, 1], easeInCubic: [0.32, 0, 0.67, 0], easeOutCubic: [0.33, 1, 0.68, 1], easeInOutCubic: [0.65, 0, 0.35, 1], easeInQuart: [0.5, 0, 0.75, 0], easeOutQuart: [0.25, 1, 0.5, 1], easeInOutQuart: [0.76, 0, 0.24, 1], easeInQuint: [0.64, 0, 0.78, 0], easeOutQuint: [0.22, 1, 0.36, 1], easeInOutQuint: [0.83, 0, 0.17, 1], easeInExpo: [0.7, 0, 0.84, 0], easeOutExpo: [0.16, 1, 0.3, 1], easeInOutExpo: [0.87, 0, 0.13, 1], easeInCirc: [0.55, 0, 1, 0.45], easeOutCirc: [0, 0.55, 0.45, 1], easeInOutCirc: [0.85, 0, 0.15, 1], easeInBack: [0.36, 0, 0.66, -0.56], easeOutBack: [0.34, 1.56, 0.64, 1], easeInOutBack: [0.68, -0.6, 0.32, 1.6] };
function z([e2, n2, t2, u2]) {
  const o2 = (e3, n3) => 1 - 3 * n3 + 3 * e3, a2 = (e3, n3) => 3 * n3 - 6 * e3, i2 = (e3) => 3 * e3, s2 = (e3, n3, t3) => ((o2(n3, t3) * e3 + a2(n3, t3)) * e3 + i2(n3)) * e3;
  return (r2) => e2 === n2 && t2 === u2 ? r2 : s2(((n3) => {
    let u3 = n3;
    for (let d2 = 0; d2 < 4; ++d2) {
      const d3 = (r3 = u3, 3 * o2(l2 = e2, c2 = t2) * r3 * r3 + 2 * a2(l2, c2) * r3 + i2(l2));
      if (0 === d3)
        return u3;
      u3 -= (s2(u3, e2, t2) - n3) / d3;
    }
    var r3, l2, c2;
    return u3;
  })(r2), n2, u2);
}
function $(e2, n2 = {}) {
  const { delay: u2 = 0, disabled: o2 = false, duration: a2 = 1e3, onFinished: i2 = T, onStarted: r2 = T, transition: l2 = E } = n2, v2 = t(() => {
    const e3 = c(l2);
    return "function" == typeof e3 ? e3 : z(e3);
  }), p2 = t(() => {
    const n3 = c(e2);
    return y(n3) ? n3 : n3.map(c);
  }), w2 = t(() => y(p2.value) ? [p2.value] : p2.value), m2 = s(w2.value.slice(0));
  let O2, g2, b2, I2, P2;
  const { resume: x2, pause: D2 } = function(e3, n3 = {}) {
    const { immediate: t2 = true, window: u3 = j } = n3, o3 = s(false);
    let a3 = null;
    function i3() {
      o3.value && u3 && (e3(), a3 = u3.requestAnimationFrame(i3));
    }
    function r3() {
      !o3.value && u3 && (o3.value = true, i3());
    }
    function l3() {
      o3.value = false, null != a3 && u3 && (u3.cancelAnimationFrame(a3), a3 = null);
    }
    return t2 && r3(), f(l3), { isActive: o3, pause: l3, resume: r3 };
  }(() => {
    const e3 = Date.now(), n3 = (t2 = 1 - (b2 - e3) / O2, u3 = 0, o3 = 1, Math.min(o3, Math.max(u3, t2)));
    var t2, u3, o3;
    m2.value = P2.map((e4, t3) => {
      var u4;
      return e4 + (null != (u4 = g2[t3]) ? u4 : 0) * v2.value(n3);
    }), n3 >= 1 && (D2(), i2());
  }, { immediate: false }), Q2 = () => {
    D2(), O2 = c(a2), g2 = m2.value.map((e3, n3) => {
      var t2, u3;
      return (null != (t2 = w2.value[n3]) ? t2 : 0) - (null != (u3 = m2.value[n3]) ? u3 : 0);
    }), P2 = m2.value.slice(0), I2 = Date.now(), b2 = I2 + O2, x2(), r2();
  }, R2 = function(e3, n3, t2 = {}) {
    const { immediate: u3 = true } = t2, o3 = s(false);
    let a3 = null;
    function i3() {
      a3 && (clearTimeout(a3), a3 = null);
    }
    function r3() {
      o3.value = false, i3();
    }
    function l3(...t3) {
      i3(), o3.value = true, a3 = setTimeout(() => {
        o3.value = false, a3 = null, e3(...t3);
      }, c(n3));
    }
    return u3 && (o3.value = true, h && l3()), f(r3), { isPending: o3, start: l3, stop: r3 };
  }(Q2, u2, { immediate: false });
  return d(w2, () => {
    c(o2) ? m2.value = w2.value.slice(0) : c(u2) <= 0 ? Q2() : R2.start();
  }, { deep: true }), t(() => {
    const e3 = c(o2) ? w2 : m2;
    return y(p2.value) ? e3.value[0] : e3.value;
  });
}
function M({ window: e2 = j, initialWidth: n2 = 1 / 0, initialHeight: t2 = 1 / 0 } = {}) {
  const u2 = s(n2), o2 = s(t2), a2 = () => {
    e2 && (u2.value = e2.innerWidth, o2.value = e2.innerHeight);
  };
  return a2(), x(a2), Q("resize", a2, { passive: true }), { width: u2, height: o2 };
}
export { W as T, F as a, P as b, Q as c, G as d, $ as e, L as f, I as i, R as o, m as r, S as t, M as u };
