// @layui/layui-vue/es/_chunks/@umijs/index.js
var e = "undefined" != typeof globalThis ? globalThis : "undefined" != typeof window ? window : "undefined" != typeof global ? global : "undefined" != typeof self ? self : {}, t = { exports: {} };
!function(e2) {
  /*! *****************************************************************************
      Copyright (c) Microsoft Corporation.
  
      Permission to use, copy, modify, and/or distribute this software for any
      purpose with or without fee is hereby granted.
  
      THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
      REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
      AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
      INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
      LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
      OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
      PERFORMANCE OF THIS SOFTWARE.
      ***************************************************************************** */
  var t2 = function() {
    return t2 = Object.assign || function(e3) {
      for (var t3, r2 = 1, n2 = arguments.length; r2 < n2; r2++)
        for (var o2 in t3 = arguments[r2])
          Object.prototype.hasOwnProperty.call(t3, o2) && (e3[o2] = t3[o2]);
      return e3;
    }, t2.apply(this, arguments);
  };
  function r(e3, t3, r2, n2) {
    function o2(e4) {
      return e4 instanceof r2 ? e4 : new r2(function(t4) {
        t4(e4);
      });
    }
    return new (r2 || (r2 = Promise))(function(r3, a2) {
      function i2(e4) {
        try {
          s2(n2.next(e4));
        } catch (e5) {
          a2(e5);
        }
      }
      function u2(e4) {
        try {
          s2(n2.throw(e4));
        } catch (e5) {
          a2(e5);
        }
      }
      function s2(e4) {
        e4.done ? r3(e4.value) : o2(e4.value).then(i2, u2);
      }
      s2((n2 = n2.apply(e3, t3 || [])).next());
    });
  }
  function n(e3, t3) {
    var r2, n2, o2, a2, i2 = { label: 0, sent: function() {
      if (1 & o2[0])
        throw o2[1];
      return o2[1];
    }, trys: [], ops: [] };
    return a2 = { next: u2(0), throw: u2(1), return: u2(2) }, "function" == typeof Symbol && (a2[Symbol.iterator] = function() {
      return this;
    }), a2;
    function u2(e4) {
      return function(t4) {
        return s2([e4, t4]);
      };
    }
    function s2(a3) {
      if (r2)
        throw new TypeError("Generator is already executing.");
      for (; i2; )
        try {
          if (r2 = 1, n2 && (o2 = 2 & a3[0] ? n2.return : a3[0] ? n2.throw || ((o2 = n2.return) && o2.call(n2), 0) : n2.next) && !(o2 = o2.call(n2, a3[1])).done)
            return o2;
          switch (n2 = 0, o2 && (a3 = [2 & a3[0], o2.value]), a3[0]) {
            case 0:
            case 1:
              o2 = a3;
              break;
            case 4:
              return i2.label++, { value: a3[1], done: false };
            case 5:
              i2.label++, n2 = a3[1], a3 = [0];
              continue;
            case 7:
              a3 = i2.ops.pop(), i2.trys.pop();
              continue;
            default:
              if (!((o2 = (o2 = i2.trys).length > 0 && o2[o2.length - 1]) || 6 !== a3[0] && 2 !== a3[0])) {
                i2 = 0;
                continue;
              }
              if (3 === a3[0] && (!o2 || a3[1] > o2[0] && a3[1] < o2[3])) {
                i2.label = a3[1];
                break;
              }
              if (6 === a3[0] && i2.label < o2[1]) {
                i2.label = o2[1], o2 = a3;
                break;
              }
              if (o2 && i2.label < o2[2]) {
                i2.label = o2[2], i2.ops.push(a3);
                break;
              }
              o2[2] && i2.ops.pop(), i2.trys.pop();
              continue;
          }
          a3 = t3.call(e3, i2);
        } catch (e4) {
          a3 = [6, e4], n2 = 0;
        } finally {
          r2 = o2 = 0;
        }
      if (5 & a3[0])
        throw a3[1];
      return { value: a3[0] ? a3[1] : void 0, done: true };
    }
  }
  function o(e3) {
    var t3 = "function" == typeof Symbol && Symbol.iterator, r2 = t3 && e3[t3], n2 = 0;
    if (r2)
      return r2.call(e3);
    if (e3 && "number" == typeof e3.length)
      return { next: function() {
        return e3 && n2 >= e3.length && (e3 = void 0), { value: e3 && e3[n2++], done: !e3 };
      } };
    throw new TypeError(t3 ? "Object is not iterable." : "Symbol.iterator is not defined.");
  }
  function a(e3, t3) {
    var r2 = "function" == typeof Symbol && e3[Symbol.iterator];
    if (!r2)
      return e3;
    var n2, o2, a2 = r2.call(e3), i2 = [];
    try {
      for (; (void 0 === t3 || t3-- > 0) && !(n2 = a2.next()).done; )
        i2.push(n2.value);
    } catch (e4) {
      o2 = { error: e4 };
    } finally {
      try {
        n2 && !n2.done && (r2 = a2.return) && r2.call(a2);
      } finally {
        if (o2)
          throw o2.error;
      }
    }
    return i2;
  }
  function i(e3, t3, r2) {
    if (r2 || 2 === arguments.length)
      for (var n2, o2 = 0, a2 = t3.length; o2 < a2; o2++)
        !n2 && o2 in t3 || (n2 || (n2 = Array.prototype.slice.call(t3, 0, o2)), n2[o2] = t3[o2]);
    return e3.concat(n2 || Array.prototype.slice.call(t3));
  }
  var u = { UI_GET_DATA: "ui-get-data", UI_GET_ACTIVE_TAB_INFO: "ui-get-active-tab-info", UI_SUBSCRIBE_TO_CHANGES: "ui-subscribe-to-changes", UI_UNSUBSCRIBE_FROM_CHANGES: "ui-unsubscribe-from-changes", UI_CHANGE_SETTINGS: "ui-change-settings", UI_SET_THEME: "ui-set-theme", UI_SET_SHORTCUT: "ui-set-shortcut", UI_TOGGLE_URL: "ui-toggle-url", UI_MARK_NEWS_AS_READ: "ui-mark-news-as-read", UI_LOAD_CONFIG: "ui-load-config", UI_APPLY_DEV_DYNAMIC_THEME_FIXES: "ui-apply-dev-dynamic-theme-fixes", UI_RESET_DEV_DYNAMIC_THEME_FIXES: "ui-reset-dev-dynamic-theme-fixes", UI_APPLY_DEV_INVERSION_FIXES: "ui-apply-dev-inversion-fixes", UI_RESET_DEV_INVERSION_FIXES: "ui-reset-dev-inversion-fixes", UI_APPLY_DEV_STATIC_THEMES: "ui-apply-dev-static-themes", UI_RESET_DEV_STATIC_THEMES: "ui-reset-dev-static-themes", UI_SAVE_FILE: "ui-save-file", UI_REQUEST_EXPORT_CSS: "ui-request-export-css", BG_CHANGES: "bg-changes", BG_ADD_CSS_FILTER: "bg-add-css-filter", BG_ADD_STATIC_THEME: "bg-add-static-theme", BG_ADD_SVG_FILTER: "bg-add-svg-filter", BG_ADD_DYNAMIC_THEME: "bg-add-dynamic-theme", BG_EXPORT_CSS: "bg-export-css", BG_UNSUPPORTED_SENDER: "bg-unsupported-sender", BG_CLEAN_UP: "bg-clean-up", BG_RELOAD: "bg-reload", BG_FETCH_RESPONSE: "bg-fetch-response", BG_UI_UPDATE: "bg-ui-update", BG_CSS_UPDATE: "bg-css-update", CS_COLOR_SCHEME_CHANGE: "cs-color-scheme-change", CS_FRAME_CONNECT: "cs-frame-connect", CS_FRAME_FORGET: "cs-frame-forget", CS_FRAME_FREEZE: "cs-frame-freeze", CS_FRAME_RESUME: "cs-frame-resume", CS_EXPORT_CSS_RESPONSE: "cs-export-css-response", CS_FETCH: "cs-fetch" }, s = "undefined" == typeof navigator ? "some useragent" : navigator.userAgent.toLowerCase(), c = "undefined" == typeof navigator ? "some platform" : navigator.platform.toLowerCase(), l = s.includes("chrome") || s.includes("chromium"), d = s.includes("thunderbird"), f = s.includes("firefox") || s.includes("librewolf") || d;
  s.includes("vivaldi"), s.includes("yabrowser"), s.includes("opr") || s.includes("opera"), s.includes("edg");
  var h = s.includes("safari") && !l, p = c.startsWith("win"), v = c.startsWith("mac");
  s.includes("mobile");
  var m, g = "function" == typeof ShadowRoot, b = "function" == typeof MediaQueryList && "function" == typeof MediaQueryList.prototype.addEventListener;
  (m = s.match(/chrom[e|ium]\/([^ ]+)/)) && m[1] && m[1];
  var y = function() {
    try {
      return document.querySelector(":defined"), true;
    } catch (e3) {
      return false;
    }
  }();
  function S(e3, t3, o2) {
    return r(this, void 0, void 0, function() {
      var r2;
      return n(this, function(n2) {
        switch (n2.label) {
          case 0:
            return [4, fetch(e3, { cache: "force-cache", credentials: "omit", referrer: o2 })];
          case 1:
            if (r2 = n2.sent(), f && "text/css" === t3 && e3.startsWith("moz-extension://") && e3.endsWith(".css"))
              return [2, r2];
            if (t3 && !r2.headers.get("Content-Type").startsWith(t3))
              throw new Error("Mime type mismatch when loading " + e3);
            if (!r2.ok)
              throw new Error("Unable to load " + e3 + " " + r2.status + " " + r2.statusText);
            return [2, r2];
        }
      });
    });
  }
  function _(e3, t3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(r2) {
        switch (r2.label) {
          case 0:
            return [4, S(e3, t3)];
          case 1:
            return [4, k(r2.sent())];
          case 2:
            return [2, r2.sent()];
        }
      });
    });
  }
  function k(e3) {
    return r(this, void 0, void 0, function() {
      var t3;
      return n(this, function(r2) {
        switch (r2.label) {
          case 0:
            return [4, e3.blob()];
          case 1:
            return t3 = r2.sent(), [4, new Promise(function(e4) {
              var r3 = new FileReader();
              r3.onloadend = function() {
                return e4(r3.result);
              }, r3.readAsDataURL(t3);
            })];
          case 2:
            return [2, r2.sent()];
        }
      });
    });
  }
  "undefined" != typeof globalThis && globalThis.chrome && globalThis.chrome.runtime && globalThis.chrome.runtime.getManifest && globalThis.chrome.runtime.getManifest().manifest_version;
  var w = function(e3) {
    return r(void 0, void 0, void 0, function() {
      return n(this, function(t3) {
        return [2, Promise.reject(new Error(["Embedded Dark Reader cannot access a cross-origin resource", e3, "Overview your URLs and CORS policies or use", "`DarkReader.setFetchMethod(fetch: (url) => Promise<Response>))`.", "See if using `DarkReader.setFetchMethod(window.fetch)`", "before `DarkReader.enable()` works."].join(" ")))];
      });
    });
  }, E = w;
  function x(e3) {
    E = e3 || w;
  }
  function C(e3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(t3) {
        switch (t3.label) {
          case 0:
            return [4, E(e3)];
          case 1:
            return [2, t3.sent()];
        }
      });
    });
  }
  function T() {
    if ("undefined" != typeof chrome) {
      "undefined" == typeof window || window.chrome || (window.chrome = {}), "undefined" == typeof chrome || chrome.runtime || (chrome.runtime = {});
      var e3 = /* @__PURE__ */ new Set();
      if ("function" == typeof chrome.runtime.sendMessage) {
        var t3 = chrome.runtime.sendMessage;
        chrome.runtime.sendMessage = function() {
          for (var e4 = [], r2 = 0; r2 < arguments.length; r2++)
            e4[r2] = arguments[r2];
          s2.apply(void 0, i([], a(e4), false)), t3.apply(chrome.runtime, e4);
        };
      } else
        chrome.runtime.sendMessage = s2;
      if (chrome.runtime.onMessage || (chrome.runtime.onMessage = {}), "function" == typeof chrome.runtime.onMessage.addListener) {
        var o2 = chrome.runtime.onMessage.addListener;
        chrome.runtime.onMessage.addListener = function() {
          for (var e4 = [], t4 = 0; t4 < arguments.length; t4++)
            e4[t4] = arguments[t4];
          c2(e4[0]), o2.apply(chrome.runtime.onMessage, e4);
        };
      } else
        chrome.runtime.onMessage.addListener = function() {
          for (var e4 = [], t4 = 0; t4 < arguments.length; t4++)
            e4[t4] = arguments[t4];
          return c2(e4[0]);
        };
    }
    function s2() {
      for (var t4 = [], o3 = 0; o3 < arguments.length; o3++)
        t4[o3] = arguments[o3];
      return r(this, void 0, void 0, function() {
        var r2, o4, a2, i2, s3, c3, l2;
        return n(this, function(n2) {
          switch (n2.label) {
            case 0:
              if (!t4[0] || t4[0].type !== u.CS_FETCH)
                return [3, 8];
              r2 = t4[0].id, n2.label = 1;
            case 1:
              return n2.trys.push([1, 7, , 8]), o4 = t4[0].data, a2 = o4.url, i2 = o4.responseType, [4, C(a2)];
            case 2:
              return s3 = n2.sent(), "data-url" !== i2 ? [3, 4] : [4, k(s3)];
            case 3:
              return c3 = n2.sent(), [3, 6];
            case 4:
              return [4, s3.text()];
            case 5:
              c3 = n2.sent(), n2.label = 6;
            case 6:
              return e3.forEach(function(e4) {
                return e4({ type: u.BG_FETCH_RESPONSE, data: c3, error: null, id: r2 });
              }), [3, 8];
            case 7:
              return l2 = n2.sent(), console.error(l2), e3.forEach(function(e4) {
                return e4({ type: u.BG_FETCH_RESPONSE, data: null, error: l2, id: r2 });
              }), [3, 8];
            case 8:
              return [2];
          }
        });
      });
    }
    function c2(t4) {
      e3.add(t4);
    }
  }
  T();
  var R = { cssFilter: "cssFilter", svgFilter: "svgFilter", staticTheme: "staticTheme", dynamicTheme: "dynamicTheme" }, V = { darkScheme: { background: "#181a1b", text: "#e8e6e3" }, lightScheme: { background: "#dcdad7", text: "#181a1b" } }, M = { mode: 1, brightness: 100, contrast: 100, grayscale: 0, sepia: 0, useFont: false, fontFamily: v ? "Helvetica Neue" : p ? "Segoe UI" : "Open Sans", textStroke: 0, engine: R.dynamicTheme, stylesheet: "", darkSchemeBackgroundColor: V.darkScheme.background, darkSchemeTextColor: V.darkScheme.text, lightSchemeBackgroundColor: V.lightScheme.background, lightSchemeTextColor: V.lightScheme.text, scrollbarColor: v ? "" : "auto", selectionColor: "auto", styleSystemControls: true, lightColorScheme: "Default", darkColorScheme: "Default" };
  function A(e3) {
    return null != e3.length;
  }
  function L(e3, t3) {
    var r2, n2;
    if (A(e3))
      for (var a2 = 0, i2 = e3.length; a2 < i2; a2++)
        t3(e3[a2]);
    else
      try {
        for (var u2 = o(e3), s2 = u2.next(); !s2.done; s2 = u2.next())
          t3(s2.value);
      } catch (e4) {
        r2 = { error: e4 };
      } finally {
        try {
          s2 && !s2.done && (n2 = u2.return) && n2.call(u2);
        } finally {
          if (r2)
            throw r2.error;
        }
      }
  }
  function P(e3, t3) {
    L(t3, function(t4) {
      return e3.push(t4);
    });
  }
  function O(e3) {
    for (var t3 = [], r2 = 0, n2 = e3.length; r2 < n2; r2++)
      t3.push(e3[r2]);
    return t3;
  }
  function D() {
  }
  function I() {
  }
  function F(e3) {
    var t3, r2 = false, n2 = null, o2 = function() {
      for (var o3 = [], u3 = 0; u3 < arguments.length; u3++)
        o3[u3] = arguments[u3];
      t3 = o3, n2 ? r2 = true : (e3.apply(void 0, i([], a(t3), false)), n2 = requestAnimationFrame(function() {
        n2 = null, r2 && (e3.apply(void 0, i([], a(t3), false)), r2 = false);
      }));
    }, u2 = function() {
      cancelAnimationFrame(n2), r2 = false, n2 = null;
    };
    return Object.assign(o2, { cancel: u2 });
  }
  function N() {
    var e3 = [], t3 = null;
    function r2() {
      for (var r3; r3 = e3.shift(); )
        r3();
      t3 = null;
    }
    function n2(n3) {
      e3.push(n3), t3 || (t3 = requestAnimationFrame(r2));
    }
    function o2() {
      e3.splice(0), cancelAnimationFrame(t3), t3 = null;
    }
    return { add: n2, cancel: o2 };
  }
  function j(e3) {
    var t3 = 0;
    return e3.seconds && (t3 += 1e3 * e3.seconds), e3.minutes && (t3 += 60 * e3.minutes * 1e3), e3.hours && (t3 += 60 * e3.hours * 60 * 1e3), e3.days && (t3 += 24 * e3.days * 60 * 60 * 1e3), t3;
  }
  function B(e3) {
    e3 && e3.parentNode && e3.parentNode.removeChild(e3);
  }
  function U(e3, t3, r2) {
    void 0 === r2 && (r2 = Function.prototype);
    var n2 = 10, o2 = j({ seconds: 2 }), a2 = j({ seconds: 10 }), i2 = e3.previousSibling, u2 = e3.parentNode;
    if (!u2)
      throw new Error("Unable to watch for node position: parent element not found");
    if ("prev-sibling" === t3 && !i2)
      throw new Error("Unable to watch for node position: there is no previous sibling");
    var s2 = 0, c2 = null, l2 = null, d2 = F(function() {
      if (!l2) {
        s2++;
        var h3 = Date.now();
        if (null == c2)
          c2 = h3;
        else if (s2 >= n2) {
          if (h3 - c2 < a2)
            return void (l2 = setTimeout(function() {
              c2 = null, s2 = 0, l2 = null, d2();
            }, o2));
          c2 = h3, s2 = 1;
        }
        if ("parent" === t3 && i2 && i2.parentNode !== u2)
          p2();
        else {
          if ("prev-sibling" === t3) {
            if (null == i2.parentNode)
              return void p2();
            i2.parentNode !== u2 && m2(i2.parentNode);
          }
          u2.insertBefore(e3, i2 ? i2.nextSibling : u2.firstChild), f2.takeRecords(), r2 && r2();
        }
      }
    }), f2 = new MutationObserver(function() {
      ("parent" === t3 && e3.parentNode !== u2 || "prev-sibling" === t3 && e3.previousSibling !== i2) && d2();
    }), h2 = function() {
      f2.observe(u2, { childList: true });
    }, p2 = function() {
      clearTimeout(l2), f2.disconnect(), d2.cancel();
    }, v2 = function() {
      f2.takeRecords();
    }, m2 = function(e4) {
      u2 = e4, p2(), h2();
    };
    return h2(), { run: h2, stop: p2, skip: v2 };
  }
  function q(e3, t3) {
    if (null != e3)
      for (var r2 = document.createTreeWalker(e3, NodeFilter.SHOW_ELEMENT, { acceptNode: function(e4) {
        return null == e4.shadowRoot ? NodeFilter.FILTER_SKIP : NodeFilter.FILTER_ACCEPT;
      } }), n2 = e3.shadowRoot ? r2.currentNode : r2.nextNode(); null != n2; n2 = r2.nextNode())
        t3(n2), q(n2.shadowRoot, t3);
  }
  function W() {
    return "undefined" != typeof document && ("complete" === document.readyState || "interactive" === document.readyState);
  }
  var H = /* @__PURE__ */ new Set();
  function G(e3) {
    H.add(e3);
  }
  function $(e3) {
    H.delete(e3);
  }
  function z() {
    return "complete" === document.readyState;
  }
  var Q = /* @__PURE__ */ new Set();
  function K(e3) {
    Q.add(e3);
  }
  function X() {
    Q.clear();
  }
  if (!W()) {
    var Y = function() {
      W() && (H.forEach(function(e3) {
        return e3();
      }), H.clear(), z() && (document.removeEventListener("readystatechange", Y), Q.forEach(function(e3) {
        return e3();
      }), Q.clear()));
    };
    "undefined" != typeof document && document.addEventListener("readystatechange", Y);
  }
  var Z = 1e3;
  function J(e3) {
    if (e3.length > Z)
      return true;
    for (var t3 = 0, r2 = 0; r2 < e3.length; r2++)
      if ((t3 += e3[r2].addedNodes.length) > Z)
        return true;
    return false;
  }
  function ee(e3) {
    var t3 = /* @__PURE__ */ new Set(), r2 = /* @__PURE__ */ new Set(), n2 = /* @__PURE__ */ new Set();
    e3.forEach(function(e4) {
      L(e4.addedNodes, function(e5) {
        e5 instanceof Element && e5.isConnected && t3.add(e5);
      }), L(e4.removedNodes, function(e5) {
        e5 instanceof Element && (e5.isConnected ? (n2.add(e5), t3.delete(e5)) : r2.add(e5));
      });
    });
    var o2 = [], a2 = [];
    return t3.forEach(function(e4) {
      t3.has(e4.parentElement) && o2.push(e4);
    }), r2.forEach(function(e4) {
      r2.has(e4.parentElement) && a2.push(e4);
    }), o2.forEach(function(e4) {
      return t3.delete(e4);
    }), a2.forEach(function(e4) {
      return r2.delete(e4);
    }), { additions: t3, moves: n2, deletions: r2 };
  }
  var te, re = /* @__PURE__ */ new Map(), ne = /* @__PURE__ */ new WeakMap();
  function oe(e3, t3) {
    var r2, n2, o2;
    if (re.has(e3))
      r2 = re.get(e3), n2 = ne.get(r2);
    else {
      var a2 = false, i2 = false;
      (r2 = new MutationObserver(function(t4) {
        if (J(t4))
          !a2 || W() ? n2.forEach(function(t5) {
            return (0, t5.onHugeMutations)(e3);
          }) : i2 || (G(o2 = function() {
            return n2.forEach(function(t5) {
              return (0, t5.onHugeMutations)(e3);
            });
          }), i2 = true), a2 = true;
        else {
          var r3 = ee(t4);
          n2.forEach(function(e4) {
            return (0, e4.onMinorMutations)(r3);
          });
        }
      })).observe(e3, { childList: true, subtree: true }), re.set(e3, r2), n2 = /* @__PURE__ */ new Set(), ne.set(r2, n2);
    }
    return n2.add(t3), { disconnect: function() {
      n2.delete(t3), o2 && $(o2), 0 === n2.size && (r2.disconnect(), ne.delete(r2), re.delete(e3));
    } };
  }
  var ae = /* @__PURE__ */ new Map();
  function ie(e3) {
    return te || (te = document.createElement("a")), te.href = e3, te.href;
  }
  function ue(e3, t3) {
    void 0 === t3 && (t3 = null);
    var r2 = e3 + (t3 ? ";" + t3 : "");
    if (ae.has(r2))
      return ae.get(r2);
    if (t3) {
      var n2 = new URL(e3, ie(t3));
      return ae.set(r2, n2), n2;
    }
    var o2 = new URL(ie(e3));
    return ae.set(e3, o2), o2;
  }
  function se(e3, t3) {
    if (t3.match(/^data\\?\:/))
      return t3;
    if (/^\/\//.test(t3))
      return "" + location.protocol + t3;
    var r2 = ue(e3);
    return ue(t3, r2.href).href;
  }
  function ce(e3) {
    if (e3.startsWith("data:"))
      return true;
    var t3 = ue(e3);
    return t3.protocol === location.protocol && t3.hostname === location.hostname && t3.port === location.port && t3.pathname === location.pathname;
  }
  function le(e3, t3, r2) {
    L(e3, function(e4) {
      if (e4.selectorText)
        t3(e4);
      else if (e4.href)
        try {
          le(e4.styleSheet.cssRules, t3, r2);
        } catch (e5) {
          D("Found a non-loaded link."), r2 && r2();
        }
      else if (e4.media) {
        var n2 = Array.from(e4.media), o2 = n2.some(function(e5) {
          return e5.startsWith("screen") || e5.startsWith("all") || e5.startsWith("(");
        }), a2 = n2.some(function(e5) {
          return e5.startsWith("print") || e5.startsWith("speech");
        });
        !o2 && a2 || le(e4.cssRules, t3, r2);
      } else
        e4.conditionText ? CSS.supports(e4.conditionText) && le(e4.cssRules, t3, r2) : I("CSSRule type not supported", e4);
    });
  }
  var de = ["background", "border", "border-color", "border-bottom", "border-left", "border-right", "border-top", "outline", "outline-color"], fe = h ? de.map(function(e3) {
    return [e3, new RegExp(e3 + ":\\s*(.*?)\\s*;")];
  }) : null;
  function he(e3, t3) {
    L(e3, function(r3) {
      var n2 = e3.getPropertyValue(r3).trim();
      n2 && t3(r3, n2);
    });
    var r2 = e3.cssText;
    r2.includes("var(") && (h ? fe.forEach(function(e4) {
      var n2 = a(e4, 2), o2 = n2[0], i2 = n2[1], u2 = r2.match(i2);
      if (u2 && u2[1]) {
        var s2 = u2[1].trim();
        t3(o2, s2);
      }
    }) : de.forEach(function(r3) {
      var n2 = e3.getPropertyValue(r3);
      n2 && n2.includes("var(") && t3(r3, n2);
    }));
  }
  var pe = /url\((('.+?')|(".+?")|([^\)]*?))\)/g, ve = /@import\s*(url\()?(('.+?')|(".+?")|([^\)]*?))\)? ?(screen)?;?/g;
  function me(e3) {
    return e3.replace(/^url\((.*)\)$/, "$1").trim().replace(/^"(.*)"$/, "$1").replace(/^'(.*)'$/, "$1");
  }
  function ge(e3) {
    var t3 = ue(e3);
    return "" + t3.origin + t3.pathname.replace(/\?.*$/, "").replace(/(\/)([^\/]+)$/i, "$1");
  }
  function be(e3, t3) {
    return e3.replace(pe, function(e4) {
      var r2 = me(e4);
      try {
        return 'url("' + se(t3, r2) + '")';
      } catch (t4) {
        return e4;
      }
    });
  }
  var ye = /\/\*[\s\S]*?\*\//g;
  function Se(e3) {
    return e3.replace(ye, "");
  }
  var _e = /@font-face\s*{[^}]*}/g;
  function ke(e3) {
    return e3.replace(_e, "");
  }
  function we(e3) {
    var t3 = e3.h, r2 = e3.s, n2 = e3.l, o2 = e3.a, i2 = void 0 === o2 ? 1 : o2;
    if (0 === r2) {
      var u2 = a([n2, n2, n2].map(function(e4) {
        return Math.round(255 * e4);
      }), 3), s2 = u2[0], c2 = u2[1];
      return { r: s2, g: u2[2], b: c2, a: i2 };
    }
    var l2 = (1 - Math.abs(2 * n2 - 1)) * r2, d2 = l2 * (1 - Math.abs(t3 / 60 % 2 - 1)), f2 = n2 - l2 / 2, h2 = a((t3 < 60 ? [l2, d2, 0] : t3 < 120 ? [d2, l2, 0] : t3 < 180 ? [0, l2, d2] : t3 < 240 ? [0, d2, l2] : t3 < 300 ? [d2, 0, l2] : [l2, 0, d2]).map(function(e4) {
      return Math.round(255 * (e4 + f2));
    }), 3);
    return { r: h2[0], g: h2[1], b: h2[2], a: i2 };
  }
  function Ee(e3) {
    var t3 = e3.r, r2 = e3.g, n2 = e3.b, o2 = e3.a, a2 = void 0 === o2 ? 1 : o2, i2 = t3 / 255, u2 = r2 / 255, s2 = n2 / 255, c2 = Math.max(i2, u2, s2), l2 = Math.min(i2, u2, s2), d2 = c2 - l2, f2 = (c2 + l2) / 2;
    if (0 === d2)
      return { h: 0, s: 0, l: f2, a: a2 };
    var h2 = 60 * (c2 === i2 ? (u2 - s2) / d2 % 6 : c2 === u2 ? (s2 - i2) / d2 + 2 : (i2 - u2) / d2 + 4);
    return h2 < 0 && (h2 += 360), { h: h2, s: d2 / (1 - Math.abs(2 * f2 - 1)), l: f2, a: a2 };
  }
  function xe(e3, t3) {
    void 0 === t3 && (t3 = 0);
    var r2 = e3.toFixed(t3);
    if (0 === t3)
      return r2;
    var n2 = r2.indexOf(".");
    if (n2 >= 0) {
      var o2 = r2.match(/0+$/);
      if (o2)
        return o2.index === n2 + 1 ? r2.substring(0, n2) : r2.substring(0, o2.index);
    }
    return r2;
  }
  function Ce(e3) {
    var t3 = e3.r, r2 = e3.g, n2 = e3.b, o2 = e3.a;
    return null != o2 && o2 < 1 ? "rgba(" + xe(t3) + ", " + xe(r2) + ", " + xe(n2) + ", " + xe(o2, 2) + ")" : "rgb(" + xe(t3) + ", " + xe(r2) + ", " + xe(n2) + ")";
  }
  function Te(e3) {
    var t3 = e3.r, r2 = e3.g, n2 = e3.b, o2 = e3.a;
    return "#" + (null != o2 && o2 < 1 ? [t3, r2, n2, Math.round(255 * o2)] : [t3, r2, n2]).map(function(e4) {
      return (e4 < 16 ? "0" : "") + e4.toString(16);
    }).join("");
  }
  function Re(e3) {
    var t3 = e3.h, r2 = e3.s, n2 = e3.l, o2 = e3.a;
    return null != o2 && o2 < 1 ? "hsla(" + xe(t3) + ", " + xe(100 * r2) + "%, " + xe(100 * n2) + "%, " + xe(o2, 2) + ")" : "hsl(" + xe(t3) + ", " + xe(100 * r2) + "%, " + xe(100 * n2) + "%)";
  }
  var Ve = /^rgba?\([^\(\)]+\)$/, Me = /^hsla?\([^\(\)]+\)$/, Ae = /^#[0-9a-f]+$/i;
  function Le(e3) {
    var t3 = e3.trim().toLowerCase();
    if (t3.match(Ve))
      return Fe(t3);
    if (t3.match(Me))
      return Be(t3);
    if (t3.match(Ae))
      return Ue(t3);
    if (ze.has(t3))
      return qe(t3);
    if (Qe.has(t3))
      return We(t3);
    if ("transparent" === e3)
      return { r: 0, g: 0, b: 0, a: 0 };
    throw new Error("Unable to parse " + e3);
  }
  function Pe(e3) {
    var t3 = [], r2 = 0, n2 = false, o2 = e3.indexOf("(");
    e3 = e3.substring(o2 + 1, e3.length - 1);
    for (var a2 = 0; a2 < e3.length; a2++) {
      var i2 = e3[a2];
      i2 >= "0" && i2 <= "9" || "." === i2 || "+" === i2 || "-" === i2 ? n2 = true : !n2 || " " !== i2 && "," !== i2 ? n2 || (r2 = a2 + 1) : (t3.push(e3.substring(r2, a2)), n2 = false, r2 = a2 + 1);
    }
    return n2 && t3.push(e3.substring(r2, e3.length)), t3;
  }
  function Oe(e3, t3, r2) {
    var n2 = Pe(e3), o2 = Object.entries(r2);
    return n2.map(function(e4) {
      return e4.trim();
    }).map(function(e4, r3) {
      var n3, i2 = o2.find(function(t4) {
        var r4 = a(t4, 1)[0];
        return e4.endsWith(r4);
      });
      return n3 = i2 ? parseFloat(e4.substring(0, e4.length - i2[0].length)) / i2[1] * t3[r3] : parseFloat(e4), t3[r3] > 1 ? Math.round(n3) : n3;
    });
  }
  var De = [255, 255, 255, 1], Ie = { "%": 100 };
  function Fe(e3) {
    var t3 = a(Oe(e3, De, Ie), 4), r2 = t3[0], n2 = t3[1], o2 = t3[2], i2 = t3[3];
    return { r: r2, g: n2, b: o2, a: void 0 === i2 ? 1 : i2 };
  }
  var Ne = [360, 1, 1, 1], je = { "%": 100, deg: 360, rad: 2 * Math.PI, turn: 1 };
  function Be(e3) {
    var t3 = a(Oe(e3, Ne, je), 4), r2 = t3[0], n2 = t3[1], o2 = t3[2], i2 = t3[3];
    return we({ h: r2, s: n2, l: o2, a: void 0 === i2 ? 1 : i2 });
  }
  function Ue(e3) {
    var t3 = e3.substring(1);
    switch (t3.length) {
      case 3:
      case 4:
        var r2 = a([0, 1, 2].map(function(e4) {
          return parseInt("" + t3[e4] + t3[e4], 16);
        }), 3);
        return { r: r2[0], g: r2[1], b: r2[2], a: 3 === t3.length ? 1 : parseInt("" + t3[3] + t3[3], 16) / 255 };
      case 6:
      case 8:
        var n2 = a([0, 2, 4].map(function(e4) {
          return parseInt(t3.substring(e4, e4 + 2), 16);
        }), 3);
        return { r: n2[0], g: n2[1], b: n2[2], a: 6 === t3.length ? 1 : parseInt(t3.substring(6, 8), 16) / 255 };
    }
    throw new Error("Unable to parse " + e3);
  }
  function qe(e3) {
    var t3 = ze.get(e3);
    return { r: t3 >> 16 & 255, g: t3 >> 8 & 255, b: t3 >> 0 & 255, a: 1 };
  }
  function We(e3) {
    var t3 = Qe.get(e3);
    return { r: t3 >> 16 & 255, g: t3 >> 8 & 255, b: t3 >> 0 & 255, a: 1 };
  }
  var He = function(e3) {
    return e3 >= "0" && e3 <= "9";
  }, Ge = function(e3) {
    return Math.floor(Math.log10(e3)) + 1;
  };
  function $e(e3) {
    for (var t3 = 0, r2 = function(t4, r3, n3) {
      e3 = e3.substring(0, t4) + n3 + e3.substring(r3);
    }, n2 = function() {
      for (var r3 = 0, n3 = 1; n3 < 4; n3++) {
        var o3 = e3[t3 + n3];
        if (" " === o3)
          break;
        if (!He(o3))
          break;
        r3 *= 10, r3 += Number(o3);
      }
      var a3 = Ge(r3);
      if ("%" === e3[(t3 += a3) + 1])
        return t3++, r3;
    }; 0 !== (t3 = e3.indexOf("calc(")); ) {
      var o2 = t3;
      t3 += 4;
      var a2 = n2();
      if (!a2)
        break;
      if (" " !== e3[t3 + 1])
        break;
      t3++;
      var i2 = e3[t3 + 1];
      if ("+" !== i2 && "-" !== i2)
        break;
      if (t3++, " " !== e3[t3 + 1])
        break;
      t3++;
      var u2 = n2();
      if (!u2)
        break;
      r2(o2, t3 + 2, "+" === i2 ? a2 + u2 + "%" : a2 - u2 + "%");
    }
    return e3;
  }
  var ze = new Map(Object.entries({ aliceblue: 15792383, antiquewhite: 16444375, aqua: 65535, aquamarine: 8388564, azure: 15794175, beige: 16119260, bisque: 16770244, black: 0, blanchedalmond: 16772045, blue: 255, blueviolet: 9055202, brown: 10824234, burlywood: 14596231, cadetblue: 6266528, chartreuse: 8388352, chocolate: 13789470, coral: 16744272, cornflowerblue: 6591981, cornsilk: 16775388, crimson: 14423100, cyan: 65535, darkblue: 139, darkcyan: 35723, darkgoldenrod: 12092939, darkgray: 11119017, darkgrey: 11119017, darkgreen: 25600, darkkhaki: 12433259, darkmagenta: 9109643, darkolivegreen: 5597999, darkorange: 16747520, darkorchid: 10040012, darkred: 9109504, darksalmon: 15308410, darkseagreen: 9419919, darkslateblue: 4734347, darkslategray: 3100495, darkslategrey: 3100495, darkturquoise: 52945, darkviolet: 9699539, deeppink: 16716947, deepskyblue: 49151, dimgray: 6908265, dimgrey: 6908265, dodgerblue: 2003199, firebrick: 11674146, floralwhite: 16775920, forestgreen: 2263842, fuchsia: 16711935, gainsboro: 14474460, ghostwhite: 16316671, gold: 16766720, goldenrod: 14329120, gray: 8421504, grey: 8421504, green: 32768, greenyellow: 11403055, honeydew: 15794160, hotpink: 16738740, indianred: 13458524, indigo: 4915330, ivory: 16777200, khaki: 15787660, lavender: 15132410, lavenderblush: 16773365, lawngreen: 8190976, lemonchiffon: 16775885, lightblue: 11393254, lightcoral: 15761536, lightcyan: 14745599, lightgoldenrodyellow: 16448210, lightgray: 13882323, lightgrey: 13882323, lightgreen: 9498256, lightpink: 16758465, lightsalmon: 16752762, lightseagreen: 2142890, lightskyblue: 8900346, lightslategray: 7833753, lightslategrey: 7833753, lightsteelblue: 11584734, lightyellow: 16777184, lime: 65280, limegreen: 3329330, linen: 16445670, magenta: 16711935, maroon: 8388608, mediumaquamarine: 6737322, mediumblue: 205, mediumorchid: 12211667, mediumpurple: 9662683, mediumseagreen: 3978097, mediumslateblue: 8087790, mediumspringgreen: 64154, mediumturquoise: 4772300, mediumvioletred: 13047173, midnightblue: 1644912, mintcream: 16121850, mistyrose: 16770273, moccasin: 16770229, navajowhite: 16768685, navy: 128, oldlace: 16643558, olive: 8421376, olivedrab: 7048739, orange: 16753920, orangered: 16729344, orchid: 14315734, palegoldenrod: 15657130, palegreen: 10025880, paleturquoise: 11529966, palevioletred: 14381203, papayawhip: 16773077, peachpuff: 16767673, peru: 13468991, pink: 16761035, plum: 14524637, powderblue: 11591910, purple: 8388736, rebeccapurple: 6697881, red: 16711680, rosybrown: 12357519, royalblue: 4286945, saddlebrown: 9127187, salmon: 16416882, sandybrown: 16032864, seagreen: 3050327, seashell: 16774638, sienna: 10506797, silver: 12632256, skyblue: 8900331, slateblue: 6970061, slategray: 7372944, slategrey: 7372944, snow: 16775930, springgreen: 65407, steelblue: 4620980, tan: 13808780, teal: 32896, thistle: 14204888, tomato: 16737095, turquoise: 4251856, violet: 15631086, wheat: 16113331, white: 16777215, whitesmoke: 16119285, yellow: 16776960, yellowgreen: 10145074 })), Qe = new Map(Object.entries({ ActiveBorder: 3906044, ActiveCaption: 0, AppWorkspace: 11184810, Background: 6513614, ButtonFace: 16777215, ButtonHighlight: 15329769, ButtonShadow: 10461343, ButtonText: 0, CaptionText: 0, GrayText: 8355711, Highlight: 11720703, HighlightText: 0, InactiveBorder: 16777215, InactiveCaption: 16777215, InactiveCaptionText: 0, InfoBackground: 16514245, InfoText: 0, Menu: 16185078, MenuText: 16777215, Scrollbar: 11184810, ThreeDDarkShadow: 0, ThreeDFace: 12632256, ThreeDHighlight: 16777215, ThreeDLightShadow: 16777215, ThreeDShadow: 0, Window: 15527148, WindowFrame: 11184810, WindowText: 0, "-webkit-focus-ring-color": 15046400 }).map(function(e3) {
    var t3 = a(e3, 2), r2 = t3[0], n2 = t3[1];
    return [r2.toLowerCase(), n2];
  }));
  function Ke(e3, t3, r2, n2, o2) {
    return (e3 - t3) * (o2 - n2) / (r2 - t3) + n2;
  }
  function Xe(e3, t3, r2) {
    return Math.min(r2, Math.max(t3, e3));
  }
  function Ye(e3, t3) {
    for (var r2 = [], n2 = 0, o2 = e3.length; n2 < o2; n2++) {
      r2[n2] = [];
      for (var a2 = 0, i2 = t3[0].length; a2 < i2; a2++) {
        for (var u2 = 0, s2 = 0, c2 = e3[0].length; s2 < c2; s2++)
          u2 += e3[n2][s2] * t3[s2][a2];
        r2[n2][a2] = u2;
      }
    }
    return r2;
  }
  function Ze(e3, t3, r2) {
    void 0 === r2 && (r2 = 0);
    for (var n2, o2 = []; n2 = e3.exec(t3); )
      o2.push(n2[r2]);
    return o2;
  }
  function Je(e3) {
    function t3(e4) {
      return e4.replace(/^\s+/, "");
    }
    function r2(e4) {
      return 0 === e4 ? "" : " ".repeat(4 * e4);
    }
    if (e3.length < 5e4)
      for (var n2 = /[^{}]+{\s*}/; n2.test(e3); )
        e3 = e3.replace(n2, "");
    for (var o2 = e3.replace(/\s{2,}/g, " ").replace(/\{/g, "{\n").replace(/\}/g, "\n}\n").replace(/\;(?![^\(|\"]*(\)|\"))/g, ";\n").replace(/\,(?![^\(|\"]*(\)|\"))/g, ",\n").replace(/\n\s*\n/g, "\n").split("\n"), a2 = 0, i2 = [], u2 = 0, s2 = o2.length; u2 < s2; u2++) {
      var c2 = o2[u2] + "\n";
      c2.includes("{") ? i2.push(r2(a2++) + t3(c2)) : c2.includes("}") ? i2.push(r2(--a2) + t3(c2)) : i2.push(r2(a2) + t3(c2));
    }
    return i2.join("").trim();
  }
  function et(e3, t3) {
    void 0 === t3 && (t3 = 0);
    for (var r2 = e3.length, n2 = 0, o2 = -1, a2 = t3; a2 < r2; a2++)
      if (0 === n2) {
        if ((i2 = e3.indexOf("(", a2)) < 0)
          break;
        o2 = i2, n2++, a2 = i2;
      } else {
        var i2, u2 = e3.indexOf(")", a2);
        if (u2 < 0)
          break;
        if ((i2 = e3.indexOf("(", a2)) < 0 || u2 < i2) {
          if (0 == --n2)
            return { start: o2, end: u2 + 1 };
          a2 = u2;
        } else
          n2++, a2 = i2;
      }
    return null;
  }
  function tt(e3) {
    var t3 = nt.identity();
    return 0 !== e3.sepia && (t3 = Ye(t3, nt.sepia(e3.sepia / 100))), 0 !== e3.grayscale && (t3 = Ye(t3, nt.grayscale(e3.grayscale / 100))), 100 !== e3.contrast && (t3 = Ye(t3, nt.contrast(e3.contrast / 100))), 100 !== e3.brightness && (t3 = Ye(t3, nt.brightness(e3.brightness / 100))), 1 === e3.mode && (t3 = Ye(t3, nt.invertNHue())), t3;
  }
  function rt(e3, t3) {
    var r2 = a(e3, 3), n2 = Ye(t3, [[r2[0] / 255], [r2[1] / 255], [r2[2] / 255], [1], [1]]);
    return [0, 1, 2].map(function(e4) {
      return Xe(Math.round(255 * n2[e4][0]), 0, 255);
    });
  }
  var nt = { identity: function() {
    return [[1, 0, 0, 0, 0], [0, 1, 0, 0, 0], [0, 0, 1, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]];
  }, invertNHue: function() {
    return [[0.333, -0.667, -0.667, 0, 1], [-0.667, 0.333, -0.667, 0, 1], [-0.667, -0.667, 0.333, 0, 1], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]];
  }, brightness: function(e3) {
    return [[e3, 0, 0, 0, 0], [0, e3, 0, 0, 0], [0, 0, e3, 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]];
  }, contrast: function(e3) {
    var t3 = (1 - e3) / 2;
    return [[e3, 0, 0, 0, t3], [0, e3, 0, 0, t3], [0, 0, e3, 0, t3], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]];
  }, sepia: function(e3) {
    return [[0.393 + 0.607 * (1 - e3), 0.769 - 0.769 * (1 - e3), 0.189 - 0.189 * (1 - e3), 0, 0], [0.349 - 0.349 * (1 - e3), 0.686 + 0.314 * (1 - e3), 0.168 - 0.168 * (1 - e3), 0, 0], [0.272 - 0.272 * (1 - e3), 0.534 - 0.534 * (1 - e3), 0.131 + 0.869 * (1 - e3), 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]];
  }, grayscale: function(e3) {
    return [[0.2126 + 0.7874 * (1 - e3), 0.7152 - 0.7152 * (1 - e3), 0.0722 - 0.0722 * (1 - e3), 0, 0], [0.2126 - 0.2126 * (1 - e3), 0.7152 + 0.2848 * (1 - e3), 0.0722 - 0.0722 * (1 - e3), 0, 0], [0.2126 - 0.2126 * (1 - e3), 0.7152 - 0.7152 * (1 - e3), 0.0722 + 0.9278 * (1 - e3), 0, 0], [0, 0, 0, 1, 0], [0, 0, 0, 0, 1]];
  } };
  function ot(e3) {
    return e3[1 === e3.mode ? "darkSchemeBackgroundColor" : "lightSchemeBackgroundColor"];
  }
  function at(e3) {
    return e3[1 === e3.mode ? "darkSchemeTextColor" : "lightSchemeTextColor"];
  }
  var it = /* @__PURE__ */ new Map(), ut = /* @__PURE__ */ new Map();
  function st(e3) {
    if (ut.has(e3))
      return ut.get(e3);
    var t3 = Ee(Le(e3));
    return ut.set(e3, t3), t3;
  }
  function ct() {
    it.clear(), ut.clear();
  }
  var lt = ["r", "g", "b", "a"], dt = ["mode", "brightness", "contrast", "grayscale", "sepia", "darkSchemeBackgroundColor", "darkSchemeTextColor", "lightSchemeBackgroundColor", "lightSchemeTextColor"];
  function ft(e3, t3) {
    var r2 = "";
    return lt.forEach(function(t4) {
      r2 += e3[t4] + ";";
    }), dt.forEach(function(e4) {
      r2 += t3[e4] + ";";
    }), r2;
  }
  function ht(e3, t3, r2, n2, o2) {
    var i2;
    it.has(r2) ? i2 = it.get(r2) : (i2 = /* @__PURE__ */ new Map(), it.set(r2, i2));
    var u2 = ft(e3, t3);
    if (i2.has(u2))
      return i2.get(u2);
    var s2 = we(r2(Ee(e3), null == n2 ? null : st(n2), null == o2 ? null : st(o2))), c2 = s2.r, l2 = s2.g, d2 = s2.b, f2 = s2.a, h2 = a(rt([c2, l2, d2], tt(t3)), 3), p2 = h2[0], v2 = h2[1], m2 = h2[2], g2 = 1 === f2 ? Te({ r: p2, g: v2, b: m2 }) : Ce({ r: p2, g: v2, b: m2, a: f2 });
    return i2.set(u2, g2), g2;
  }
  function pt(e3) {
    return e3;
  }
  function vt(e3, t3) {
    return ht(e3, t3, pt);
  }
  function mt(e3, t3) {
    var r2 = ot(t3);
    return ht(e3, t3, gt, at(t3), r2);
  }
  function gt(e3, t3, r2) {
    var n2 = e3.h, o2 = e3.s, a2 = e3.l, i2 = e3.a, u2 = a2 < 0.5, s2 = n2, c2 = a2;
    return (u2 ? a2 < 0.2 || o2 < 0.12 : o2 < 0.24 || a2 > 0.8 && n2 > 200 && n2 < 280) && (u2 ? (s2 = t3.h, c2 = t3.s) : (s2 = r2.h, c2 = r2.s)), { h: s2, s: c2, l: Ke(a2, 0, 1, t3.l, r2.l), a: i2 };
  }
  var bt = 0.4;
  function yt(e3, t3) {
    var r2 = e3.h, n2 = e3.s, o2 = e3.l, a2 = e3.a, i2 = n2 < 0.12 || o2 > 0.8 && r2 > 200 && r2 < 280;
    if (o2 < 0.5) {
      var u2 = Ke(o2, 0, 0.5, 0, bt);
      return i2 ? { h: t3.h, s: t3.s, l: u2, a: a2 } : { h: r2, s: n2, l: u2, a: a2 };
    }
    var s2 = Ke(o2, 0.5, 1, bt, t3.l);
    if (i2)
      return { h: t3.h, s: t3.s, l: s2, a: a2 };
    var c2 = r2;
    return r2 > 60 && r2 < 180 && (c2 = r2 > 120 ? Ke(r2, 120, 180, 135, 180) : Ke(r2, 60, 120, 60, 105)), { h: c2, s: n2, l: s2, a: a2 };
  }
  function St(e3, r2) {
    if (0 === r2.mode)
      return mt(e3, r2);
    var n2 = ot(r2);
    return ht(e3, t2(t2({}, r2), { mode: 0 }), yt, n2);
  }
  var _t, kt = 0.55;
  function wt(e3) {
    return Ke(e3, 205, 245, 205, 220);
  }
  function Et(e3, t3) {
    var r2 = e3.h, n2 = e3.s, o2 = e3.l, a2 = e3.a, i2 = o2 < 0.2 || n2 < 0.24, u2 = !i2 && r2 > 205 && r2 < 245;
    if (o2 > 0.5) {
      var s2 = Ke(o2, 0.5, 1, kt, t3.l);
      if (i2)
        return { h: t3.h, s: t3.s, l: s2, a: a2 };
      var c2 = r2;
      return u2 && (c2 = wt(r2)), { h: c2, s: n2, l: s2, a: a2 };
    }
    if (i2)
      return { h: t3.h, s: t3.s, l: Ke(o2, 0, 0.5, t3.l, kt), a: a2 };
    var l2, d2 = r2;
    return u2 ? (d2 = wt(r2), l2 = Ke(o2, 0, 0.5, t3.l, Math.min(1, kt + 0.05))) : l2 = Ke(o2, 0, 0.5, t3.l, kt), { h: d2, s: n2, l: l2, a: a2 };
  }
  function xt(e3, r2) {
    if (0 === r2.mode)
      return mt(e3, r2);
    var n2 = at(r2);
    return ht(e3, t2(t2({}, r2), { mode: 0 }), Et, n2);
  }
  function Ct(e3, t3, r2) {
    var n2 = e3.h, o2 = e3.s, a2 = e3.l, i2 = e3.a, u2 = n2, s2 = o2;
    return (a2 < 0.2 || o2 < 0.24) && (a2 < 0.5 ? (u2 = t3.h, s2 = t3.s) : (u2 = r2.h, s2 = r2.s)), { h: u2, s: s2, l: Ke(a2, 0, 1, 0.5, 0.2), a: i2 };
  }
  function Tt(e3, r2) {
    if (0 === r2.mode)
      return mt(e3, r2);
    var n2 = at(r2), o2 = ot(r2);
    return ht(e3, t2(t2({}, r2), { mode: 0 }), Ct, n2, o2);
  }
  function Rt(e3, t3) {
    return St(e3, t3);
  }
  function Vt(e3, t3) {
    return St(e3, t3);
  }
  function Mt(e3) {
    var t3 = [];
    return t3.push('*:not(pre, pre *, code, .far, .fa, .glyphicon, [class*="vjs-"], .fab, .fa-github, .fas, .material-icons, .icofont, .typcn, mu, [class*="mu-"], .glyphicon, .icon) {'), e3.useFont && e3.fontFamily && t3.push("  font-family: " + e3.fontFamily + " !important;"), e3.textStroke > 0 && (t3.push("  -webkit-text-stroke: " + e3.textStroke + "px !important;"), t3.push("  text-stroke: " + e3.textStroke + "px !important;")), t3.push("}"), t3.join("\n");
  }
  function At(e3) {
    var t3 = [];
    return e3.mode === _t.dark && t3.push("invert(100%) hue-rotate(180deg)"), 100 !== e3.brightness && t3.push("brightness(" + e3.brightness + "%)"), 100 !== e3.contrast && t3.push("contrast(" + e3.contrast + "%)"), 0 !== e3.grayscale && t3.push("grayscale(" + e3.grayscale + "%)"), 0 !== e3.sepia && t3.push("sepia(" + e3.sepia + "%)"), 0 === t3.length ? null : t3.join(" ");
  }
  function Lt(e3) {
    return e3.slice(0, 4).map(function(e4) {
      return e4.map(function(e5) {
        return e5.toFixed(3);
      }).join(" ");
    }).join(" ");
  }
  function Pt(e3) {
    return Lt(tt(e3));
  }
  !function(e3) {
    e3[e3.light = 0] = "light", e3[e3.dark = 1] = "dark";
  }(_t || (_t = {}));
  var Ot = 0, Dt = /* @__PURE__ */ new Map(), It = /* @__PURE__ */ new Map();
  function Ft(e3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(t3) {
        return [2, new Promise(function(t4, r2) {
          var n2 = ++Ot;
          Dt.set(n2, t4), It.set(n2, r2), "undefined" != typeof chrome && chrome.runtime.sendMessage({ type: u.CS_FETCH, data: e3, id: n2 });
        })];
      });
    });
  }
  "undefined" != typeof chrome && chrome.runtime.onMessage.addListener(function(e3) {
    var t3 = e3.type, r2 = e3.data, n2 = e3.error, o2 = e3.id;
    if (t3 === u.BG_FETCH_RESPONSE) {
      var a2 = Dt.get(o2), i2 = It.get(o2);
      Dt.delete(o2), It.delete(o2), n2 ? i2 && i2(n2) : a2 && a2(r2);
    }
  });
  var Nt = new (function() {
    function e3() {
      this.queue = [], this.timerId = null, this.frameDuration = 1e3 / 60;
    }
    return e3.prototype.addToQueue = function(e4) {
      this.queue.push(e4), this.startQueue();
    }, e3.prototype.stopQueue = function() {
      null !== this.timerId && (cancelAnimationFrame(this.timerId), this.timerId = null), this.queue = [];
    }, e3.prototype.startQueue = function() {
      var e4 = this;
      this.timerId || (this.timerId = requestAnimationFrame(function() {
        e4.timerId = null;
        for (var t3, r2 = Date.now(); t3 = e4.queue.shift(); )
          if (t3(), Date.now() - r2 >= e4.frameDuration) {
            e4.startQueue();
            break;
          }
      }));
    }, e3;
  }())();
  function jt(e3) {
    return r(this, void 0, void 0, function() {
      var o2 = this;
      return n(this, function(a2) {
        return [2, new Promise(function(a3, i2) {
          return r(o2, void 0, void 0, function() {
            var r2, o3, u2, s2;
            return n(this, function(n2) {
              switch (n2.label) {
                case 0:
                  return e3.startsWith("data:") ? (r2 = e3, [3, 4]) : [3, 1];
                case 1:
                  return n2.trys.push([1, 3, , 4]), [4, Bt(e3)];
                case 2:
                  return r2 = n2.sent(), [3, 4];
                case 3:
                  return o3 = n2.sent(), i2(o3), [2];
                case 4:
                  return n2.trys.push([4, 6, , 7]), [4, Ut(r2)];
                case 5:
                  return u2 = n2.sent(), Nt.addToQueue(function() {
                    a3(t2({ src: e3, dataURL: r2, width: u2.naturalWidth, height: u2.naturalHeight }, Qt(u2)));
                  }), [3, 7];
                case 6:
                  return s2 = n2.sent(), i2(s2), [3, 7];
                case 7:
                  return [2];
              }
            });
          });
        })];
      });
    });
  }
  function Bt(e3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(t3) {
        switch (t3.label) {
          case 0:
            return new URL(e3).origin !== location.origin ? [3, 2] : [4, _(e3)];
          case 1:
          case 3:
            return [2, t3.sent()];
          case 2:
            return [4, Ft({ url: e3, responseType: "data-url" })];
        }
      });
    });
  }
  function Ut(e3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(t3) {
        return [2, new Promise(function(t4, r2) {
          var n2 = new Image();
          n2.onload = function() {
            return t4(n2);
          }, n2.onerror = function() {
            return r2("Unable to load image " + e3);
          }, n2.src = e3;
        })];
      });
    });
  }
  var qt, Wt, Ht = 1024;
  function Gt() {
    var e3 = Ht, t3 = Ht;
    (qt = document.createElement("canvas")).width = e3, qt.height = t3, (Wt = qt.getContext("2d")).imageSmoothingEnabled = false;
  }
  function $t() {
    qt = null, Wt = null;
  }
  var zt = 5242880;
  function Qt(e3) {
    qt || Gt();
    var t3 = e3.naturalWidth, r2 = e3.naturalHeight;
    if (0 === r2 || 0 === t3)
      return I("logWarn(Image is empty " + e3.currentSrc + ")"), null;
    if (t3 * r2 * 4 > zt)
      return { isDark: false, isLight: false, isTransparent: false, isLarge: false, isTooLarge: true };
    var n2 = t3 * r2, o2 = Math.min(1, Math.sqrt(Ht / n2)), a2 = Math.ceil(t3 * o2), i2 = Math.ceil(r2 * o2);
    Wt.clearRect(0, 0, a2, i2), Wt.drawImage(e3, 0, 0, t3, r2, 0, 0, a2, i2);
    var u2, s2, c2, l2, d2, f2, h2, p2 = Wt.getImageData(0, 0, a2, i2).data, v2 = 0.05, m2 = 0.4, g2 = 0.7, b2 = 0, y2 = 0, S2 = 0;
    for (c2 = 0; c2 < i2; c2++)
      for (s2 = 0; s2 < a2; s2++)
        l2 = p2[0 + (u2 = 4 * (c2 * a2 + s2))] / 255, d2 = p2[u2 + 1] / 255, f2 = p2[u2 + 2] / 255, p2[u2 + 3] / 255 < v2 ? b2++ : ((h2 = 0.2126 * l2 + 0.7152 * d2 + 0.0722 * f2) < m2 && y2++, h2 > g2 && S2++);
    var _2 = a2 * i2, k2 = _2 - b2;
    return { isDark: y2 / k2 >= 0.7, isLight: S2 / k2 >= 0.7, isTransparent: b2 / _2 >= 0.1, isLarge: n2 >= 48e4, isTooLarge: false };
  }
  function Kt(e3, t3) {
    var r2 = e3.dataURL, n2 = e3.width, o2 = e3.height, a2 = ['<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="' + n2 + '" height="' + o2 + '">', "<defs>", '<filter id="darkreader-image-filter">', '<feColorMatrix type="matrix" values="' + Pt(t3) + '" />', "</filter>", "</defs>", '<image width="' + n2 + '" height="' + o2 + '" filter="url(#darkreader-image-filter)" xlink:href="' + r2 + '" />', "</svg>"].join("");
    return "data:image/svg+xml;base64," + btoa(a2);
  }
  function Xt() {
    Nt && Nt.stopQueue(), $t();
  }
  function Yt(e3, t3) {
    return Boolean(e3 && e3.getPropertyPriority(t3));
  }
  function Zt(e3, t3, r2, n2, o2, a2) {
    if (e3.startsWith("--")) {
      if (i2 = mr(n2, e3, t3, r2, o2, a2))
        return { property: e3, value: i2, important: Yt(r2.style, e3), sourceValue: t3 };
    } else if (t3.includes("var(")) {
      if (i2 = gr(n2, e3, t3))
        return { property: e3, value: i2, important: Yt(r2.style, e3), sourceValue: t3 };
    } else if (e3.includes("color") && "-webkit-print-color-adjust" !== e3 || "fill" === e3 || "stroke" === e3 || "stop-color" === e3) {
      if (i2 = sr(e3, t3))
        return { property: e3, value: i2, important: Yt(r2.style, e3), sourceValue: t3 };
    } else if ("background-image" === e3 || "list-style-image" === e3) {
      if (i2 = hr(t3, r2, o2, a2))
        return { property: e3, value: i2, important: Yt(r2.style, e3), sourceValue: t3 };
    } else if (e3.includes("shadow")) {
      var i2;
      if (i2 = vr(t3))
        return { property: e3, value: i2, important: Yt(r2.style, e3), sourceValue: t3 };
    }
    return null;
  }
  function Jt(e3, t3, r2) {
    var n2 = [];
    return t3 || (n2.push("html {"), n2.push("    background-color: " + St({ r: 255, g: 255, b: 255 }, e3) + " !important;"), n2.push("}")), n2.push((t3 ? "" : "html, body, ") + (r2 ? "input, textarea, select, button" : "") + " {"), n2.push("    background-color: " + St({ r: 255, g: 255, b: 255 }, e3) + ";"), n2.push("}"), n2.push("html, body, " + (r2 ? "input, textarea, select, button" : "") + " {"), n2.push("    border-color: " + Tt({ r: 76, g: 76, b: 76 }, e3) + ";"), n2.push("    color: " + xt({ r: 0, g: 0, b: 0 }, e3) + ";"), n2.push("}"), n2.push("a {"), n2.push("    color: " + xt({ r: 0, g: 64, b: 255 }, e3) + ";"), n2.push("}"), n2.push("table {"), n2.push("    border-color: " + Tt({ r: 128, g: 128, b: 128 }, e3) + ";"), n2.push("}"), n2.push("::placeholder {"), n2.push("    color: " + xt({ r: 169, g: 169, b: 169 }, e3) + ";"), n2.push("}"), n2.push("input:-webkit-autofill,"), n2.push("textarea:-webkit-autofill,"), n2.push("select:-webkit-autofill {"), n2.push("    background-color: " + St({ r: 250, g: 255, b: 189 }, e3) + " !important;"), n2.push("    color: " + xt({ r: 0, g: 0, b: 0 }, e3) + " !important;"), n2.push("}"), e3.scrollbarColor && n2.push(rr(e3)), e3.selectionColor && n2.push(tr(e3)), n2.join("\n");
  }
  function er(e3) {
    var r2, n2;
    if ("auto" === e3.selectionColor)
      r2 = St({ r: 0, g: 96, b: 212 }, t2(t2({}, e3), { grayscale: 0 })), n2 = xt({ r: 255, g: 255, b: 255 }, t2(t2({}, e3), { grayscale: 0 }));
    else {
      var o2 = Ee(Le(e3.selectionColor));
      r2 = e3.selectionColor, n2 = o2.l < 0.5 ? "#FFF" : "#000";
    }
    return { backgroundColorSelection: r2, foregroundColorSelection: n2 };
  }
  function tr(e3) {
    var t3 = [], r2 = er(e3), n2 = r2.backgroundColorSelection, o2 = r2.foregroundColorSelection;
    return ["::selection", "::-moz-selection"].forEach(function(e4) {
      t3.push(e4 + " {"), t3.push("    background-color: " + n2 + " !important;"), t3.push("    color: " + o2 + " !important;"), t3.push("}");
    }), t3.join("\n");
  }
  function rr(e3) {
    var r2, n2, o2, a2, i2, u2, s2 = [];
    if ("auto" === e3.scrollbarColor)
      r2 = St({ r: 241, g: 241, b: 241 }, e3), n2 = xt({ r: 96, g: 96, b: 96 }, e3), o2 = St({ r: 176, g: 176, b: 176 }, e3), a2 = St({ r: 144, g: 144, b: 144 }, e3), i2 = St({ r: 96, g: 96, b: 96 }, e3), u2 = St({ r: 255, g: 255, b: 255 }, e3);
    else {
      var c2 = Ee(Le(e3.scrollbarColor)), l2 = c2.l > 0.5, d2 = function(e4) {
        return t2(t2({}, c2), { l: Xe(c2.l + e4, 0, 1) });
      }, h2 = function(e4) {
        return t2(t2({}, c2), { l: Xe(c2.l - e4, 0, 1) });
      };
      r2 = Re(h2(0.4)), n2 = Re(l2 ? h2(0.4) : d2(0.4)), o2 = Re(c2), a2 = Re(d2(0.1)), i2 = Re(d2(0.2));
    }
    return s2.push("::-webkit-scrollbar {"), s2.push("    background-color: " + r2 + ";"), s2.push("    color: " + n2 + ";"), s2.push("}"), s2.push("::-webkit-scrollbar-thumb {"), s2.push("    background-color: " + o2 + ";"), s2.push("}"), s2.push("::-webkit-scrollbar-thumb:hover {"), s2.push("    background-color: " + a2 + ";"), s2.push("}"), s2.push("::-webkit-scrollbar-thumb:active {"), s2.push("    background-color: " + i2 + ";"), s2.push("}"), s2.push("::-webkit-scrollbar-corner {"), s2.push("    background-color: " + u2 + ";"), s2.push("}"), f && (s2.push("* {"), s2.push("    scrollbar-color: " + o2 + " " + r2 + ";"), s2.push("}")), s2.join("\n");
  }
  function nr(e3, t3) {
    var r2 = t3.strict, n2 = [], o2 = location.hostname.endsWith("microsoft.com");
    return n2.push("html, body, " + (r2 ? "body :not(iframe)" + (o2 ? ':not(div[style^="position:absolute;top:0;left:-"]' : "") : "body > :not(iframe)") + " {"), n2.push("    background-color: " + St({ r: 255, g: 255, b: 255 }, e3) + " !important;"), n2.push("    border-color: " + Tt({ r: 64, g: 64, b: 64 }, e3) + " !important;"), n2.push("    color: " + xt({ r: 0, g: 0, b: 0 }, e3) + " !important;"), n2.push("}"), n2.join("\n");
  }
  var or = /* @__PURE__ */ new Set(["inherit", "transparent", "initial", "currentcolor", "none", "unset"]), ar = /* @__PURE__ */ new Map();
  function ir(e3) {
    if (e3 = e3.trim(), ar.has(e3))
      return ar.get(e3);
    e3.includes("calc(") && (e3 = $e(e3));
    var t3 = Le(e3);
    return ar.set(e3, t3), t3;
  }
  function ur(e3) {
    try {
      return ir(e3);
    } catch (e4) {
      return null;
    }
  }
  function sr(e3, t3) {
    if (or.has(t3.toLowerCase()))
      return t3;
    try {
      var r2 = ir(t3);
      return e3.includes("background") ? function(e4) {
        return St(r2, e4);
      } : e3.includes("border") || e3.includes("outline") ? function(e4) {
        return Tt(r2, e4);
      } : function(e4) {
        return xt(r2, e4);
      };
    } catch (e4) {
      return null;
    }
  }
  var cr = /[\-a-z]+gradient\(([^\(\)]*(\(([^\(\)]*(\(.*?\)))*[^\(\)]*\))){0,15}[^\(\)]*\)/g, lr = /* @__PURE__ */ new Map(), dr = /* @__PURE__ */ new Map();
  function fr(e3, t3) {
    if (!e3 || 0 === t3.length)
      return false;
    if (t3.some(function(e4) {
      return "*" === e4;
    }))
      return true;
    for (var r2 = e3.split(/,\s*/g), n2 = function(e4) {
      var n3 = t3[e4];
      if (r2.some(function(e5) {
        return e5 === n3;
      }))
        return { value: true };
    }, o2 = 0; o2 < t3.length; o2++) {
      var a2 = n2(o2);
      if ("object" == typeof a2)
        return a2.value;
    }
    return false;
  }
  function hr(e3, o2, a2, i2) {
    var u2 = this;
    try {
      var s2 = Ze(cr, e3), c2 = Ze(pe, e3);
      if (0 === c2.length && 0 === s2.length)
        return e3;
      var l2 = function(t3) {
        var r2 = 0;
        return t3.map(function(t4) {
          var n2 = e3.indexOf(t4, r2);
          return r2 = n2 + t4.length, { match: t4, index: n2 };
        });
      }, d2 = l2(c2).map(function(e4) {
        return t2({ type: "url" }, e4);
      }).concat(l2(s2).map(function(e4) {
        return t2({ type: "gradient" }, e4);
      })).sort(function(e4, t3) {
        return e4.index - t3.index;
      }), f2 = function(e4) {
        var t3 = e4.match(/^(.*-gradient)\((.*)\)$/), r2 = t3[1], n2 = t3[2], o3 = /^(from|color-stop|to)\(([^\(\)]*?,\s*)?(.*?)\)$/, a3 = Ze(/([^\(\),]+(\([^\(\)]*(\([^\(\)]*\)*[^\(\)]*)?\))?[^\(\),]*),?/g, n2, 1).map(function(e5) {
          var t4 = ur(e5 = e5.trim());
          if (t4)
            return function(e6) {
              return Vt(t4, e6);
            };
          var r3 = e5.lastIndexOf(" ");
          if (t4 = ur(e5.substring(0, r3)))
            return function(n4) {
              return Vt(t4, n4) + " " + e5.substring(r3 + 1);
            };
          var n3 = e5.match(o3);
          return n3 && (t4 = ur(n3[3])) ? function(e6) {
            return n3[1] + "(" + (n3[2] ? n3[2] + ", " : "") + Vt(t4, e6) + ")";
          } : function() {
            return e5;
          };
        });
        return function(e5) {
          return r2 + "(" + a3.map(function(t4) {
            return t4(e5);
          }).join(", ") + ")";
        };
      }, h2 = function(e4) {
        var t3;
        if (fr(o2.selectorText, a2))
          return null;
        var s3 = me(e4), c3 = o2.parentStyleSheet, l3 = c3 && c3.href ? ge(c3.href) : (null === (t3 = c3.ownerNode) || void 0 === t3 ? void 0 : t3.baseURI) || location.origin, d3 = 'url("' + (s3 = se(l3, s3)) + '")';
        return function(e5) {
          return r(u2, void 0, void 0, function() {
            var t4, r2;
            return n(this, function(n2) {
              switch (n2.label) {
                case 0:
                  return lr.has(s3) ? (t4 = lr.get(s3), [3, 7]) : [3, 1];
                case 1:
                  return n2.trys.push([1, 6, , 7]), dr.has(s3) ? (r2 = dr.get(s3), [4, new Promise(function(e6) {
                    return r2.push(e6);
                  })]) : [3, 3];
                case 2:
                  return (t4 = n2.sent()) ? [3, 5] : [2, null];
                case 3:
                  return dr.set(s3, []), [4, jt(s3)];
                case 4:
                  t4 = n2.sent(), lr.set(s3, t4), dr.get(s3).forEach(function(e6) {
                    return e6(t4);
                  }), dr.delete(s3), n2.label = 5;
                case 5:
                  return i2() ? [2, null] : [3, 7];
                case 6:
                  return I(n2.sent()), dr.has(s3) && (dr.get(s3).forEach(function(e6) {
                    return e6(null);
                  }), dr.delete(s3)), [2, d3];
                case 7:
                  return [2, p2(t4, e5) || d3];
              }
            });
          });
        };
      }, p2 = function(e4, r2) {
        var n2, o3 = e4.isDark, a3 = e4.isLight, i3 = e4.isTransparent, u3 = e4.isLarge, s3 = e4.isTooLarge, c3 = e4.width;
        return s3 ? n2 = 'url("' + e4.src + '")' : o3 && i3 && 1 === r2.mode && !u3 && c3 > 2 ? (D("Inverting dark image " + e4.src), n2 = 'url("' + Kt(e4, t2(t2({}, r2), { sepia: Xe(r2.sepia + 10, 0, 100) })) + '")') : a3 && !i3 && 1 === r2.mode ? u3 ? n2 = "none" : (D("Dimming light image " + e4.src), n2 = 'url("' + Kt(e4, r2) + '")') : 0 === r2.mode && a3 && !u3 ? (D("Applying filter to image " + e4.src), n2 = 'url("' + Kt(e4, t2(t2({}, r2), { brightness: Xe(r2.brightness - 10, 5, 200), sepia: Xe(r2.sepia + 10, 0, 100) })) + '")') : n2 = null, n2;
      }, v2 = [], m2 = 0;
      return d2.forEach(function(t3, r2) {
        var n2 = t3.match, o3 = t3.type, a3 = t3.index, i3 = m2, u3 = a3 + n2.length;
        m2 = u3, v2.push(function() {
          return e3.substring(i3, a3);
        }), v2.push("url" === o3 ? h2(n2) : f2(n2)), r2 === d2.length - 1 && v2.push(function() {
          return e3.substring(u3);
        });
      }), function(e4) {
        var t3 = v2.filter(Boolean).map(function(t4) {
          return t4(e4);
        });
        return t3.some(function(e5) {
          return e5 instanceof Promise;
        }) ? Promise.all(t3).then(function(e5) {
          return e5.join("");
        }) : t3.join("");
      };
    } catch (e4) {
      return null;
    }
  }
  function pr(e3) {
    try {
      var t3 = 0, r2 = Ze(/(^|\s)(?!calc)([a-z]+\(.+?\)|#[0-9a-f]+|[a-z]+)(.*?(inset|outset)?($|,))/gi, e3, 2), n2 = 0, o2 = r2.map(function(o3, a2) {
        var i2 = t3, u2 = e3.indexOf(o3, t3), s2 = u2 + o3.length;
        t3 = s2;
        var c2 = ur(o3);
        return c2 ? function(t4) {
          return "" + e3.substring(i2, u2) + Rt(c2, t4) + (a2 === r2.length - 1 ? e3.substring(s2) : "");
        } : (n2++, function() {
          return e3.substring(i2, s2);
        });
      });
      return function(e4) {
        var t4 = o2.map(function(t5) {
          return t5(e4);
        }).join("");
        return { matchesLength: r2.length, unparseableMatchesLength: n2, result: t4 };
      };
    } catch (e4) {
      return null;
    }
  }
  function vr(e3) {
    var t3 = pr(e3);
    return t3 ? function(e4) {
      return t3(e4).result;
    } : null;
  }
  function mr(e3, t3, r2, n2, o2, a2) {
    return e3.getModifierForVariable({ varName: t3, sourceValue: r2, rule: n2, ignoredImgSelectors: o2, isCancelled: a2 });
  }
  function gr(e3, t3, r2) {
    return e3.getModifierForVarDependant(t3, r2);
  }
  function br() {
    ar.clear(), ct(), lr.clear(), Xt(), dr.clear();
  }
  var yr = 1, Sr = 2, _r = 4, kr = 8, wr = new (function() {
    function e3() {
      this.varTypes = /* @__PURE__ */ new Map(), this.rulesQueue = [], this.definedVars = /* @__PURE__ */ new Set(), this.varRefs = /* @__PURE__ */ new Map(), this.unknownColorVars = /* @__PURE__ */ new Set(), this.unknownBgVars = /* @__PURE__ */ new Set(), this.undefinedVars = /* @__PURE__ */ new Set(), this.initialVarTypes = /* @__PURE__ */ new Map(), this.changedTypeVars = /* @__PURE__ */ new Set(), this.typeChangeSubscriptions = /* @__PURE__ */ new Map(), this.unstableVarValues = /* @__PURE__ */ new Map();
    }
    return e3.prototype.clear = function() {
      this.varTypes.clear(), this.rulesQueue.splice(0), this.definedVars.clear(), this.varRefs.clear(), this.unknownColorVars.clear(), this.unknownBgVars.clear(), this.undefinedVars.clear(), this.initialVarTypes.clear(), this.changedTypeVars.clear(), this.typeChangeSubscriptions.clear(), this.unstableVarValues.clear();
    }, e3.prototype.isVarType = function(e4, t3) {
      return this.varTypes.has(e4) && (this.varTypes.get(e4) & t3) > 0;
    }, e3.prototype.addRulesForMatching = function(e4) {
      this.rulesQueue.push(e4);
    }, e3.prototype.matchVariablesAndDependants = function() {
      var e4 = this;
      this.changedTypeVars.clear(), this.initialVarTypes = new Map(this.varTypes), this.collectRootVariables(), this.collectVariablesAndVarDep(this.rulesQueue), this.rulesQueue.splice(0), this.collectRootVarDependants(), this.varRefs.forEach(function(t3, r2) {
        t3.forEach(function(t4) {
          e4.varTypes.has(r2) && e4.resolveVariableType(t4, e4.varTypes.get(r2));
        });
      }), this.unknownColorVars.forEach(function(t3) {
        e4.unknownBgVars.has(t3) ? (e4.unknownColorVars.delete(t3), e4.unknownBgVars.delete(t3), e4.resolveVariableType(t3, yr)) : e4.isVarType(t3, yr | Sr | _r) ? e4.unknownColorVars.delete(t3) : e4.undefinedVars.add(t3);
      }), this.unknownBgVars.forEach(function(t3) {
        null != e4.findVarRef(t3, function(t4) {
          return e4.unknownColorVars.has(t4) || e4.isVarType(t4, Sr | _r);
        }) ? e4.itarateVarRefs(t3, function(t4) {
          e4.resolveVariableType(t4, yr);
        }) : e4.isVarType(t3, yr | kr) ? e4.unknownBgVars.delete(t3) : e4.undefinedVars.add(t3);
      }), this.changedTypeVars.forEach(function(t3) {
        e4.typeChangeSubscriptions.has(t3) && e4.typeChangeSubscriptions.get(t3).forEach(function(e5) {
          e5();
        });
      }), this.changedTypeVars.clear();
    }, e3.prototype.getModifierForVariable = function(e4) {
      var t3 = this;
      return function(r2) {
        var n2 = e4.varName, o2 = e4.sourceValue, a2 = e4.rule, i2 = e4.ignoredImgSelectors, u2 = e4.isCancelled, s2 = function() {
          var e5 = [], s3 = function(a3, i3, u3) {
            if (t3.isVarType(n2, a3)) {
              var s4, c4 = i3(n2);
              if (Dr(o2))
                if (Ir(o2)) {
                  var l4 = Wr(o2, t3.unstableVarValues);
                  l4 || (l4 = a3 === yr ? "#ffffff" : "#000000"), s4 = u3(l4, r2);
                } else
                  s4 = Rr(o2, function(e6) {
                    return i3(e6);
                  }, function(e6) {
                    return u3(e6, r2);
                  });
              else
                s4 = u3(o2, r2);
              e5.push({ property: c4, value: s4 });
            }
          };
          if (s3(yr, Mr, Br), s3(Sr, Ar, Ur), s3(_r, Lr, qr), t3.isVarType(n2, kr)) {
            var c3 = Pr(n2), l3 = o2;
            Dr(o2) && (l3 = Rr(o2, function(e6) {
              return Mr(e6);
            }, function(e6) {
              return Br(e6, r2);
            }));
            var d3 = hr(l3, a2, i2, u2);
            l3 = "function" == typeof d3 ? d3(r2) : d3, e5.push({ property: c3, value: l3 });
          }
          return e5;
        }, c2 = /* @__PURE__ */ new Set(), l2 = function(e5) {
          var r3 = function() {
            var t4 = s2();
            e5(t4);
          };
          c2.add(r3), t3.subscribeForVarTypeChange(n2, r3);
        }, d2 = function() {
          c2.forEach(function(e5) {
            t3.unsubscribeFromVariableTypeChanges(n2, e5);
          });
        };
        return { declarations: s2(), onTypeChange: { addListener: l2, removeListeners: d2 } };
      };
    }, e3.prototype.getModifierForVarDependant = function(e4, t3) {
      var r2 = this;
      if (t3.match(/^\s*(rgb|hsl)a?\(/)) {
        var n2 = e4.startsWith("background"), o2 = "color" === e4 || "caret-color" === e4;
        return function(e5) {
          var a2 = Wr(t3, r2.unstableVarValues);
          return a2 || (a2 = n2 ? "#ffffff" : "#000000"), (n2 ? Br : o2 ? Ur : qr)(a2, e5);
        };
      }
      return "background-color" === e4 ? function(e5) {
        return Rr(t3, function(e6) {
          return Mr(e6);
        }, function(t4) {
          return Br(t4, e5);
        });
      } : "color" === e4 || "caret-color" === e4 ? function(e5) {
        return Rr(t3, function(e6) {
          return Ar(e6);
        }, function(t4) {
          return Ur(t4, e5);
        });
      } : "background" === e4 || "background-image" === e4 || "box-shadow" === e4 ? function(n3) {
        var o3 = /* @__PURE__ */ new Set(), a2 = function() {
          var a3 = Rr(t3, function(e5) {
            return r2.isVarType(e5, yr) ? Mr(e5) : r2.isVarType(e5, kr) ? Pr(e5) : (o3.add(e5), e5);
          }, function(e5) {
            return Br(e5, n3);
          });
          if ("box-shadow" === e4) {
            var i3 = pr(a3)(n3);
            if (i3.unparseableMatchesLength !== i3.matchesLength)
              return i3.result;
          }
          return a3;
        }, i2 = a2();
        return o3.size > 0 ? new Promise(function(e5) {
          var t4 = o3.values().next().value, n4 = function() {
            r2.unsubscribeFromVariableTypeChanges(t4, n4);
            var o4 = a2();
            e5(o4);
          };
          r2.subscribeForVarTypeChange(t4, n4);
        }) : i2;
      } : e4.startsWith("border") || e4.startsWith("outline") ? function(e5) {
        return Rr(t3, function(e6) {
          return Lr(e6);
        }, function(t4) {
          return qr(t4, e5);
        });
      } : null;
    }, e3.prototype.subscribeForVarTypeChange = function(e4, t3) {
      this.typeChangeSubscriptions.has(e4) || this.typeChangeSubscriptions.set(e4, /* @__PURE__ */ new Set());
      var r2 = this.typeChangeSubscriptions.get(e4);
      r2.has(t3) || r2.add(t3);
    }, e3.prototype.unsubscribeFromVariableTypeChanges = function(e4, t3) {
      this.typeChangeSubscriptions.has(e4) && this.typeChangeSubscriptions.get(e4).delete(t3);
    }, e3.prototype.collectVariablesAndVarDep = function(e4) {
      var t3 = this;
      e4.forEach(function(e5) {
        le(e5, function(e6) {
          e6.style && he(e6.style, function(e7, r2) {
            Or(e7) && t3.inspectVariable(e7, r2), Dr(r2) && t3.inspectVarDependant(e7, r2);
          });
        });
      });
    }, e3.prototype.collectRootVariables = function() {
      var e4 = this;
      he(document.documentElement.style, function(t3, r2) {
        Or(t3) && e4.inspectVariable(t3, r2);
      });
    }, e3.prototype.inspectVariable = function(e4, t3) {
      this.unstableVarValues.set(e4, t3), Dr(t3) && Ir(t3) && (this.unknownColorVars.add(e4), this.definedVars.add(e4)), this.definedVars.has(e4) || (this.definedVars.add(e4), ur(t3) ? this.unknownColorVars.add(e4) : (t3.includes("url(") || t3.includes("linear-gradient(") || t3.includes("radial-gradient(")) && this.resolveVariableType(e4, kr));
    }, e3.prototype.resolveVariableType = function(e4, t3) {
      var r2 = this.initialVarTypes.get(e4) || 0, n2 = (this.varTypes.get(e4) || 0) | t3;
      this.varTypes.set(e4, n2), (n2 !== r2 || this.undefinedVars.has(e4)) && (this.changedTypeVars.add(e4), this.undefinedVars.delete(e4)), this.unknownColorVars.delete(e4), this.unknownBgVars.delete(e4);
    }, e3.prototype.collectRootVarDependants = function() {
      var e4 = this;
      he(document.documentElement.style, function(t3, r2) {
        Dr(r2) && e4.inspectVarDependant(t3, r2);
      });
    }, e3.prototype.inspectVarDependant = function(e4, t3) {
      var r2 = this;
      Or(e4) ? this.iterateVarDeps(t3, function(t4) {
        r2.varRefs.has(e4) || r2.varRefs.set(e4, /* @__PURE__ */ new Set()), r2.varRefs.get(e4).add(t4);
      }) : "background-color" === e4 || "box-shadow" === e4 ? this.iterateVarDeps(t3, function(e5) {
        return r2.resolveVariableType(e5, yr);
      }) : "color" === e4 || "caret-color" === e4 ? this.iterateVarDeps(t3, function(e5) {
        return r2.resolveVariableType(e5, Sr);
      }) : e4.startsWith("border") || e4.startsWith("outline") ? this.iterateVarDeps(t3, function(e5) {
        return r2.resolveVariableType(e5, _r);
      }) : "background" !== e4 && "background-image" !== e4 || this.iterateVarDeps(t3, function(e5) {
        if (!r2.isVarType(e5, yr | kr)) {
          var t4 = null != r2.findVarRef(e5, function(e6) {
            return r2.unknownColorVars.has(e6) || r2.isVarType(e6, Sr | _r);
          });
          r2.itarateVarRefs(e5, function(e6) {
            t4 ? r2.resolveVariableType(e6, yr) : r2.unknownBgVars.add(e6);
          });
        }
      });
    }, e3.prototype.iterateVarDeps = function(e4, t3) {
      var r2 = /* @__PURE__ */ new Set();
      Vr(e4, function(e5) {
        return r2.add(e5);
      }), r2.forEach(function(e5) {
        return t3(e5);
      });
    }, e3.prototype.findVarRef = function(e4, t3, r2) {
      var n2, a2;
      if (void 0 === r2 && (r2 = /* @__PURE__ */ new Set()), r2.has(e4))
        return null;
      if (r2.add(e4), t3(e4))
        return e4;
      var i2 = this.varRefs.get(e4);
      if (!i2 || 0 === i2.size)
        return null;
      try {
        for (var u2 = o(i2), s2 = u2.next(); !s2.done; s2 = u2.next()) {
          var c2 = s2.value, l2 = this.findVarRef(c2, t3, r2);
          if (l2)
            return l2;
        }
      } catch (e5) {
        n2 = { error: e5 };
      } finally {
        try {
          s2 && !s2.done && (a2 = u2.return) && a2.call(u2);
        } finally {
          if (n2)
            throw n2.error;
        }
      }
      return null;
    }, e3.prototype.itarateVarRefs = function(e4, t3) {
      this.findVarRef(e4, function(e5) {
        return t3(e5), false;
      });
    }, e3.prototype.setOnRootVariableChange = function(e4) {
      this.onRootVariableDefined = e4;
    }, e3.prototype.putRootVars = function(e4, t3) {
      var r2, n2, i2 = this, u2 = e4.sheet;
      u2.cssRules.length > 0 && u2.deleteRule(0);
      var s2 = /* @__PURE__ */ new Map();
      he(document.documentElement.style, function(e5, r3) {
        Or(e5) && (i2.isVarType(e5, yr) && s2.set(Mr(e5), Br(r3, t3)), i2.isVarType(e5, Sr) && s2.set(Ar(e5), Ur(r3, t3)), i2.isVarType(e5, _r) && s2.set(Lr(e5), qr(r3, t3)), i2.subscribeForVarTypeChange(e5, i2.onRootVariableDefined));
      });
      var c2 = [];
      c2.push(":root {");
      try {
        for (var l2 = o(s2), d2 = l2.next(); !d2.done; d2 = l2.next()) {
          var f2 = a(d2.value, 2), h2 = f2[0], p2 = f2[1];
          c2.push("    " + h2 + ": " + p2 + ";");
        }
      } catch (e5) {
        r2 = { error: e5 };
      } finally {
        try {
          d2 && !d2.done && (n2 = l2.return) && n2.call(l2);
        } finally {
          if (r2)
            throw r2.error;
        }
      }
      c2.push("}");
      var v2 = c2.join("\n");
      u2.insertRule(v2);
    }, e3;
  }())();
  function Er(e3, t3) {
    void 0 === t3 && (t3 = 0);
    var r2 = e3.indexOf("var(", t3);
    if (r2 >= 0) {
      var n2 = et(e3, r2 + 3);
      return n2 ? { start: r2, end: n2.end } : null;
    }
  }
  function xr(e3) {
    for (var t3, r2 = [], n2 = 0; t3 = Er(e3, n2); ) {
      var o2 = t3.start, a2 = t3.end;
      r2.push({ start: o2, end: a2, value: e3.substring(o2, a2) }), n2 = t3.end + 1;
    }
    return r2;
  }
  function Cr(e3, t3) {
    var r2 = xr(e3), n2 = r2.length;
    if (0 === n2)
      return e3;
    var o2 = e3.length, a2 = r2.map(function(e4) {
      return t3(e4.value);
    }), i2 = [];
    i2.push(e3.substring(0, r2[0].start));
    for (var u2 = 0; u2 < n2; u2++) {
      i2.push(a2[u2]);
      var s2 = r2[u2].end, c2 = u2 < n2 - 1 ? r2[u2 + 1].start : o2;
      i2.push(e3.substring(s2, c2));
    }
    return i2.join("");
  }
  function Tr(e3) {
    var t3, r2, n2 = e3.indexOf(",");
    return n2 >= 0 ? (t3 = e3.substring(4, n2).trim(), r2 = e3.substring(n2 + 1, e3.length - 1).trim()) : (t3 = e3.substring(4, e3.length - 1).trim(), r2 = ""), { name: t3, fallback: r2 };
  }
  function Rr(e3, t3, r2) {
    return Cr(e3, function(e4) {
      var n2 = Tr(e4), o2 = n2.name, a2 = n2.fallback, i2 = t3(o2);
      return a2 ? "var(" + i2 + ", " + (Dr(a2) ? Rr(a2, t3, r2) : r2 ? r2(a2) : a2) + ")" : "var(" + i2 + ")";
    });
  }
  function Vr(e3, t3) {
    Rr(e3, function(e4) {
      return t3(e4), e4;
    });
  }
  function Mr(e3) {
    return "--darkreader-bg" + e3;
  }
  function Ar(e3) {
    return "--darkreader-text" + e3;
  }
  function Lr(e3) {
    return "--darkreader-border" + e3;
  }
  function Pr(e3) {
    return "--darkreader-bgimg" + e3;
  }
  function Or(e3) {
    return e3.startsWith("--");
  }
  function Dr(e3) {
    return e3.includes("var(");
  }
  function Ir(e3) {
    return e3.match(/^\s*(rgb|hsl)a?\(/);
  }
  var Fr = /^\d{1,3}, ?\d{1,3}, ?\d{1,3}$/;
  function Nr(e3) {
    if (Fr.test(e3)) {
      var t3 = e3.split(","), r2 = "rgb(";
      return t3.forEach(function(e4) {
        r2 += e4.trim() + ", ";
      }), r2 = r2.substr(0, r2.length - 2), { isRaw: true, color: r2 += ")" };
    }
    return { isRaw: false, color: e3 };
  }
  function jr(e3, t3, r2) {
    var n2 = Nr(e3), o2 = n2.isRaw, a2 = n2.color, i2 = ur(a2);
    if (i2) {
      var u2 = r2(i2, t3);
      if (o2) {
        var s2 = ur(u2);
        return s2 ? s2.r + ", " + s2.g + ", " + s2.b : u2;
      }
      return u2;
    }
    return a2;
  }
  function Br(e3, t3) {
    return jr(e3, t3, St);
  }
  function Ur(e3, t3) {
    return jr(e3, t3, xt);
  }
  function qr(e3, t3) {
    return jr(e3, t3, Tt);
  }
  function Wr(e3, t3, r2) {
    void 0 === r2 && (r2 = /* @__PURE__ */ new Set());
    var n2 = false, o2 = Cr(e3, function(e4) {
      var o3 = Tr(e4), a2 = o3.name, i2 = o3.fallback;
      if (r2.has(a2))
        return n2 = true, null;
      r2.add(a2);
      var u2 = t3.get(a2) || i2, s2 = null;
      return u2 && (s2 = Dr(u2) ? Wr(u2, t3, r2) : u2), s2 || (n2 = true, null);
    });
    return n2 ? null : o2;
  }
  var Hr = { "background-color": { customProp: "--darkreader-inline-bgcolor", cssProp: "background-color", dataAttr: "data-darkreader-inline-bgcolor" }, "background-image": { customProp: "--darkreader-inline-bgimage", cssProp: "background-image", dataAttr: "data-darkreader-inline-bgimage" }, "border-color": { customProp: "--darkreader-inline-border", cssProp: "border-color", dataAttr: "data-darkreader-inline-border" }, "border-bottom-color": { customProp: "--darkreader-inline-border-bottom", cssProp: "border-bottom-color", dataAttr: "data-darkreader-inline-border-bottom" }, "border-left-color": { customProp: "--darkreader-inline-border-left", cssProp: "border-left-color", dataAttr: "data-darkreader-inline-border-left" }, "border-right-color": { customProp: "--darkreader-inline-border-right", cssProp: "border-right-color", dataAttr: "data-darkreader-inline-border-right" }, "border-top-color": { customProp: "--darkreader-inline-border-top", cssProp: "border-top-color", dataAttr: "data-darkreader-inline-border-top" }, "box-shadow": { customProp: "--darkreader-inline-boxshadow", cssProp: "box-shadow", dataAttr: "data-darkreader-inline-boxshadow" }, color: { customProp: "--darkreader-inline-color", cssProp: "color", dataAttr: "data-darkreader-inline-color" }, fill: { customProp: "--darkreader-inline-fill", cssProp: "fill", dataAttr: "data-darkreader-inline-fill" }, stroke: { customProp: "--darkreader-inline-stroke", cssProp: "stroke", dataAttr: "data-darkreader-inline-stroke" }, "outline-color": { customProp: "--darkreader-inline-outline", cssProp: "outline-color", dataAttr: "data-darkreader-inline-outline" }, "stop-color": { customProp: "--darkreader-inline-stopcolor", cssProp: "stop-color", dataAttr: "data-darkreader-inline-stopcolor" } }, Gr = Object.values(Hr), $r = {};
  Gr.forEach(function(e3) {
    var t3 = e3.cssProp, r2 = e3.customProp;
    return $r[r2] = t3;
  });
  var zr = ["style", "fill", "stop-color", "stroke", "bgcolor", "color"], Qr = zr.map(function(e3) {
    return "[" + e3 + "]";
  }).join(", ");
  function Kr() {
    return Gr.map(function(e3) {
      var t3 = e3.dataAttr, r2 = e3.customProp;
      return ["[" + t3 + "] {", "  " + e3.cssProp + ": var(" + r2 + ") !important;", "}"].join("\n");
    }).join("\n");
  }
  function Xr(e3) {
    var t3 = [];
    return e3 instanceof Element && e3.matches(Qr) && t3.push(e3), (e3 instanceof Element || g && e3 instanceof ShadowRoot || e3 instanceof Document) && P(t3, e3.querySelectorAll(Qr)), t3;
  }
  var Yr = /* @__PURE__ */ new Map(), Zr = /* @__PURE__ */ new Map();
  function Jr(e3, t3) {
    en(document, e3, t3), q(document.documentElement, function(r2) {
      en(r2.shadowRoot, e3, t3);
    });
  }
  function en(e3, t3, r2) {
    Yr.has(e3) && (Yr.get(e3).disconnect(), Zr.get(e3).disconnect());
    var n2 = /* @__PURE__ */ new WeakSet();
    function o2(e4) {
      Xr(e4).forEach(function(e5) {
        n2.has(e5) || (n2.add(e5), t3(e5));
      }), q(e4, function(o3) {
        n2.has(e4) || (n2.add(e4), r2(o3.shadowRoot), en(o3.shadowRoot, t3, r2));
      });
    }
    var u2 = oe(e3, { onMinorMutations: function(e4) {
      e4.additions.forEach(function(e5) {
        return o2(e5);
      });
    }, onHugeMutations: function() {
      o2(e3);
    } });
    Yr.set(e3, u2);
    var s2 = 0, c2 = null, l2 = j({ seconds: 10 }), d2 = j({ seconds: 2 }), f2 = 50, h2 = [], p2 = null, v2 = F(function(e4) {
      e4.forEach(function(e5) {
        zr.includes(e5.attributeName) && t3(e5.target);
      });
    }), m2 = new MutationObserver(function(e4) {
      if (p2)
        h2.push.apply(h2, i([], a(e4), false));
      else {
        s2++;
        var t4 = Date.now();
        if (null == c2)
          c2 = t4;
        else if (s2 >= f2) {
          if (t4 - c2 < l2)
            return p2 = setTimeout(function() {
              c2 = null, s2 = 0, p2 = null;
              var e5 = h2;
              h2 = [], v2(e5);
            }, d2), void h2.push.apply(h2, i([], a(e4), false));
          c2 = t4, s2 = 1;
        }
        v2(e4);
      }
    });
    m2.observe(e3, { attributes: true, attributeFilter: zr.concat(Gr.map(function(e4) {
      return e4.dataAttr;
    })), subtree: true }), Zr.set(e3, m2);
  }
  function tn() {
    Yr.forEach(function(e3) {
      return e3.disconnect();
    }), Zr.forEach(function(e3) {
      return e3.disconnect();
    }), Yr.clear(), Zr.clear();
  }
  var rn = /* @__PURE__ */ new WeakMap(), nn = ["brightness", "contrast", "grayscale", "sepia", "mode"];
  function on(e3, t3) {
    return zr.map(function(t4) {
      return t4 + '="' + e3.getAttribute(t4) + '"';
    }).concat(nn.map(function(e4) {
      return e4 + '="' + t3[e4] + '"';
    })).join(" ");
  }
  function an(e3, t3) {
    for (var r2 = 0, n2 = t3.length; r2 < n2; r2++) {
      var o2 = t3[r2];
      if (e3.matches(o2))
        return true;
    }
    return false;
  }
  function un(e3, t3, r2, n2) {
    if (on(e3, t3) !== rn.get(e3)) {
      var o2 = new Set(Object.keys(Hr));
      if (r2.length > 0 && an(e3, r2))
        o2.forEach(function(t4) {
          e3.removeAttribute(Hr[t4].dataAttr);
        });
      else {
        if (e3.hasAttribute("bgcolor") && (((s2 = e3.getAttribute("bgcolor")).match(/^[0-9a-f]{3}$/i) || s2.match(/^[0-9a-f]{6}$/i)) && (s2 = "#" + s2), c2("background-color", "background-color", s2)), e3.hasAttribute("color") && "mask-icon" !== e3.rel && (((s2 = e3.getAttribute("color")).match(/^[0-9a-f]{3}$/i) || s2.match(/^[0-9a-f]{6}$/i)) && (s2 = "#" + s2), c2("color", "color", s2)), e3 instanceof SVGElement) {
          if (e3.hasAttribute("fill")) {
            var a2 = 32, i2 = e3.getAttribute("fill");
            if ("none" !== i2)
              if (e3 instanceof SVGTextElement)
                c2("fill", "color", i2);
              else {
                var u2 = function() {
                  var t4 = e3.getBoundingClientRect(), r3 = t4.width, n3 = t4.height;
                  c2("fill", r3 > a2 || n3 > a2 ? "background-color" : "color", i2);
                };
                z() ? u2() : K(u2);
              }
          }
          e3.hasAttribute("stop-color") && c2("stop-color", "background-color", e3.getAttribute("stop-color"));
        }
        if (e3.hasAttribute("stroke")) {
          var s2 = e3.getAttribute("stroke");
          c2("stroke", e3 instanceof SVGLineElement || e3 instanceof SVGTextElement ? "border-color" : "color", s2);
        }
        e3.style && he(e3.style, function(t4, r3) {
          if ("background-image" !== t4 || !r3.includes("url"))
            if (Hr.hasOwnProperty(t4))
              c2(t4, t4, r3);
            else {
              var n3 = $r[t4];
              if (n3 && !e3.style.getPropertyValue(n3) && !e3.hasAttribute(n3)) {
                if ("background-color" === n3 && e3.hasAttribute("bgcolor"))
                  return;
                e3.style.setProperty(t4, "");
              }
            }
        }), e3.style && e3 instanceof SVGTextElement && e3.style.fill && c2("fill", "color", e3.style.getPropertyValue("fill")), L(o2, function(t4) {
          e3.removeAttribute(Hr[t4].dataAttr);
        }), rn.set(e3, on(e3, t3));
      }
    }
    function c2(r3, a3, i3) {
      var u3 = Hr[r3], s3 = u3.customProp, c3 = u3.dataAttr, l2 = Zt(a3, i3, {}, wr, n2, null);
      if (l2) {
        var d2 = l2.value;
        "function" == typeof d2 && (d2 = d2(t3)), e3.style.setProperty(s3, d2), e3.hasAttribute(c3) || e3.setAttribute(c3, ""), o2.delete(r3);
      }
    }
  }
  var sn = "theme-color", cn = 'meta[name="' + sn + '"]', ln = null, dn = null;
  function fn(e3, t3) {
    ln = ln || e3.content;
    try {
      var r2 = Le(ln);
      e3.content = St(r2, t3);
    } catch (e4) {
    }
  }
  function hn(e3) {
    var t3 = document.querySelector(cn);
    t3 ? fn(t3, e3) : (dn && dn.disconnect(), (dn = new MutationObserver(function(t4) {
      e:
        for (var r2 = 0; r2 < t4.length; r2++)
          for (var n2 = t4[r2].addedNodes, o2 = 0; o2 < n2.length; o2++) {
            var a2 = n2[o2];
            if (a2 instanceof HTMLMetaElement && a2.name === sn) {
              dn.disconnect(), dn = null, fn(a2, e3);
              break e;
            }
          }
    })).observe(document.head, { childList: true }));
  }
  function pn() {
    dn && (dn.disconnect(), dn = null);
    var e3 = document.querySelector(cn);
    e3 && ln && (e3.content = ln);
  }
  var vn = ["mode", "brightness", "contrast", "grayscale", "sepia", "darkSchemeBackgroundColor", "darkSchemeTextColor", "lightSchemeBackgroundColor", "lightSchemeTextColor"];
  function mn(e3) {
    return vn.map(function(t3) {
      return t3 + ":" + e3[t3];
    }).join(";");
  }
  var gn = N();
  function bn() {
    var e3 = 0, t3 = /* @__PURE__ */ new Set(), r2 = /* @__PURE__ */ new Map(), n2 = /* @__PURE__ */ new Set(), o2 = null, u2 = false, s2 = false;
    function c2() {
      return u2 && !s2;
    }
    function l2(c3) {
      var l3 = c3.sourceCSSRules, d2 = c3.theme, f2 = c3.ignoreImageAnalysis, h2 = c3.force, p2 = c3.prepareSheet, v2 = c3.isAsyncCancelled, m2 = 0 === r2.size, g2 = new Set(r2.keys()), b2 = mn(d2), y2 = b2 !== o2;
      u2 && (s2 = true);
      var S2 = [];
      if (le(l3, function(e4) {
        var n3 = e4.cssText, o3 = false;
        if (g2.delete(n3), e4.parentRule instanceof CSSMediaRule && (n3 += ";" + e4.parentRule.media.mediaText), t3.has(n3) || (t3.add(n3), o3 = true), o3) {
          m2 = true;
          var a2 = [];
          e4.style && he(e4.style, function(t4, r3) {
            var n4 = Zt(t4, r3, e4, wr, f2, v2);
            n4 && a2.push(n4);
          });
          var i2 = null;
          if (a2.length > 0) {
            var u3 = e4.parentRule;
            i2 = { selector: e4.selectorText, declarations: a2, parentRule: u3 }, S2.push(i2);
          }
          r2.set(n3, i2);
        } else
          S2.push(r2.get(n3));
      }, function() {
        u2 = true;
      }), g2.forEach(function(e4) {
        t3.delete(e4), r2.delete(e4);
      }), o2 = b2, h2 || m2 || y2) {
        e3++;
        var _2 = /* @__PURE__ */ new Map(), k2 = /* @__PURE__ */ new Map(), w2 = 0, E2 = 0, x2 = { rule: null, rules: [], isGroup: true }, C2 = /* @__PURE__ */ new WeakMap();
        n2.forEach(function(e4) {
          return e4();
        }), n2.clear(), S2.filter(function(e4) {
          return e4;
        }).forEach(function(t4) {
          var r3 = t4.selector, o3 = t4.declarations, u3 = V2(t4.parentRule), s3 = { selector: r3, declarations: [], isGroup: false }, c4 = s3.declarations;
          function l4(t5, r4, n3, o4) {
            var a2 = ++w2, i2 = { property: t5, value: null, important: n3, asyncKey: a2, sourceValue: o4 };
            c4.push(i2);
            var u4 = e3;
            r4.then(function(t6) {
              t6 && !v2() && u4 === e3 && (i2.value = t6, gn.add(function() {
                v2() || u4 !== e3 || A2(a2);
              }));
            });
          }
          function f3(t5, r4, o4, u4) {
            var s4 = r4, d3 = s4.declarations, f4 = s4.onTypeChange, h3 = ++E2, p3 = e3, m3 = c4.length, g3 = [];
            if (0 === d3.length) {
              var b3 = { property: t5, value: u4, important: o4, sourceValue: u4, varKey: h3 };
              c4.push(b3), g3 = [b3];
            }
            d3.forEach(function(e4) {
              if (e4.value instanceof Promise)
                l4(e4.property, e4.value, o4, u4);
              else {
                var t6 = { property: e4.property, value: e4.value, important: o4, sourceValue: u4, varKey: h3 };
                c4.push(t6), g3.push(t6);
              }
            }), f4.addListener(function(t6) {
              if (!v2() && p3 === e3) {
                var r5 = t6.map(function(e4) {
                  return { property: e4.property, value: e4.value, important: o4, sourceValue: u4, varKey: h3 };
                }), n3 = c4.indexOf(g3[0], m3);
                c4.splice.apply(c4, i([n3, g3.length], a(r5), false)), g3 = r5, L2(h3);
              }
            }), n2.add(function() {
              return f4.removeListeners();
            });
          }
          u3.rules.push(s3), o3.forEach(function(e4) {
            var t5 = e4.property, r4 = e4.value, n3 = e4.important, o4 = e4.sourceValue;
            if ("function" == typeof r4) {
              var a2 = r4(d2);
              a2 instanceof Promise ? l4(t5, a2, n3, o4) : t5.startsWith("--") ? f3(t5, a2, n3, o4) : c4.push({ property: t5, value: a2, important: n3, sourceValue: o4 });
            } else
              c4.push({ property: t5, value: r4, important: n3, sourceValue: o4 });
          });
        });
        var T2 = p2();
        M2();
      }
      function R2(e4, t4, r3) {
        var n3 = r3.selector, o3 = r3.declarations, a2 = function(e5) {
          var t5 = e5.property, r4 = e5.value, n4 = e5.important, o4 = e5.sourceValue;
          return t5 + ": " + (null == r4 ? o4 : r4) + (n4 ? " !important" : "") + ";";
        }, i2 = "";
        o3.forEach(function(e5) {
          i2 += a2(e5) + " ";
        });
        var u3 = n3 + " { " + i2 + " }";
        e4.insertRule(u3, t4);
      }
      function V2(e4) {
        if (null == e4)
          return x2;
        if (C2.has(e4))
          return C2.get(e4);
        var t4 = { rule: e4, rules: [], isGroup: true };
        return C2.set(e4, t4), V2(e4.parentRule).rules.push(t4), t4;
      }
      function M2() {
        function e4(e5, t5) {
          var r3 = e5.rule;
          if (r3 instanceof CSSMediaRule) {
            var n3 = r3.media, o3 = t5.cssRules.length;
            return t5.insertRule("@media " + n3.mediaText + " {}", o3), t5.cssRules[o3];
          }
          return t5;
        }
        function t4(r3, n3, o3) {
          r3.rules.forEach(function(r4) {
            r4.isGroup ? t4(r4, e4(r4, n3), o3) : o3(r4, n3);
          });
        }
        t4(x2, T2, function(e5, t5) {
          var r3 = t5.cssRules.length;
          e5.declarations.forEach(function(n3) {
            var o3 = n3.asyncKey, a2 = n3.varKey;
            null != o3 && _2.set(o3, { rule: e5, target: t5, index: r3 }), null != a2 && k2.set(a2, { rule: e5, target: t5, index: r3 });
          }), R2(t5, r3, e5);
        });
      }
      function A2(e4) {
        var t4 = _2.get(e4), r3 = t4.rule, n3 = t4.target, o3 = t4.index;
        n3.deleteRule(o3), R2(n3, o3, r3), _2.delete(e4);
      }
      function L2(e4) {
        var t4 = k2.get(e4), r3 = t4.rule, n3 = t4.target, o3 = t4.index;
        n3.deleteRule(o3), R2(n3, o3, r3);
      }
    }
    return { modifySheet: l2, shouldRebuildStyle: c2 };
  }
  var yn = 'style, link[rel*="stylesheet" i]:not([disabled])';
  function Sn(e3) {
    return (e3 instanceof HTMLStyleElement || e3 instanceof SVGStyleElement || e3 instanceof HTMLLinkElement && e3.rel && e3.rel.toLowerCase().includes("stylesheet") && !e3.disabled && (!f || !e3.href.startsWith("moz-extension://"))) && !e3.classList.contains("darkreader") && "print" !== e3.media.toLowerCase() && !e3.classList.contains("stylus");
  }
  function _n(e3, t3, r2) {
    return void 0 === t3 && (t3 = []), void 0 === r2 && (r2 = true), Sn(e3) ? t3.push(e3) : (e3 instanceof Element || g && e3 instanceof ShadowRoot || e3 === document) && (L(e3.querySelectorAll(yn), function(e4) {
      return _n(e4, t3, false);
    }), r2 && q(e3, function(e4) {
      return _n(e4.shadowRoot, t3, false);
    })), t3;
  }
  var kn = /* @__PURE__ */ new WeakSet(), wn = /* @__PURE__ */ new WeakSet(), En = false;
  "undefined" != typeof document && document.addEventListener("__darkreader__inlineScriptsAllowed", function() {
    En = true;
  });
  var xn = 0, Cn = /* @__PURE__ */ new Map();
  function Tn() {
    Cn.clear();
  }
  function Rn(e3, t3) {
    for (var o2 = t3.update, i2 = t3.loadingStart, u2 = t3.loadingEnd, s2 = [], c2 = e3; (c2 = c2.nextElementSibling) && c2.matches(".darkreader"); )
      s2.push(c2);
    var f2 = s2.find(function(e4) {
      return e4.matches(".darkreader--cors") && !wn.has(e4);
    }) || null, p2 = s2.find(function(e4) {
      return e4.matches(".darkreader--sync") && !kn.has(e4);
    }) || null, v2 = null, m2 = null, g2 = false, b2 = true, y2 = bn(), S2 = new MutationObserver(function() {
      o2();
    }), _2 = { attributes: true, childList: true, subtree: true, characterData: true };
    function k2() {
      return e3 instanceof HTMLStyleElement && e3.textContent.trim().match(ve);
    }
    function w2(e4, t4) {
      var r2 = false;
      if (e4) {
        var n2 = void 0;
        e:
          for (var o3 = 0, a2 = e4.length; o3 < a2; o3++)
            if ((n2 = e4[o3]).href) {
              if (!t4) {
                r2 = true;
                break e;
              }
              if (n2.href.startsWith("http") && !n2.href.startsWith(location.origin)) {
                r2 = true;
                break e;
              }
            }
      }
      return r2;
    }
    function E2() {
      if (f2)
        return f2.sheet.cssRules;
      if (k2())
        return null;
      var t4 = I2();
      return e3 instanceof HTMLLinkElement && !ce(e3.href) && w2(t4, false) || w2(t4, true) ? null : t4;
    }
    function x2() {
      f2 ? (e3.nextSibling !== f2 && e3.parentNode.insertBefore(f2, e3.nextSibling), f2.nextSibling !== p2 && e3.parentNode.insertBefore(p2, f2.nextSibling)) : e3.nextSibling !== p2 && e3.parentNode.insertBefore(p2, e3.nextSibling);
    }
    function C2() {
      (p2 = e3 instanceof SVGStyleElement ? document.createElementNS("http://www.w3.org/2000/svg", "style") : document.createElement("style")).classList.add("darkreader"), p2.classList.add("darkreader--sync"), p2.media = "screen", !l && e3.title && (p2.title = e3.title), kn.add(p2);
    }
    var T2 = false, R2 = false, V2 = ++xn;
    function M2() {
      return r(this, void 0, void 0, function() {
        var t4, r2, o3, i3, u3, s3, c3;
        return n(this, function(n2) {
          switch (n2.label) {
            case 0:
              if (!(e3 instanceof HTMLLinkElement))
                return [3, 7];
              if (o3 = a(O2(), 2), i3 = o3[0], u3 = o3[1], (i3 || u3 || h) && (!h || e3.sheet) && !D2(u3))
                return [3, 5];
              n2.label = 1;
            case 1:
              return n2.trys.push([1, 3, , 4]), [4, Vn(e3, V2)];
            case 2:
              return n2.sent(), [3, 4];
            case 3:
              return n2.sent(), R2 = true, [3, 4];
            case 4:
              if (g2)
                return [2, null];
              c3 = a(O2(), 2), i3 = c3[0], u3 = c3[1], n2.label = 5;
            case 5:
              return i3 && !w2(i3, false) ? [2, i3] : [4, An(e3.href)];
            case 6:
              return t4 = n2.sent(), r2 = ge(e3.href), g2 ? [2, null] : [3, 8];
            case 7:
              if (!k2())
                return [2, null];
              t4 = e3.textContent.trim(), r2 = ge(location.href), n2.label = 8;
            case 8:
              if (!t4)
                return [3, 13];
              n2.label = 9;
            case 9:
              return n2.trys.push([9, 11, , 12]), [4, Ln(t4, r2)];
            case 10:
              return s3 = n2.sent(), f2 = Pn(e3, s3), [3, 12];
            case 11:
              return n2.sent(), [3, 12];
            case 12:
              if (f2)
                return v2 = U(f2, "prev-sibling"), [2, f2.sheet.cssRules];
              n2.label = 13;
            case 13:
              return [2, null];
          }
        });
      });
    }
    function A2(e4) {
      var t4 = E2();
      return t4 ? { rules: t4 } : (e4.secondRound || T2 || R2 || (T2 = true, i2(), M2().then(function(e5) {
        T2 = false, u2(), e5 && o2();
      }).catch(function(e5) {
        T2 = false, u2();
      })), null);
    }
    var L2 = false;
    function P2(e4, t4) {
      var r2 = E2();
      function n2(e5) {
        try {
          if (e5.replaceSync)
            return void e5.replaceSync("");
        } catch (e6) {
        }
        for (var t5 = e5.cssRules.length - 1; t5 >= 0; t5--)
          e5.deleteRule(t5);
      }
      function a2() {
        return p2 || C2(), m2 && m2.stop(), x2(), null == p2.sheet && (p2.textContent = ""), n2(p2.sheet), m2 ? m2.run() : m2 = U(p2, "prev-sibling", function() {
          L2 = true, i3();
        }), p2.sheet;
      }
      function i3() {
        var n3 = L2;
        L2 = false, y2.modifySheet({ prepareSheet: a2, sourceCSSRules: r2, theme: e4, ignoreImageAnalysis: t4, force: n3, isAsyncCancelled: function() {
          return g2;
        } }), b2 = 0 === p2.sheet.cssRules.length, y2.shouldRebuildStyle() && K(function() {
          return o2();
        });
      }
      r2 && (g2 = false, i3());
    }
    function O2() {
      try {
        return null == e3.sheet ? [null, null] : [e3.sheet.cssRules, null];
      } catch (e4) {
        return [null, e4];
      }
    }
    function D2(e4) {
      return e4 && e4.message && e4.message.includes("loading");
    }
    function I2() {
      var e4 = a(O2(), 2), t4 = e4[0];
      return e4[1] ? null : t4;
    }
    function F2() {
      Q2(), d || En && e3.sheet || H2();
    }
    var N2 = null, j2 = null;
    function q2() {
      var e4 = I2();
      return e4 ? e4.length : null;
    }
    function W2() {
      return q2() !== N2;
    }
    function H2() {
      N2 = q2(), G2();
      var t4 = function() {
        W2() && (N2 = q2(), o2()), En && e3.sheet ? G2() : j2 = requestAnimationFrame(t4);
      };
      t4();
    }
    function G2() {
      cancelAnimationFrame(j2);
    }
    var $2 = false;
    function z2() {
      function e4() {
        $2 = false, g2 || o2();
      }
      En = true, G2(), $2 || ($2 = true, "function" == typeof queueMicrotask ? queueMicrotask(e4) : requestAnimationFrame(e4));
    }
    function Q2() {
      e3.addEventListener("__darkreader__updateSheet", z2);
    }
    function X2() {
      e3.removeEventListener("__darkreader__updateSheet", z2);
    }
    function Y2() {
      X2(), G2();
    }
    function Z2() {
      S2.disconnect(), g2 = true, v2 && v2.stop(), m2 && m2.stop(), Y2();
    }
    function J2() {
      if (Z2(), B(f2), B(p2), u2(), Cn.has(V2)) {
        var e4 = Cn.get(V2);
        Cn.delete(V2), e4 && e4();
      }
    }
    function ee2() {
      S2.observe(e3, _2), e3 instanceof HTMLStyleElement && F2();
    }
    var te2 = 10, re2 = 0;
    function ne2() {
      p2 && (++re2 > te2 || (x2(), v2 && v2.skip(), m2 && m2.skip(), b2 || (L2 = true, o2())));
    }
    return { details: A2, render: P2, pause: Z2, destroy: J2, watch: ee2, restore: ne2 };
  }
  function Vn(e3, t3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(r2) {
        return [2, new Promise(function(r3, n2) {
          var o2 = function() {
            e3.removeEventListener("load", a2), e3.removeEventListener("error", i2), Cn.delete(t3);
          }, a2 = function() {
            o2(), r3();
          }, i2 = function() {
            o2(), n2("Linkelement " + t3 + " couldn't be loaded. " + e3.href);
          };
          Cn.set(t3, function() {
            o2(), n2();
          }), e3.addEventListener("load", a2), e3.addEventListener("error", i2), e3.href || i2();
        })];
      });
    });
  }
  function Mn(e3) {
    return me(e3.substring(7).trim().replace(/;$/, "").replace(/screen$/, ""));
  }
  function An(e3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(t3) {
        switch (t3.label) {
          case 0:
            return e3.startsWith("data:") ? [4, fetch(e3)] : [3, 3];
          case 1:
            return [4, t3.sent().text()];
          case 2:
          case 4:
            return [2, t3.sent()];
          case 3:
            return [4, Ft({ url: e3, responseType: "text", mimeType: "text/css", origin: window.location.origin })];
        }
      });
    });
  }
  function Ln(e3, t3, a2) {
    return void 0 === a2 && (a2 = /* @__PURE__ */ new Map()), r(this, void 0, void 0, function() {
      var r2, i2, u2, s2, c2, l2, d2, f2, h2, p2;
      return n(this, function(n2) {
        switch (n2.label) {
          case 0:
            e3 = be(e3 = ke(e3 = Se(e3)), t3), r2 = Ze(ve, e3), n2.label = 1;
          case 1:
            n2.trys.push([1, 10, 11, 12]), i2 = o(r2), u2 = i2.next(), n2.label = 2;
          case 2:
            return u2.done ? [3, 9] : (s2 = u2.value, c2 = Mn(s2), l2 = se(t3, c2), d2 = void 0, a2.has(l2) ? (d2 = a2.get(l2), [3, 7]) : [3, 3]);
          case 3:
            return n2.trys.push([3, 6, , 7]), [4, An(l2)];
          case 4:
            return d2 = n2.sent(), a2.set(l2, d2), [4, Ln(d2, ge(l2), a2)];
          case 5:
            return d2 = n2.sent(), [3, 7];
          case 6:
            return n2.sent(), d2 = "", [3, 7];
          case 7:
            e3 = e3.split(s2).join(d2), n2.label = 8;
          case 8:
            return u2 = i2.next(), [3, 2];
          case 9:
            return [3, 12];
          case 10:
            return f2 = n2.sent(), h2 = { error: f2 }, [3, 12];
          case 11:
            try {
              u2 && !u2.done && (p2 = i2.return) && p2.call(i2);
            } finally {
              if (h2)
                throw h2.error;
            }
            return [7];
          case 12:
            return [2, e3 = e3.trim()];
        }
      });
    });
  }
  function Pn(e3, t3) {
    if (!t3)
      return null;
    var r2 = document.createElement("style");
    return r2.classList.add("darkreader"), r2.classList.add("darkreader--cors"), r2.media = "screen", r2.textContent = t3, e3.parentNode.insertBefore(r2, e3.nextSibling), r2.sheet.disabled = true, wn.add(r2), r2;
  }
  var On, Dn, In = [], Fn = /* @__PURE__ */ new Map();
  function Nn(e3) {
    y && L(e3.querySelectorAll(":not(:defined)"), function(e4) {
      var t3 = e4.tagName.toLowerCase();
      if (!t3.includes("-")) {
        var r2 = e4.getAttribute("is");
        if (!r2)
          return;
        t3 = r2;
      }
      Fn.has(t3) || (Fn.set(t3, /* @__PURE__ */ new Set()), qn(t3).then(function() {
        if (Dn) {
          var e5 = Fn.get(t3);
          Fn.delete(t3), Dn(Array.from(e5));
        }
      })), Fn.get(t3).add(e4);
    });
  }
  var jn = false;
  "undefined" != typeof document && document.addEventListener("__darkreader__inlineScriptsAllowed", function() {
    jn = true;
  });
  var Bn = /* @__PURE__ */ new Map();
  function Un(e3) {
    jn = true, Bn.has(e3.detail.tag) && Bn.get(e3.detail.tag)();
  }
  function qn(e3) {
    return r(this, void 0, void 0, function() {
      return n(this, function(t3) {
        return [2, new Promise(function(t4) {
          if (window.customElements && "function" == typeof customElements.whenDefined)
            customElements.whenDefined(e3).then(function() {
              return t4();
            });
          else if (jn)
            Bn.set(e3, t4), document.dispatchEvent(new CustomEvent("__darkreader__addUndefinedResolver", { detail: { tag: e3 } }));
          else {
            var r2 = function() {
              var n2 = Fn.get(e3);
              n2 && n2.size > 0 && (n2.values().next().value.matches(":defined") ? t4() : requestAnimationFrame(r2));
            };
            requestAnimationFrame(r2);
          }
        })];
      });
    });
  }
  function Wn(e3) {
    Dn = e3;
  }
  function Hn() {
    Dn = null, Fn.clear(), document.removeEventListener("__darkreader__isDefined", Un);
  }
  function Gn(e3, t3, r2) {
    zn();
    var n2 = new Set(e3), o2 = /* @__PURE__ */ new WeakMap(), a2 = /* @__PURE__ */ new WeakMap();
    function i2(e4) {
      o2.set(e4, e4.previousElementSibling), a2.set(e4, e4.nextElementSibling);
    }
    function u2(e4) {
      o2.delete(e4), a2.delete(e4);
    }
    function s2(e4) {
      return e4.previousElementSibling !== o2.get(e4) || e4.nextElementSibling !== a2.get(e4);
    }
    function c2(e4) {
      var r3 = e4.createdStyles, o3 = e4.removedStyles, a3 = e4.movedStyles;
      r3.forEach(function(e5) {
        return i2(e5);
      }), a3.forEach(function(e5) {
        return i2(e5);
      }), o3.forEach(function(e5) {
        return u2(e5);
      }), r3.forEach(function(e5) {
        return n2.add(e5);
      }), o3.forEach(function(e5) {
        return n2.delete(e5);
      }), r3.size + o3.size + a3.size > 0 && t3({ created: Array.from(r3), removed: Array.from(o3), moved: Array.from(a3), updated: [] });
    }
    function l2(e4) {
      var t4 = e4.additions, r3 = e4.moves, n3 = e4.deletions, o3 = /* @__PURE__ */ new Set(), a3 = /* @__PURE__ */ new Set(), i3 = /* @__PURE__ */ new Set();
      t4.forEach(function(e5) {
        return _n(e5).forEach(function(e6) {
          return o3.add(e6);
        });
      }), n3.forEach(function(e5) {
        return _n(e5).forEach(function(e6) {
          return a3.add(e6);
        });
      }), r3.forEach(function(e5) {
        return _n(e5).forEach(function(e6) {
          return i3.add(e6);
        });
      }), c2({ createdStyles: o3, removedStyles: a3, movedStyles: i3 }), t4.forEach(function(e5) {
        q(e5, p2), Nn(e5);
      });
    }
    function d2(e4) {
      var t4 = new Set(_n(e4)), r3 = /* @__PURE__ */ new Set(), o3 = /* @__PURE__ */ new Set(), a3 = /* @__PURE__ */ new Set();
      t4.forEach(function(e5) {
        n2.has(e5) || r3.add(e5);
      }), n2.forEach(function(e5) {
        t4.has(e5) || o3.add(e5);
      }), t4.forEach(function(e5) {
        r3.has(e5) || o3.has(e5) || !s2(e5) || a3.add(e5);
      }), c2({ createdStyles: r3, removedStyles: o3, movedStyles: a3 }), q(e4, p2), Nn(e4);
    }
    function f2(e4) {
      var r3 = /* @__PURE__ */ new Set(), n3 = /* @__PURE__ */ new Set();
      e4.forEach(function(e5) {
        var t4 = e5.target;
        t4.isConnected && (Sn(t4) ? r3.add(t4) : t4 instanceof HTMLLinkElement && t4.disabled && n3.add(t4));
      }), r3.size + n3.size > 0 && t3({ updated: Array.from(r3), created: [], removed: Array.from(n3), moved: [] });
    }
    function h2(e4) {
      var t4 = oe(e4, { onMinorMutations: l2, onHugeMutations: d2 }), r3 = new MutationObserver(f2);
      r3.observe(e4, { attributes: true, attributeFilter: ["rel", "disabled", "media"], subtree: true }), In.push(t4, r3), On.add(e4);
    }
    function p2(e4) {
      var t4 = e4.shadowRoot;
      null == t4 || On.has(t4) || (h2(t4), r2(t4));
    }
    e3.forEach(i2), h2(document), q(document.documentElement, p2), Wn(function(e4) {
      var r3 = [];
      e4.forEach(function(e5) {
        return P(r3, _n(e5.shadowRoot));
      }), t3({ created: r3, updated: [], removed: [], moved: [] }), e4.forEach(function(e5) {
        var t4 = e5.shadowRoot;
        null != t4 && (p2(e5), q(t4, p2), Nn(t4));
      });
    }), document.addEventListener("__darkreader__isDefined", Un), Nn(document);
  }
  function $n() {
    In.forEach(function(e3) {
      return e3.disconnect();
    }), In.splice(0, In.length), On = /* @__PURE__ */ new WeakSet();
  }
  function zn() {
    $n(), Hn();
  }
  function Qn(e3) {
    return (e3 < 16 ? "0" : "") + e3.toString(16);
  }
  function Kn() {
    if ("undefined" == typeof crypto)
      return "xxx-xxxx";
    if ("randomUUID" in crypto) {
      var e3 = crypto.randomUUID();
      return e3.substring(0, 8) + e3.substring(9, 13) + e3.substring(14, 18) + e3.substring(19, 23) + e3.substring(24);
    }
    return Array.from(crypto.getRandomValues(new Uint8Array(16))).map(function(e4) {
      return Qn(e4);
    }).join("");
  }
  var Xn = /* @__PURE__ */ new WeakMap(), Yn = /* @__PURE__ */ new WeakSet();
  function Zn(e3) {
    var t3 = false;
    function r2(t4, r3) {
      var n3 = i([], a(e3.adoptedStyleSheets), false), o3 = n3.indexOf(t4), u2 = n3.indexOf(r3);
      o3 !== u2 - 1 && (u2 >= 0 && n3.splice(u2, 1), n3.splice(o3 + 1, 0, r3), e3.adoptedStyleSheets = n3);
    }
    function n2() {
      t3 = true;
      var r3 = i([], a(e3.adoptedStyleSheets), false);
      e3.adoptedStyleSheets.forEach(function(e4) {
        if (Yn.has(e4)) {
          var t4 = r3.indexOf(e4);
          t4 >= 0 && r3.splice(t4, 1), Xn.delete(e4), Yn.delete(e4);
        }
      }), e3.adoptedStyleSheets = r3;
    }
    function o2(n3, o3) {
      e3.adoptedStyleSheets.forEach(function(e4) {
        if (!Yn.has(e4)) {
          var a2 = e4.rules, i2 = new CSSStyleSheet();
          bn().modifySheet({ prepareSheet: u2, sourceCSSRules: a2, theme: n3, ignoreImageAnalysis: o3, force: false, isAsyncCancelled: function() {
            return t3;
          } });
        }
        function u2() {
          for (var t4 = i2.cssRules.length - 1; t4 >= 0; t4--)
            i2.deleteRule(t4);
          return r2(e4, i2), Xn.set(e4, i2), Yn.add(i2), i2;
        }
      });
    }
    return { render: o2, destroy: n2 };
  }
  function Jn(e3) {
    document.dispatchEvent(new CustomEvent("__darkreader__inlineScriptsAllowed"));
    var t3 = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, "addRule"), r2 = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, "insertRule"), n2 = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, "deleteRule"), o2 = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, "removeRule"), u2 = e3 ? Object.getOwnPropertyDescriptor(Document.prototype, "styleSheets") : null, s2 = location.hostname.endsWith("baidu.com"), c2 = s2 ? Object.getOwnPropertyDescriptor(Element.prototype, "getElementsByTagName") : null, l2 = function() {
      Object.defineProperty(CSSStyleSheet.prototype, "addRule", t3), Object.defineProperty(CSSStyleSheet.prototype, "insertRule", r2), Object.defineProperty(CSSStyleSheet.prototype, "deleteRule", n2), Object.defineProperty(CSSStyleSheet.prototype, "removeRule", o2), document.removeEventListener("__darkreader__cleanUp", l2), document.removeEventListener("__darkreader__addUndefinedResolver", d2), e3 && Object.defineProperty(Document.prototype, "styleSheets", u2), s2 && Object.defineProperty(Element.prototype, "getElementsByTagName", c2);
    }, d2 = function(e4) {
      customElements.whenDefined(e4.detail.tag).then(function() {
        document.dispatchEvent(new CustomEvent("__darkreader__isDefined", { detail: { tag: e4.detail.tag } }));
      });
    };
    document.addEventListener("__darkreader__cleanUp", l2), document.addEventListener("__darkreader__addUndefinedResolver", d2);
    var f2 = new Event("__darkreader__updateSheet");
    function h2(e4, r3, n3) {
      return t3.value.call(this, e4, r3, n3), this.ownerNode && !this.ownerNode.classList.contains("darkreader") && this.ownerNode.dispatchEvent(f2), -1;
    }
    function p2(e4, t4) {
      var n3 = r2.value.call(this, e4, t4);
      return this.ownerNode && !this.ownerNode.classList.contains("darkreader") && this.ownerNode.dispatchEvent(f2), n3;
    }
    function v2(e4) {
      n2.value.call(this, e4), this.ownerNode && !this.ownerNode.classList.contains("darkreader") && this.ownerNode.dispatchEvent(f2);
    }
    function m2(e4) {
      o2.value.call(this, e4), this.ownerNode && !this.ownerNode.classList.contains("darkreader") && this.ownerNode.dispatchEvent(f2);
    }
    function g2() {
      var e4 = i([], a(u2.get.call(this)), false).filter(function(e5) {
        return !e5.ownerNode.classList.contains("darkreader");
      });
      return Object.setPrototypeOf(e4, StyleSheetList.prototype);
    }
    function b2(e4) {
      var t4 = this;
      if ("style" !== e4)
        return c2.value.call(this, e4);
      var r3 = function() {
        var r4 = c2.value.call(t4, e4);
        return Object.setPrototypeOf(i([], a(r4), false).filter(function(e5) {
          return !e5.classList.contains("darkreader");
        }), NodeList.prototype);
      }, n3 = r3();
      return n3 = new Proxy(n3, { get: function(e5, t5) {
        return r3()[Number(t5)];
      } });
    }
    Object.defineProperty(CSSStyleSheet.prototype, "addRule", Object.assign({}, t3, { value: h2 })), Object.defineProperty(CSSStyleSheet.prototype, "insertRule", Object.assign({}, r2, { value: p2 })), Object.defineProperty(CSSStyleSheet.prototype, "deleteRule", Object.assign({}, n2, { value: v2 })), Object.defineProperty(CSSStyleSheet.prototype, "removeRule", Object.assign({}, o2, { value: m2 })), e3 && Object.defineProperty(Document.prototype, "styleSheets", Object.assign({}, u2, { get: g2 })), s2 && Object.defineProperty(Element.prototype, "getElementsByTagName", Object.assign({}, c2, { value: b2 }));
  }
  var eo = Kn(), to = /* @__PURE__ */ new Map(), ro = [], no = null, oo = null, ao = null, io = null, uo = null;
  function so(e3, t3) {
    void 0 === t3 && (t3 = document.head || document);
    var r2 = t3.querySelector("." + e3);
    return r2 || ((r2 = document.createElement("style")).classList.add("darkreader"), r2.classList.add(e3), r2.media = "screen", r2.textContent = ""), r2;
  }
  function co(e3, t3) {
    void 0 === t3 && (t3 = document.head || document);
    var r2 = t3.querySelector("." + e3);
    return r2 || ((r2 = document.createElement("script")).classList.add("darkreader"), r2.classList.add(e3)), r2;
  }
  var lo = /* @__PURE__ */ new Map();
  function fo(e3, t3) {
    lo.has(t3) && lo.get(t3).stop(), lo.set(t3, U(e3, "parent"));
  }
  function ho() {
    L(lo.values(), function(e3) {
      return e3.stop();
    }), lo.clear();
  }
  function po() {
    var e3 = so("darkreader--fallback", document);
    e3.textContent = nr(no, { strict: true }), document.head.insertBefore(e3, document.head.firstChild), fo(e3, "fallback");
    var r2 = so("darkreader--user-agent");
    r2.textContent = Jt(no, ao, no.styleSystemControls), document.head.insertBefore(r2, e3.nextSibling), fo(r2, "user-agent");
    var n2 = so("darkreader--text");
    no.useFont || no.textStroke > 0 ? n2.textContent = Mt(no) : n2.textContent = "", document.head.insertBefore(n2, e3.nextSibling), fo(n2, "text");
    var o2 = so("darkreader--invert");
    oo && Array.isArray(oo.invert) && oo.invert.length > 0 ? o2.textContent = [oo.invert.join(", ") + " {", "    filter: " + At(t2(t2({}, no), { contrast: 0 === no.mode ? no.contrast : Xe(no.contrast - 10, 0, 100) })) + " !important;", "}"].join("\n") : o2.textContent = "", document.head.insertBefore(o2, n2.nextSibling), fo(o2, "invert");
    var a2 = so("darkreader--inline");
    a2.textContent = Kr(), document.head.insertBefore(a2, o2.nextSibling), fo(a2, "inline");
    var i2 = so("darkreader--override");
    i2.textContent = oo && oo.css ? go(oo.css) : "", document.head.appendChild(i2), fo(i2, "override");
    var u2 = so("darkreader--variables"), s2 = er(no), c2 = no.darkSchemeBackgroundColor, l2 = no.darkSchemeTextColor, d2 = no.lightSchemeBackgroundColor, f2 = no.lightSchemeTextColor, h2 = no.mode, p2 = 0 === h2 ? d2 : c2, v2 = 0 === h2 ? f2 : l2;
    p2 = St(Le(p2), no), v2 = xt(Le(v2), no), u2.textContent = [":root {", "   --darkreader-neutral-background: " + p2 + ";", "   --darkreader-neutral-text: " + v2 + ";", "   --darkreader-selection-background: " + s2.backgroundColorSelection + ";", "   --darkreader-selection-text: " + s2.foregroundColorSelection + ";", "}"].join("\n"), document.head.insertBefore(u2, a2.nextSibling), fo(u2, "variables");
    var m2 = so("darkreader--root-vars");
    document.head.insertBefore(m2, u2.nextSibling);
    var g2 = co("darkreader--proxy");
    g2.append("(" + Jn + ")(!" + (oo && oo.disableStyleSheetsProxy) + ")"), document.head.insertBefore(g2, m2.nextSibling), g2.remove();
  }
  var vo = /* @__PURE__ */ new Set();
  function mo(e3) {
    var r2 = so("darkreader--inline", e3);
    r2.textContent = Kr(), e3.insertBefore(r2, e3.firstChild);
    var n2 = so("darkreader--override", e3);
    n2.textContent = oo && oo.css ? go(oo.css) : "", e3.insertBefore(n2, r2.nextSibling);
    var o2 = so("darkreader--invert", e3);
    oo && Array.isArray(oo.invert) && oo.invert.length > 0 ? o2.textContent = [oo.invert.join(", ") + " {", "    filter: " + At(t2(t2({}, no), { contrast: 0 === no.mode ? no.contrast : Xe(no.contrast - 10, 0, 100) })) + " !important;", "}"].join("\n") : o2.textContent = "", e3.insertBefore(o2, n2.nextSibling), vo.add(e3);
  }
  function go(e3) {
    return e3.replace(/\${(.+?)}/g, function(e4, t3) {
      var r2 = ur(t3);
      return r2 ? vt(r2, no) : t3;
    });
  }
  function bo() {
    var e3 = document.querySelector(".darkreader--fallback");
    e3 && (e3.textContent = "");
  }
  function yo() {
    xo();
    var e3 = _n(document).filter(function(e4) {
      return !to.has(e4);
    }).map(function(e4) {
      return ko(e4);
    });
    e3.map(function(e4) {
      return e4.details({ secondRound: false });
    }).filter(function(e4) {
      return e4 && e4.rules.length > 0;
    }).forEach(function(e4) {
      wr.addRulesForMatching(e4.rules);
    }), wr.matchVariablesAndDependants(), wr.setOnRootVariableChange(function() {
      wr.putRootVars(document.head.querySelector(".darkreader--root-vars"), no);
    }), wr.putRootVars(document.head.querySelector(".darkreader--root-vars"), no), to.forEach(function(e4) {
      return e4.render(no, io);
    }), 0 === _o.size && bo(), e3.forEach(function(e4) {
      return e4.watch();
    });
    var t3 = O(document.querySelectorAll(Qr));
    q(document.documentElement, function(e4) {
      mo(e4.shadowRoot);
      var r2 = e4.shadowRoot.querySelectorAll(Qr);
      r2.length > 0 && P(t3, r2);
    }), t3.forEach(function(e4) {
      return un(e4, no, uo, io);
    }), Lo(document);
  }
  var So = 0, _o = /* @__PURE__ */ new Set();
  function ko(e3) {
    var t3 = ++So;
    function r2() {
      if (!W() || !Ro) {
        _o.add(t3), D("Current amount of styles loading: " + _o.size);
        var e4 = document.querySelector(".darkreader--fallback");
        e4.textContent || (e4.textContent = nr(no, { strict: false }));
      }
    }
    function n2() {
      _o.delete(t3), D("Removed loadingStyle " + t3 + ", now awaiting: " + _o.size), 0 === _o.size && W() && bo();
    }
    function o2() {
      var e4 = a2.details({ secondRound: true });
      e4 && (wr.addRulesForMatching(e4.rules), wr.matchVariablesAndDependants(), a2.render(no, io));
    }
    var a2 = Rn(e3, { update: o2, loadingStart: r2, loadingEnd: n2 });
    return to.set(e3, a2), a2;
  }
  function wo(e3) {
    var t3 = to.get(e3);
    t3 && (t3.destroy(), to.delete(e3));
  }
  var Eo = F(function(e3) {
    to.forEach(function(e4) {
      return e4.render(no, io);
    }), ro.forEach(function(e4) {
      return e4.render(no, io);
    }), e3 && e3();
  }), xo = function() {
    Eo.cancel();
  };
  function Co() {
    0 !== _o.size || bo();
  }
  var To = null, Ro = "undefined" != typeof document && !document.hidden;
  function Vo(e3) {
    var t3 = Boolean(To);
    To = function() {
      document.hidden || (Mo(), e3(), Ro = true);
    }, t3 || document.addEventListener("visibilitychange", To);
  }
  function Mo() {
    document.removeEventListener("visibilitychange", To), To = null;
  }
  function Ao() {
    function e3() {
      yo(), Po();
    }
    po(), document.hidden ? Vo(e3) : e3(), hn(no);
  }
  function Lo(e3) {
    if (Array.isArray(e3.adoptedStyleSheets) && e3.adoptedStyleSheets.length > 0) {
      var t3 = Zn(e3);
      ro.push(t3), t3.render(no, io);
    }
  }
  function Po() {
    Gn(Array.from(to.keys()), function(e3) {
      var t3 = e3.created, r2 = e3.updated, n2 = e3.removed, o2 = e3.moved, a2 = n2, i2 = t3.concat(r2).concat(o2).filter(function(e4) {
        return !to.has(e4);
      }), u2 = o2.filter(function(e4) {
        return to.has(e4);
      });
      a2.forEach(function(e4) {
        return wo(e4);
      });
      var s2 = i2.map(function(e4) {
        return ko(e4);
      });
      s2.map(function(e4) {
        return e4.details({ secondRound: false });
      }).filter(function(e4) {
        return e4 && e4.rules.length > 0;
      }).forEach(function(e4) {
        wr.addRulesForMatching(e4.rules);
      }), wr.matchVariablesAndDependants(), s2.forEach(function(e4) {
        return e4.render(no, io);
      }), s2.forEach(function(e4) {
        return e4.watch();
      }), u2.forEach(function(e4) {
        return to.get(e4).restore();
      });
    }, function(e3) {
      mo(e3), Lo(e3);
    }), Jr(function(e3) {
      un(e3, no, uo, io), e3 === document.documentElement && e3.getAttribute("style").includes("--") && (wr.matchVariablesAndDependants(), wr.putRootVars(document.head.querySelector(".darkreader--root-vars"), no));
    }, function(e3) {
      mo(e3);
      var t3 = e3.querySelectorAll(Qr);
      t3.length > 0 && L(t3, function(e4) {
        return un(e4, no, uo, io);
      });
    }), G(Co);
  }
  function Oo() {
    to.forEach(function(e3) {
      return e3.pause();
    }), ho(), zn(), tn(), $(Co), X();
  }
  function Do() {
    var e3 = document.createElement("meta");
    e3.name = "darkreader", e3.content = eo, document.head.appendChild(e3);
  }
  function Io() {
    var e3 = document.querySelector('meta[name="darkreader"]');
    return e3 ? e3.content !== eo : (Do(), false);
  }
  function Fo(e3, t3, r2) {
    if (no = e3, (oo = t3) ? (io = Array.isArray(oo.ignoreImageAnalysis) ? oo.ignoreImageAnalysis : [], uo = Array.isArray(oo.ignoreInlineStyle) ? oo.ignoreInlineStyle : []) : (io = [], uo = []), ao = r2, document.head) {
      if (Io())
        return;
      document.documentElement.setAttribute("data-darkreader-mode", "dynamic"), document.documentElement.setAttribute("data-darkreader-scheme", no.mode ? "dark" : "dimmed"), Ao();
    } else {
      if (!f) {
        var n2 = so("darkreader--fallback");
        document.documentElement.appendChild(n2), n2.textContent = nr(no, { strict: true });
      }
      var o2 = new MutationObserver(function() {
        if (document.head) {
          if (o2.disconnect(), Io())
            return void jo();
          Ao();
        }
      });
      o2.observe(document, { childList: true, subtree: true });
    }
  }
  function No() {
    document.dispatchEvent(new CustomEvent("__darkreader__cleanUp")), B(document.head.querySelector(".darkreader--proxy"));
  }
  function jo() {
    document.documentElement.removeAttribute("data-darkreader-mode"), document.documentElement.removeAttribute("data-darkreader-scheme"), Bo(), B(document.querySelector(".darkreader--fallback")), document.head && (pn(), B(document.head.querySelector(".darkreader--user-agent")), B(document.head.querySelector(".darkreader--text")), B(document.head.querySelector(".darkreader--invert")), B(document.head.querySelector(".darkreader--inline")), B(document.head.querySelector(".darkreader--override")), B(document.head.querySelector(".darkreader--variables")), B(document.head.querySelector(".darkreader--root-vars")), B(document.head.querySelector('meta[name="darkreader"]')), No()), vo.forEach(function(e3) {
      B(e3.querySelector(".darkreader--inline")), B(e3.querySelector(".darkreader--override"));
    }), vo.clear(), L(to.keys(), function(e3) {
      return wo(e3);
    }), _o.clear(), Tn(), L(document.querySelectorAll(".darkreader"), B), ro.forEach(function(e3) {
      e3.destroy();
    }), ro.splice(0);
  }
  function Bo() {
    wr.clear(), ae.clear(), Mo(), xo(), Oo(), br();
  }
  var Uo = /url\(\"(blob\:.*?)\"\)/g;
  function qo(e3) {
    return r(this, void 0, void 0, function() {
      var t3, r2;
      return n(this, function(n2) {
        switch (n2.label) {
          case 0:
            return t3 = [], Ze(Uo, e3, 1).forEach(function(e4) {
              var r3 = _(e4);
              t3.push(r3);
            }), [4, Promise.all(t3)];
          case 1:
            return r2 = n2.sent(), [2, e3.replace(Uo, function() {
              return 'url("' + r2.shift() + '")';
            })];
        }
      });
    });
  }
  var Wo = '/*\n                        _______\n                       /       \\\n                      .==.    .==.\n                     ((  ))==((  ))\n                    / "=="    "=="\\\n                   /____|| || ||___\\\n       ________     ____    ________  ___    ___\n       |  ___  \\   /    \\   |  ___  \\ |  |  /  /\n       |  |  \\  \\ /  /\\  \\  |  |  \\  \\|  |_/  /\n       |  |   )  /  /__\\  \\ |  |__/  /|  ___  \\\n       |  |__/  /  ______  \\|  ____  \\|  |  \\  \\\n_______|_______/__/ ____ \\__\\__|___\\__\\__|___\\__\\____\n|  ___  \\ |  ____/ /    \\   |  ___  \\ |  ____|  ___  \\\n|  |  \\  \\|  |___ /  /\\  \\  |  |  \\  \\|  |___|  |  \\  \\\n|  |__/  /|  ____/  /__\\  \\ |  |   )  |  ____|  |__/  /\n|  ____  \\|  |__/  ______  \\|  |__/  /|  |___|  ____  \\\n|__|   \\__\\____/__/      \\__\\_______/ |______|__|   \\__\\\n                https://darkreader.org\n*/\n\n/*! Dark reader generated CSS | Licensed under MIT https://github.com/darkreader/darkreader/blob/master/LICENSE */\n';
  function Ho() {
    return r(this, void 0, void 0, function() {
      function e3(e4, r3) {
        var n2 = document.querySelector(e4);
        n2 && n2.textContent && (t3.push("/* " + r3 + " */"), t3.push(n2.textContent), t3.push(""));
      }
      var t3, r2, o2, a2, i2;
      return n(this, function(n2) {
        switch (n2.label) {
          case 0:
            return t3 = [Wo], e3(".darkreader--fallback", "Fallback Style"), e3(".darkreader--user-agent", "User-Agent Style"), e3(".darkreader--text", "Text Style"), e3(".darkreader--invert", "Invert Style"), e3(".darkreader--variables", "Variables Style"), r2 = [], document.querySelectorAll(".darkreader--sync").forEach(function(e4) {
              L(e4.sheet.cssRules, function(e5) {
                e5 && e5.cssText && r2.push(e5.cssText);
              });
            }), r2.length ? (o2 = Je(r2.join("\n")), t3.push("/* Modified CSS */"), i2 = (a2 = t3).push, [4, qo(o2)]) : [3, 2];
          case 1:
            i2.apply(a2, [n2.sent()]), t3.push(""), n2.label = 2;
          case 2:
            return e3(".darkreader--override", "Override Style"), [2, t3.join("\n")];
        }
      });
    });
  }
  var Go = false, $o = function() {
    if ("undefined" == typeof window)
      return false;
    try {
      return window.self !== window.top;
    } catch (e3) {
      return console.warn(e3), true;
    }
  }();
  function zo(e3, r2) {
    void 0 === e3 && (e3 = {}), void 0 === r2 && (r2 = null);
    var n2 = t2(t2({}, M), e3);
    if (n2.engine !== R.dynamicTheme)
      throw new Error("Theme engine is not supported.");
    Fo(n2, r2, $o), Go = true;
  }
  function Qo() {
    return Go;
  }
  function Ko() {
    jo(), Go = false;
  }
  var Xo = { themeOptions: null, fixes: null };
  function Yo() {
    matchMedia("(prefers-color-scheme: dark)").matches ? zo(Xo.themeOptions, Xo.fixes) : Ko();
  }
  function Zo(e3, t3) {
    void 0 === e3 && (e3 = {}), void 0 === t3 && (t3 = null);
    var r2 = matchMedia("(prefers-color-scheme: dark)");
    e3 ? (Xo = { themeOptions: e3, fixes: t3 }, Yo(), b ? r2.addEventListener("change", Yo) : r2.addListener(Yo)) : (b ? r2.removeEventListener("change", Yo) : r2.removeListener(Yo), Ko());
  }
  function Jo() {
    return r(this, void 0, void 0, function() {
      return n(this, function(e3) {
        switch (e3.label) {
          case 0:
            return [4, Ho()];
          case 1:
            return [2, e3.sent()];
        }
      });
    });
  }
  var ea = x;
  e2.auto = Zo, e2.disable = Ko, e2.enable = zo, e2.exportGeneratedCSS = Jo, e2.isEnabled = Qo, e2.setFetchMethod = ea, Object.defineProperty(e2, "__esModule", { value: true });
}(t.exports);
export { e as c, t as d };
