// @layui/layui-vue/es/_chunks/@ctrl/index.js
function t(t2, r2) {
  (function(t3) {
    return "string" == typeof t3 && -1 !== t3.indexOf(".") && 1 === parseFloat(t3);
  })(t2) && (t2 = "100%");
  var e2 = function(t3) {
    return "string" == typeof t3 && -1 !== t3.indexOf("%");
  }(t2);
  return t2 = 360 === r2 ? t2 : Math.min(r2, Math.max(0, parseFloat(t2))), e2 && (t2 = parseInt(String(t2 * r2), 10) / 100), Math.abs(t2 - r2) < 1e-6 ? 1 : t2 = 360 === r2 ? (t2 < 0 ? t2 % r2 + r2 : t2 % r2) / parseFloat(String(r2)) : t2 % r2 / parseFloat(String(r2));
}
function r(t2) {
  return Math.min(1, Math.max(0, t2));
}
function e(t2) {
  return t2 = parseFloat(t2), (isNaN(t2) || t2 < 0 || t2 > 1) && (t2 = 1), t2;
}
function n(t2) {
  return t2 <= 1 ? "".concat(100 * Number(t2), "%") : t2;
}
function a(t2) {
  return 1 === t2.length ? "0" + t2 : String(t2);
}
function o(r2, e2, n2) {
  r2 = t(r2, 255), e2 = t(e2, 255), n2 = t(n2, 255);
  var a2 = Math.max(r2, e2, n2), o2 = Math.min(r2, e2, n2), i2 = 0, h2 = 0, s2 = (a2 + o2) / 2;
  if (a2 === o2)
    h2 = 0, i2 = 0;
  else {
    var f2 = a2 - o2;
    switch (h2 = s2 > 0.5 ? f2 / (2 - a2 - o2) : f2 / (a2 + o2), a2) {
      case r2:
        i2 = (e2 - n2) / f2 + (e2 < n2 ? 6 : 0);
        break;
      case e2:
        i2 = (n2 - r2) / f2 + 2;
        break;
      case n2:
        i2 = (r2 - e2) / f2 + 4;
    }
    i2 /= 6;
  }
  return { h: i2, s: h2, l: s2 };
}
function i(t2, r2, e2) {
  return e2 < 0 && (e2 += 1), e2 > 1 && (e2 -= 1), e2 < 1 / 6 ? t2 + 6 * e2 * (r2 - t2) : e2 < 0.5 ? r2 : e2 < 2 / 3 ? t2 + (r2 - t2) * (2 / 3 - e2) * 6 : t2;
}
function h(r2, e2, n2) {
  r2 = t(r2, 255), e2 = t(e2, 255), n2 = t(n2, 255);
  var a2 = Math.max(r2, e2, n2), o2 = Math.min(r2, e2, n2), i2 = 0, h2 = a2, s2 = a2 - o2, f2 = 0 === a2 ? 0 : s2 / a2;
  if (a2 === o2)
    i2 = 0;
  else {
    switch (a2) {
      case r2:
        i2 = (e2 - n2) / s2 + (e2 < n2 ? 6 : 0);
        break;
      case e2:
        i2 = (n2 - r2) / s2 + 2;
        break;
      case n2:
        i2 = (r2 - e2) / s2 + 4;
    }
    i2 /= 6;
  }
  return { h: i2, s: f2, v: h2 };
}
function s(t2, r2, e2, n2) {
  var o2 = [a(Math.round(t2).toString(16)), a(Math.round(r2).toString(16)), a(Math.round(e2).toString(16))];
  return n2 && o2[0].startsWith(o2[0].charAt(1)) && o2[1].startsWith(o2[1].charAt(1)) && o2[2].startsWith(o2[2].charAt(1)) ? o2[0].charAt(0) + o2[1].charAt(0) + o2[2].charAt(0) : o2.join("");
}
function f(t2) {
  return c(t2) / 255;
}
function c(t2) {
  return parseInt(t2, 16);
}
var u = { aliceblue: "#f0f8ff", antiquewhite: "#faebd7", aqua: "#00ffff", aquamarine: "#7fffd4", azure: "#f0ffff", beige: "#f5f5dc", bisque: "#ffe4c4", black: "#000000", blanchedalmond: "#ffebcd", blue: "#0000ff", blueviolet: "#8a2be2", brown: "#a52a2a", burlywood: "#deb887", cadetblue: "#5f9ea0", chartreuse: "#7fff00", chocolate: "#d2691e", coral: "#ff7f50", cornflowerblue: "#6495ed", cornsilk: "#fff8dc", crimson: "#dc143c", cyan: "#00ffff", darkblue: "#00008b", darkcyan: "#008b8b", darkgoldenrod: "#b8860b", darkgray: "#a9a9a9", darkgreen: "#006400", darkgrey: "#a9a9a9", darkkhaki: "#bdb76b", darkmagenta: "#8b008b", darkolivegreen: "#556b2f", darkorange: "#ff8c00", darkorchid: "#9932cc", darkred: "#8b0000", darksalmon: "#e9967a", darkseagreen: "#8fbc8f", darkslateblue: "#483d8b", darkslategray: "#2f4f4f", darkslategrey: "#2f4f4f", darkturquoise: "#00ced1", darkviolet: "#9400d3", deeppink: "#ff1493", deepskyblue: "#00bfff", dimgray: "#696969", dimgrey: "#696969", dodgerblue: "#1e90ff", firebrick: "#b22222", floralwhite: "#fffaf0", forestgreen: "#228b22", fuchsia: "#ff00ff", gainsboro: "#dcdcdc", ghostwhite: "#f8f8ff", goldenrod: "#daa520", gold: "#ffd700", gray: "#808080", green: "#008000", greenyellow: "#adff2f", grey: "#808080", honeydew: "#f0fff0", hotpink: "#ff69b4", indianred: "#cd5c5c", indigo: "#4b0082", ivory: "#fffff0", khaki: "#f0e68c", lavenderblush: "#fff0f5", lavender: "#e6e6fa", lawngreen: "#7cfc00", lemonchiffon: "#fffacd", lightblue: "#add8e6", lightcoral: "#f08080", lightcyan: "#e0ffff", lightgoldenrodyellow: "#fafad2", lightgray: "#d3d3d3", lightgreen: "#90ee90", lightgrey: "#d3d3d3", lightpink: "#ffb6c1", lightsalmon: "#ffa07a", lightseagreen: "#20b2aa", lightskyblue: "#87cefa", lightslategray: "#778899", lightslategrey: "#778899", lightsteelblue: "#b0c4de", lightyellow: "#ffffe0", lime: "#00ff00", limegreen: "#32cd32", linen: "#faf0e6", magenta: "#ff00ff", maroon: "#800000", mediumaquamarine: "#66cdaa", mediumblue: "#0000cd", mediumorchid: "#ba55d3", mediumpurple: "#9370db", mediumseagreen: "#3cb371", mediumslateblue: "#7b68ee", mediumspringgreen: "#00fa9a", mediumturquoise: "#48d1cc", mediumvioletred: "#c71585", midnightblue: "#191970", mintcream: "#f5fffa", mistyrose: "#ffe4e1", moccasin: "#ffe4b5", navajowhite: "#ffdead", navy: "#000080", oldlace: "#fdf5e6", olive: "#808000", olivedrab: "#6b8e23", orange: "#ffa500", orangered: "#ff4500", orchid: "#da70d6", palegoldenrod: "#eee8aa", palegreen: "#98fb98", paleturquoise: "#afeeee", palevioletred: "#db7093", papayawhip: "#ffefd5", peachpuff: "#ffdab9", peru: "#cd853f", pink: "#ffc0cb", plum: "#dda0dd", powderblue: "#b0e0e6", purple: "#800080", rebeccapurple: "#663399", red: "#ff0000", rosybrown: "#bc8f8f", royalblue: "#4169e1", saddlebrown: "#8b4513", salmon: "#fa8072", sandybrown: "#f4a460", seagreen: "#2e8b57", seashell: "#fff5ee", sienna: "#a0522d", silver: "#c0c0c0", skyblue: "#87ceeb", slateblue: "#6a5acd", slategray: "#708090", slategrey: "#708090", snow: "#fffafa", springgreen: "#00ff7f", steelblue: "#4682b4", tan: "#d2b48c", teal: "#008080", thistle: "#d8bfd8", tomato: "#ff6347", turquoise: "#40e0d0", violet: "#ee82ee", wheat: "#f5deb3", white: "#ffffff", whitesmoke: "#f5f5f5", yellow: "#ffff00", yellowgreen: "#9acd32" };
function g(r2) {
  var a2, o2, h2, s2 = { r: 0, g: 0, b: 0 }, g2 = 1, d2 = null, l2 = null, b2 = null, m2 = false, y = false;
  return "string" == typeof r2 && (r2 = function(t2) {
    if (t2 = t2.trim().toLowerCase(), 0 === t2.length)
      return false;
    var r3 = false;
    if (u[t2])
      t2 = u[t2], r3 = true;
    else if ("transparent" === t2)
      return { r: 0, g: 0, b: 0, a: 0, format: "name" };
    var e2 = p.rgb.exec(t2);
    if (e2)
      return { r: e2[1], g: e2[2], b: e2[3] };
    if (e2 = p.rgba.exec(t2), e2)
      return { r: e2[1], g: e2[2], b: e2[3], a: e2[4] };
    if (e2 = p.hsl.exec(t2), e2)
      return { h: e2[1], s: e2[2], l: e2[3] };
    if (e2 = p.hsla.exec(t2), e2)
      return { h: e2[1], s: e2[2], l: e2[3], a: e2[4] };
    if (e2 = p.hsv.exec(t2), e2)
      return { h: e2[1], s: e2[2], v: e2[3] };
    if (e2 = p.hsva.exec(t2), e2)
      return { h: e2[1], s: e2[2], v: e2[3], a: e2[4] };
    if (e2 = p.hex8.exec(t2), e2)
      return { r: c(e2[1]), g: c(e2[2]), b: c(e2[3]), a: f(e2[4]), format: r3 ? "name" : "hex8" };
    if (e2 = p.hex6.exec(t2), e2)
      return { r: c(e2[1]), g: c(e2[2]), b: c(e2[3]), format: r3 ? "name" : "hex" };
    if (e2 = p.hex4.exec(t2), e2)
      return { r: c(e2[1] + e2[1]), g: c(e2[2] + e2[2]), b: c(e2[3] + e2[3]), a: f(e2[4] + e2[4]), format: r3 ? "name" : "hex8" };
    if (e2 = p.hex3.exec(t2), e2)
      return { r: c(e2[1] + e2[1]), g: c(e2[2] + e2[2]), b: c(e2[3] + e2[3]), format: r3 ? "name" : "hex" };
    return false;
  }(r2)), "object" == typeof r2 && (v(r2.r) && v(r2.g) && v(r2.b) ? (a2 = r2.r, o2 = r2.g, h2 = r2.b, s2 = { r: 255 * t(a2, 255), g: 255 * t(o2, 255), b: 255 * t(h2, 255) }, m2 = true, y = "%" === String(r2.r).substr(-1) ? "prgb" : "rgb") : v(r2.h) && v(r2.s) && v(r2.v) ? (d2 = n(r2.s), l2 = n(r2.v), s2 = function(r3, e2, n2) {
    r3 = 6 * t(r3, 360), e2 = t(e2, 100), n2 = t(n2, 100);
    var a3 = Math.floor(r3), o3 = r3 - a3, i2 = n2 * (1 - e2), h3 = n2 * (1 - o3 * e2), s3 = n2 * (1 - (1 - o3) * e2), f2 = a3 % 6;
    return { r: 255 * [n2, h3, i2, i2, s3, n2][f2], g: 255 * [s3, n2, n2, h3, i2, i2][f2], b: 255 * [i2, i2, s3, n2, n2, h3][f2] };
  }(r2.h, d2, l2), m2 = true, y = "hsv") : v(r2.h) && v(r2.s) && v(r2.l) && (d2 = n(r2.s), b2 = n(r2.l), s2 = function(r3, e2, n2) {
    var a3, o3, h3;
    if (r3 = t(r3, 360), e2 = t(e2, 100), n2 = t(n2, 100), 0 === e2)
      o3 = n2, h3 = n2, a3 = n2;
    else {
      var s3 = n2 < 0.5 ? n2 * (1 + e2) : n2 + e2 - n2 * e2, f2 = 2 * n2 - s3;
      a3 = i(f2, s3, r3 + 1 / 3), o3 = i(f2, s3, r3), h3 = i(f2, s3, r3 - 1 / 3);
    }
    return { r: 255 * a3, g: 255 * o3, b: 255 * h3 };
  }(r2.h, d2, b2), m2 = true, y = "hsl"), Object.prototype.hasOwnProperty.call(r2, "a") && (g2 = r2.a)), g2 = e(g2), { ok: m2, format: r2.format || y, r: Math.min(255, Math.max(s2.r, 0)), g: Math.min(255, Math.max(s2.g, 0)), b: Math.min(255, Math.max(s2.b, 0)), a: g2 };
}
var d = "(?:".concat("[-\\+]?\\d*\\.\\d+%?", ")|(?:").concat("[-\\+]?\\d+%?", ")"), l = "[\\s|\\(]+(".concat(d, ")[,|\\s]+(").concat(d, ")[,|\\s]+(").concat(d, ")\\s*\\)?"), b = "[\\s|\\(]+(".concat(d, ")[,|\\s]+(").concat(d, ")[,|\\s]+(").concat(d, ")[,|\\s]+(").concat(d, ")\\s*\\)?"), p = { CSS_UNIT: new RegExp(d), rgb: new RegExp("rgb" + l), rgba: new RegExp("rgba" + b), hsl: new RegExp("hsl" + l), hsla: new RegExp("hsla" + b), hsv: new RegExp("hsv" + l), hsva: new RegExp("hsva" + b), hex3: /^#?([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})$/, hex6: /^#?([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})$/, hex4: /^#?([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})$/, hex8: /^#?([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})$/ };
function v(t2) {
  return Boolean(p.CSS_UNIT.exec(String(t2)));
}
var m = function() {
  function n2(t2, r2) {
    var e2;
    if (void 0 === t2 && (t2 = ""), void 0 === r2 && (r2 = {}), t2 instanceof n2)
      return t2;
    "number" == typeof t2 && (t2 = function(t3) {
      return { r: t3 >> 16, g: (65280 & t3) >> 8, b: 255 & t3 };
    }(t2)), this.originalInput = t2;
    var a2 = g(t2);
    this.originalInput = t2, this.r = a2.r, this.g = a2.g, this.b = a2.b, this.a = a2.a, this.roundA = Math.round(100 * this.a) / 100, this.format = null !== (e2 = r2.format) && void 0 !== e2 ? e2 : a2.format, this.gradientType = r2.gradientType, this.r < 1 && (this.r = Math.round(this.r)), this.g < 1 && (this.g = Math.round(this.g)), this.b < 1 && (this.b = Math.round(this.b)), this.isValid = a2.ok;
  }
  return n2.prototype.isDark = function() {
    return this.getBrightness() < 128;
  }, n2.prototype.isLight = function() {
    return !this.isDark();
  }, n2.prototype.getBrightness = function() {
    var t2 = this.toRgb();
    return (299 * t2.r + 587 * t2.g + 114 * t2.b) / 1e3;
  }, n2.prototype.getLuminance = function() {
    var t2 = this.toRgb(), r2 = t2.r / 255, e2 = t2.g / 255, n3 = t2.b / 255;
    return 0.2126 * (r2 <= 0.03928 ? r2 / 12.92 : Math.pow((r2 + 0.055) / 1.055, 2.4)) + 0.7152 * (e2 <= 0.03928 ? e2 / 12.92 : Math.pow((e2 + 0.055) / 1.055, 2.4)) + 0.0722 * (n3 <= 0.03928 ? n3 / 12.92 : Math.pow((n3 + 0.055) / 1.055, 2.4));
  }, n2.prototype.getAlpha = function() {
    return this.a;
  }, n2.prototype.setAlpha = function(t2) {
    return this.a = e(t2), this.roundA = Math.round(100 * this.a) / 100, this;
  }, n2.prototype.isMonochrome = function() {
    return 0 === this.toHsl().s;
  }, n2.prototype.toHsv = function() {
    var t2 = h(this.r, this.g, this.b);
    return { h: 360 * t2.h, s: t2.s, v: t2.v, a: this.a };
  }, n2.prototype.toHsvString = function() {
    var t2 = h(this.r, this.g, this.b), r2 = Math.round(360 * t2.h), e2 = Math.round(100 * t2.s), n3 = Math.round(100 * t2.v);
    return 1 === this.a ? "hsv(".concat(r2, ", ").concat(e2, "%, ").concat(n3, "%)") : "hsva(".concat(r2, ", ").concat(e2, "%, ").concat(n3, "%, ").concat(this.roundA, ")");
  }, n2.prototype.toHsl = function() {
    var t2 = o(this.r, this.g, this.b);
    return { h: 360 * t2.h, s: t2.s, l: t2.l, a: this.a };
  }, n2.prototype.toHslString = function() {
    var t2 = o(this.r, this.g, this.b), r2 = Math.round(360 * t2.h), e2 = Math.round(100 * t2.s), n3 = Math.round(100 * t2.l);
    return 1 === this.a ? "hsl(".concat(r2, ", ").concat(e2, "%, ").concat(n3, "%)") : "hsla(".concat(r2, ", ").concat(e2, "%, ").concat(n3, "%, ").concat(this.roundA, ")");
  }, n2.prototype.toHex = function(t2) {
    return void 0 === t2 && (t2 = false), s(this.r, this.g, this.b, t2);
  }, n2.prototype.toHexString = function(t2) {
    return void 0 === t2 && (t2 = false), "#" + this.toHex(t2);
  }, n2.prototype.toHex8 = function(t2) {
    return void 0 === t2 && (t2 = false), function(t3, r2, e2, n3, o2) {
      var i2, h2 = [a(Math.round(t3).toString(16)), a(Math.round(r2).toString(16)), a(Math.round(e2).toString(16)), a((i2 = n3, Math.round(255 * parseFloat(i2)).toString(16)))];
      return o2 && h2[0].startsWith(h2[0].charAt(1)) && h2[1].startsWith(h2[1].charAt(1)) && h2[2].startsWith(h2[2].charAt(1)) && h2[3].startsWith(h2[3].charAt(1)) ? h2[0].charAt(0) + h2[1].charAt(0) + h2[2].charAt(0) + h2[3].charAt(0) : h2.join("");
    }(this.r, this.g, this.b, this.a, t2);
  }, n2.prototype.toHex8String = function(t2) {
    return void 0 === t2 && (t2 = false), "#" + this.toHex8(t2);
  }, n2.prototype.toHexShortString = function(t2) {
    return void 0 === t2 && (t2 = false), 1 === this.a ? this.toHexString(t2) : this.toHex8String(t2);
  }, n2.prototype.toRgb = function() {
    return { r: Math.round(this.r), g: Math.round(this.g), b: Math.round(this.b), a: this.a };
  }, n2.prototype.toRgbString = function() {
    var t2 = Math.round(this.r), r2 = Math.round(this.g), e2 = Math.round(this.b);
    return 1 === this.a ? "rgb(".concat(t2, ", ").concat(r2, ", ").concat(e2, ")") : "rgba(".concat(t2, ", ").concat(r2, ", ").concat(e2, ", ").concat(this.roundA, ")");
  }, n2.prototype.toPercentageRgb = function() {
    var r2 = function(r3) {
      return "".concat(Math.round(100 * t(r3, 255)), "%");
    };
    return { r: r2(this.r), g: r2(this.g), b: r2(this.b), a: this.a };
  }, n2.prototype.toPercentageRgbString = function() {
    var r2 = function(r3) {
      return Math.round(100 * t(r3, 255));
    };
    return 1 === this.a ? "rgb(".concat(r2(this.r), "%, ").concat(r2(this.g), "%, ").concat(r2(this.b), "%)") : "rgba(".concat(r2(this.r), "%, ").concat(r2(this.g), "%, ").concat(r2(this.b), "%, ").concat(this.roundA, ")");
  }, n2.prototype.toName = function() {
    if (0 === this.a)
      return "transparent";
    if (this.a < 1)
      return false;
    for (var t2 = "#" + s(this.r, this.g, this.b, false), r2 = 0, e2 = Object.entries(u); r2 < e2.length; r2++) {
      var n3 = e2[r2], a2 = n3[0];
      if (t2 === n3[1])
        return a2;
    }
    return false;
  }, n2.prototype.toString = function(t2) {
    var r2 = Boolean(t2);
    t2 = null != t2 ? t2 : this.format;
    var e2 = false, n3 = this.a < 1 && this.a >= 0;
    return r2 || !n3 || !t2.startsWith("hex") && "name" !== t2 ? ("rgb" === t2 && (e2 = this.toRgbString()), "prgb" === t2 && (e2 = this.toPercentageRgbString()), "hex" !== t2 && "hex6" !== t2 || (e2 = this.toHexString()), "hex3" === t2 && (e2 = this.toHexString(true)), "hex4" === t2 && (e2 = this.toHex8String(true)), "hex8" === t2 && (e2 = this.toHex8String()), "name" === t2 && (e2 = this.toName()), "hsl" === t2 && (e2 = this.toHslString()), "hsv" === t2 && (e2 = this.toHsvString()), e2 || this.toHexString()) : "name" === t2 && 0 === this.a ? this.toName() : this.toRgbString();
  }, n2.prototype.toNumber = function() {
    return (Math.round(this.r) << 16) + (Math.round(this.g) << 8) + Math.round(this.b);
  }, n2.prototype.clone = function() {
    return new n2(this.toString());
  }, n2.prototype.lighten = function(t2) {
    void 0 === t2 && (t2 = 10);
    var e2 = this.toHsl();
    return e2.l += t2 / 100, e2.l = r(e2.l), new n2(e2);
  }, n2.prototype.brighten = function(t2) {
    void 0 === t2 && (t2 = 10);
    var r2 = this.toRgb();
    return r2.r = Math.max(0, Math.min(255, r2.r - Math.round(-t2 / 100 * 255))), r2.g = Math.max(0, Math.min(255, r2.g - Math.round(-t2 / 100 * 255))), r2.b = Math.max(0, Math.min(255, r2.b - Math.round(-t2 / 100 * 255))), new n2(r2);
  }, n2.prototype.darken = function(t2) {
    void 0 === t2 && (t2 = 10);
    var e2 = this.toHsl();
    return e2.l -= t2 / 100, e2.l = r(e2.l), new n2(e2);
  }, n2.prototype.tint = function(t2) {
    return void 0 === t2 && (t2 = 10), this.mix("white", t2);
  }, n2.prototype.shade = function(t2) {
    return void 0 === t2 && (t2 = 10), this.mix("black", t2);
  }, n2.prototype.desaturate = function(t2) {
    void 0 === t2 && (t2 = 10);
    var e2 = this.toHsl();
    return e2.s -= t2 / 100, e2.s = r(e2.s), new n2(e2);
  }, n2.prototype.saturate = function(t2) {
    void 0 === t2 && (t2 = 10);
    var e2 = this.toHsl();
    return e2.s += t2 / 100, e2.s = r(e2.s), new n2(e2);
  }, n2.prototype.greyscale = function() {
    return this.desaturate(100);
  }, n2.prototype.spin = function(t2) {
    var r2 = this.toHsl(), e2 = (r2.h + t2) % 360;
    return r2.h = e2 < 0 ? 360 + e2 : e2, new n2(r2);
  }, n2.prototype.mix = function(t2, r2) {
    void 0 === r2 && (r2 = 50);
    var e2 = this.toRgb(), a2 = new n2(t2).toRgb(), o2 = r2 / 100;
    return new n2({ r: (a2.r - e2.r) * o2 + e2.r, g: (a2.g - e2.g) * o2 + e2.g, b: (a2.b - e2.b) * o2 + e2.b, a: (a2.a - e2.a) * o2 + e2.a });
  }, n2.prototype.analogous = function(t2, r2) {
    void 0 === t2 && (t2 = 6), void 0 === r2 && (r2 = 30);
    var e2 = this.toHsl(), a2 = 360 / r2, o2 = [this];
    for (e2.h = (e2.h - (a2 * t2 >> 1) + 720) % 360; --t2; )
      e2.h = (e2.h + a2) % 360, o2.push(new n2(e2));
    return o2;
  }, n2.prototype.complement = function() {
    var t2 = this.toHsl();
    return t2.h = (t2.h + 180) % 360, new n2(t2);
  }, n2.prototype.monochromatic = function(t2) {
    void 0 === t2 && (t2 = 6);
    for (var r2 = this.toHsv(), e2 = r2.h, a2 = r2.s, o2 = r2.v, i2 = [], h2 = 1 / t2; t2--; )
      i2.push(new n2({ h: e2, s: a2, v: o2 })), o2 = (o2 + h2) % 1;
    return i2;
  }, n2.prototype.splitcomplement = function() {
    var t2 = this.toHsl(), r2 = t2.h;
    return [this, new n2({ h: (r2 + 72) % 360, s: t2.s, l: t2.l }), new n2({ h: (r2 + 216) % 360, s: t2.s, l: t2.l })];
  }, n2.prototype.onBackground = function(t2) {
    var r2 = this.toRgb(), e2 = new n2(t2).toRgb(), a2 = r2.a + e2.a * (1 - r2.a);
    return new n2({ r: (r2.r * r2.a + e2.r * e2.a * (1 - r2.a)) / a2, g: (r2.g * r2.a + e2.g * e2.a * (1 - r2.a)) / a2, b: (r2.b * r2.a + e2.b * e2.a * (1 - r2.a)) / a2, a: a2 });
  }, n2.prototype.triad = function() {
    return this.polyad(3);
  }, n2.prototype.tetrad = function() {
    return this.polyad(4);
  }, n2.prototype.polyad = function(t2) {
    for (var r2 = this.toHsl(), e2 = r2.h, a2 = [this], o2 = 360 / t2, i2 = 1; i2 < t2; i2++)
      a2.push(new n2({ h: (e2 + i2 * o2) % 360, s: r2.s, l: r2.l }));
    return a2;
  }, n2.prototype.equals = function(t2) {
    return this.toRgbString() === new n2(t2).toRgbString();
  }, n2;
}();
export { m as T };
