// @layui/layui-vue/es/_chunks/vue-i18n/index.js
import { m as e, a as t, i as a, b as n, c as l, d as r, e as s, r as o, g as i, s as c, f as u, h as m, j as _, k as f, l as g, n as p, o as b, u as d, p as v, q as h, t as k, v as F, w as L, x as y, N as I, y as M, z as T, M as O, A as W, B as N, C as w, D as E, E as P } from "../@intlify/index.js";
import { h as $, Fragment as D, getCurrentInstance as S, inject as H, onMounted as j, onUnmounted as C, ref as R, computed as U, watch as A, isRef as V, createVNode as Y, Text as G } from "vue";
/*!
  * vue-i18n v9.1.10
  * (c) 2022 kazuya kawaguchi
  * Released under the MIT License.
  */
const x = "9.1.10";
function z(e2, ...t2) {
  return u(e2, null, void 0);
}
const B = "__INTLIFY_META__", q = e("__transrateVNode"), J = e("__datetimeParts"), Z = e("__numberParts");
e("__enableEmitter"), e("__disableEmitter");
const K = e("__setPluralRules");
e("__intlifyMeta");
const Q = e("__injectWithOption");
let X = 0;
function ee(e2) {
  return (t2, a2, n2, l2) => e2(a2, n2, S() || void 0, l2);
}
function te(e2, t2) {
  const { messages: a2, __i18n: n2 } = t2, l2 = m(a2) ? a2 : _(n2) ? {} : { [e2]: {} };
  if (_(n2) && n2.forEach(({ locale: e3, resource: t3 }) => {
    e3 ? (l2[e3] = l2[e3] || {}, ne(t3, l2[e3])) : ne(t3, l2);
  }), t2.flatJson)
    for (const e3 in l2)
      f(l2, e3) && g(l2[e3]);
  return l2;
}
const ae = (e2) => !l(e2) || _(e2);
function ne(e2, t2) {
  if (ae(e2) || ae(t2))
    throw z(20);
  for (const a2 in e2)
    f(e2, a2) && (ae(e2[a2]) || ae(t2[a2]) ? t2[a2] = e2[a2] : ne(e2[a2], t2[a2]));
}
const le = () => {
  const e2 = S();
  return e2 && e2.type[B] ? { [B]: e2.type[B] } : null;
};
function re(e2 = {}) {
  const { __root: s2 } = e2, o2 = void 0 === s2;
  let i2 = !r(e2.inheritLocale) || e2.inheritLocale;
  const c2 = R(s2 && i2 ? s2.locale.value : n(e2.locale) ? e2.locale : "en-US"), u2 = R(s2 && i2 ? s2.fallbackLocale.value : n(e2.fallbackLocale) || _(e2.fallbackLocale) || m(e2.fallbackLocale) || false === e2.fallbackLocale ? e2.fallbackLocale : c2.value), f2 = R(te(c2.value, e2)), g2 = R(m(e2.datetimeFormats) ? e2.datetimeFormats : { [c2.value]: {} }), v2 = R(m(e2.numberFormats) ? e2.numberFormats : { [c2.value]: {} });
  let $2 = s2 ? s2.missingWarn : !r(e2.missingWarn) && !p(e2.missingWarn) || e2.missingWarn, D2 = s2 ? s2.fallbackWarn : !r(e2.fallbackWarn) && !p(e2.fallbackWarn) || e2.fallbackWarn, S2 = s2 ? s2.fallbackRoot : !r(e2.fallbackRoot) || e2.fallbackRoot, H2 = !!e2.fallbackFormat, j2 = b(e2.missing) ? e2.missing : null, C2 = b(e2.missing) ? ee(e2.missing) : null, V2 = b(e2.postTranslation) ? e2.postTranslation : null, B2 = !r(e2.warnHtmlMessage) || e2.warnHtmlMessage, ae2 = !!e2.escapeParameter;
  const re2 = s2 ? s2.modifiers : m(e2.modifiers) ? e2.modifiers : {};
  let se2, oe2 = e2.pluralRules || s2 && s2.pluralRules;
  se2 = h({ version: x, locale: c2.value, fallbackLocale: u2.value, messages: f2.value, datetimeFormats: g2.value, numberFormats: v2.value, modifiers: re2, pluralRules: oe2, missing: null === C2 ? void 0 : C2, missingWarn: $2, fallbackWarn: D2, fallbackFormat: H2, unresolving: true, postTranslation: null === V2 ? void 0 : V2, warnHtmlMessage: B2, escapeParameter: ae2, __datetimeFormatters: m(se2) ? se2.__datetimeFormatters : void 0, __numberFormatters: m(se2) ? se2.__numberFormatters : void 0, __v_emitter: m(se2) ? se2.__v_emitter : void 0, __meta: { framework: "vue" } }), d(se2, c2.value, u2.value);
  const ie2 = U({ get: () => c2.value, set: (e3) => {
    c2.value = e3, se2.locale = c2.value;
  } }), ce2 = U({ get: () => u2.value, set: (e3) => {
    u2.value = e3, se2.fallbackLocale = u2.value, d(se2, c2.value, e3);
  } }), ue2 = U(() => f2.value), me2 = U(() => g2.value), _e2 = U(() => v2.value);
  function fe2(e3, t2, n2, l2, r2, o3) {
    let i3;
    if (c2.value, u2.value, f2.value, g2.value, v2.value, __INTLIFY_PROD_DEVTOOLS__)
      try {
        y(le()), i3 = e3(se2);
      } finally {
        y(null);
      }
    else
      i3 = e3(se2);
    if (a(i3) && i3 === I) {
      const [e4, a2] = t2();
      return s2 && S2 ? l2(s2) : r2(e4);
    }
    if (o3(i3))
      return i3;
    throw z(14);
  }
  function ge2(...e3) {
    return fe2((t2) => T(t2, ...e3), () => M(...e3), 0, (t2) => t2.t(...e3), (e4) => e4, (e4) => n(e4));
  }
  const pe2 = { normalize: function(e3) {
    return e3.map((e4) => n(e4) ? Y(G, null, e4, 0) : e4);
  }, interpolate: (e3) => e3, type: "vnode" };
  function be2(e3) {
    return f2.value[e3] || {};
  }
  X++, s2 && (A(s2.locale, (e3) => {
    i2 && (c2.value = e3, se2.locale = e3, d(se2, c2.value, u2.value));
  }), A(s2.fallbackLocale, (e3) => {
    i2 && (u2.value = e3, se2.fallbackLocale = e3, d(se2, c2.value, u2.value));
  }));
  const de2 = { id: X, locale: ie2, fallbackLocale: ce2, get inheritLocale() {
    return i2;
  }, set inheritLocale(e3) {
    i2 = e3, e3 && s2 && (c2.value = s2.locale.value, u2.value = s2.fallbackLocale.value, d(se2, c2.value, u2.value));
  }, get availableLocales() {
    return Object.keys(f2.value).sort();
  }, messages: ue2, datetimeFormats: me2, numberFormats: _e2, get modifiers() {
    return re2;
  }, get pluralRules() {
    return oe2 || {};
  }, get isGlobal() {
    return o2;
  }, get missingWarn() {
    return $2;
  }, set missingWarn(e3) {
    $2 = e3, se2.missingWarn = $2;
  }, get fallbackWarn() {
    return D2;
  }, set fallbackWarn(e3) {
    D2 = e3, se2.fallbackWarn = D2;
  }, get fallbackRoot() {
    return S2;
  }, set fallbackRoot(e3) {
    S2 = e3;
  }, get fallbackFormat() {
    return H2;
  }, set fallbackFormat(e3) {
    H2 = e3, se2.fallbackFormat = H2;
  }, get warnHtmlMessage() {
    return B2;
  }, set warnHtmlMessage(e3) {
    B2 = e3, se2.warnHtmlMessage = e3;
  }, get escapeParameter() {
    return ae2;
  }, set escapeParameter(e3) {
    ae2 = e3, se2.escapeParameter = e3;
  }, t: ge2, rt: function(...e3) {
    const [a2, n2, r2] = e3;
    if (r2 && !l(r2))
      throw z(15);
    return ge2(a2, n2, t({ resolvedMessage: true }, r2 || {}));
  }, d: function(...e3) {
    return fe2((t2) => N(t2, ...e3), () => W(...e3), 0, (t2) => t2.d(...e3), () => O, (e4) => n(e4));
  }, n: function(...e3) {
    return fe2((t2) => E(t2, ...e3), () => w(...e3), 0, (t2) => t2.n(...e3), () => O, (e4) => n(e4));
  }, te: function(e3, t2) {
    const a2 = be2(n(t2) ? t2 : c2.value);
    return null !== k(a2, e3);
  }, tm: function(e3) {
    const t2 = function(e4) {
      let t3 = null;
      const a2 = P(se2, u2.value, c2.value);
      for (let n2 = 0; n2 < a2.length; n2++) {
        const l2 = f2.value[a2[n2]] || {}, r2 = k(l2, e4);
        if (null != r2) {
          t3 = r2;
          break;
        }
      }
      return t3;
    }(e3);
    return null != t2 ? t2 : s2 && s2.tm(e3) || {};
  }, getLocaleMessage: be2, setLocaleMessage: function(e3, t2) {
    f2.value[e3] = t2, se2.messages = f2.value;
  }, mergeLocaleMessage: function(e3, t2) {
    f2.value[e3] = f2.value[e3] || {}, ne(t2, f2.value[e3]), se2.messages = f2.value;
  }, getDateTimeFormat: function(e3) {
    return g2.value[e3] || {};
  }, setDateTimeFormat: function(e3, t2) {
    g2.value[e3] = t2, se2.datetimeFormats = g2.value, F(se2, e3, t2);
  }, mergeDateTimeFormat: function(e3, a2) {
    g2.value[e3] = t(g2.value[e3] || {}, a2), se2.datetimeFormats = g2.value, F(se2, e3, a2);
  }, getNumberFormat: function(e3) {
    return v2.value[e3] || {};
  }, setNumberFormat: function(e3, t2) {
    v2.value[e3] = t2, se2.numberFormats = v2.value, L(se2, e3, t2);
  }, mergeNumberFormat: function(e3, a2) {
    v2.value[e3] = t(v2.value[e3] || {}, a2), se2.numberFormats = v2.value, L(se2, e3, a2);
  }, getPostTranslationHandler: function() {
    return b(V2) ? V2 : null;
  }, setPostTranslationHandler: function(e3) {
    V2 = e3, se2.postTranslation = e3;
  }, getMissingHandler: function() {
    return j2;
  }, setMissingHandler: function(e3) {
    null !== e3 && (C2 = ee(e3)), j2 = e3, se2.missing = C2;
  }, [q]: function(...e3) {
    return fe2((t2) => {
      let a2;
      const n2 = t2;
      try {
        n2.processor = pe2, a2 = T(n2, ...e3);
      } finally {
        n2.processor = null;
      }
      return a2;
    }, () => M(...e3), 0, (t2) => t2[q](...e3), (e4) => [Y(G, null, e4, 0)], (e4) => _(e4));
  }, [Z]: function(...e3) {
    return fe2((t2) => E(t2, ...e3), () => w(...e3), 0, (t2) => t2[Z](...e3), () => [], (e4) => n(e4) || _(e4));
  }, [J]: function(...e3) {
    return fe2((t2) => N(t2, ...e3), () => W(...e3), 0, (t2) => t2[J](...e3), () => [], (e4) => n(e4) || _(e4));
  }, [K]: function(e3) {
    oe2 = e3, se2.pluralRules = oe2;
  }, [Q]: e2.__injectWithOption };
  return de2;
}
function se(e2 = {}) {
  const l2 = re(function(e3) {
    const a2 = n(e3.locale) ? e3.locale : "en-US", l3 = n(e3.fallbackLocale) || _(e3.fallbackLocale) || m(e3.fallbackLocale) || false === e3.fallbackLocale ? e3.fallbackLocale : a2, s3 = b(e3.missing) ? e3.missing : void 0, o2 = !r(e3.silentTranslationWarn) && !p(e3.silentTranslationWarn) || !e3.silentTranslationWarn, i2 = !r(e3.silentFallbackWarn) && !p(e3.silentFallbackWarn) || !e3.silentFallbackWarn, c2 = !r(e3.fallbackRoot) || e3.fallbackRoot, u2 = !!e3.formatFallbackMessages, f2 = m(e3.modifiers) ? e3.modifiers : {}, g2 = e3.pluralizationRules, d2 = b(e3.postTranslation) ? e3.postTranslation : void 0, v2 = !n(e3.warnHtmlInMessage) || "off" !== e3.warnHtmlInMessage, h2 = !!e3.escapeParameterHtml, k2 = !r(e3.sync) || e3.sync;
    let F2 = e3.messages;
    if (m(e3.sharedMessages)) {
      const a3 = e3.sharedMessages;
      F2 = Object.keys(a3).reduce((e4, n2) => {
        const l4 = e4[n2] || (e4[n2] = {});
        return t(l4, a3[n2]), e4;
      }, F2 || {});
    }
    const { __i18n: L2, __root: y2, __injectWithOption: I2 } = e3, M2 = e3.datetimeFormats, T2 = e3.numberFormats;
    return { locale: a2, fallbackLocale: l3, messages: F2, flatJson: e3.flatJson, datetimeFormats: M2, numberFormats: T2, missing: s3, missingWarn: o2, fallbackWarn: i2, fallbackRoot: c2, fallbackFormat: u2, modifiers: f2, pluralRules: g2, postTranslation: d2, warnHtmlMessage: v2, escapeParameter: h2, inheritLocale: k2, __i18n: L2, __root: y2, __injectWithOption: I2 };
  }(e2)), s2 = { id: l2.id, get locale() {
    return l2.locale.value;
  }, set locale(e3) {
    l2.locale.value = e3;
  }, get fallbackLocale() {
    return l2.fallbackLocale.value;
  }, set fallbackLocale(e3) {
    l2.fallbackLocale.value = e3;
  }, get messages() {
    return l2.messages.value;
  }, get datetimeFormats() {
    return l2.datetimeFormats.value;
  }, get numberFormats() {
    return l2.numberFormats.value;
  }, get availableLocales() {
    return l2.availableLocales;
  }, get formatter() {
    return { interpolate: () => [] };
  }, set formatter(e3) {
  }, get missing() {
    return l2.getMissingHandler();
  }, set missing(e3) {
    l2.setMissingHandler(e3);
  }, get silentTranslationWarn() {
    return r(l2.missingWarn) ? !l2.missingWarn : l2.missingWarn;
  }, set silentTranslationWarn(e3) {
    l2.missingWarn = r(e3) ? !e3 : e3;
  }, get silentFallbackWarn() {
    return r(l2.fallbackWarn) ? !l2.fallbackWarn : l2.fallbackWarn;
  }, set silentFallbackWarn(e3) {
    l2.fallbackWarn = r(e3) ? !e3 : e3;
  }, get modifiers() {
    return l2.modifiers;
  }, get formatFallbackMessages() {
    return l2.fallbackFormat;
  }, set formatFallbackMessages(e3) {
    l2.fallbackFormat = e3;
  }, get postTranslation() {
    return l2.getPostTranslationHandler();
  }, set postTranslation(e3) {
    l2.setPostTranslationHandler(e3);
  }, get sync() {
    return l2.inheritLocale;
  }, set sync(e3) {
    l2.inheritLocale = e3;
  }, get warnHtmlInMessage() {
    return l2.warnHtmlMessage ? "warn" : "off";
  }, set warnHtmlInMessage(e3) {
    l2.warnHtmlMessage = "off" !== e3;
  }, get escapeParameterHtml() {
    return l2.escapeParameter;
  }, set escapeParameterHtml(e3) {
    l2.escapeParameter = e3;
  }, get preserveDirectiveContent() {
    return true;
  }, set preserveDirectiveContent(e3) {
  }, get pluralizationRules() {
    return l2.pluralRules || {};
  }, __composer: l2, t(...e3) {
    const [t2, a2, r2] = e3, s3 = {};
    let o2 = null, i2 = null;
    if (!n(t2))
      throw z(15);
    const c2 = t2;
    return n(a2) ? s3.locale = a2 : _(a2) ? o2 = a2 : m(a2) && (i2 = a2), _(r2) ? o2 = r2 : m(r2) && (i2 = r2), l2.t(c2, o2 || i2 || {}, s3);
  }, rt: (...e3) => l2.rt(...e3), tc(...e3) {
    const [t2, r2, s3] = e3, o2 = { plural: 1 };
    let i2 = null, c2 = null;
    if (!n(t2))
      throw z(15);
    const u2 = t2;
    return n(r2) ? o2.locale = r2 : a(r2) ? o2.plural = r2 : _(r2) ? i2 = r2 : m(r2) && (c2 = r2), n(s3) ? o2.locale = s3 : _(s3) ? i2 = s3 : m(s3) && (c2 = s3), l2.t(u2, i2 || c2 || {}, o2);
  }, te: (e3, t2) => l2.te(e3, t2), tm: (e3) => l2.tm(e3), getLocaleMessage: (e3) => l2.getLocaleMessage(e3), setLocaleMessage(e3, t2) {
    l2.setLocaleMessage(e3, t2);
  }, mergeLocaleMessage(e3, t2) {
    l2.mergeLocaleMessage(e3, t2);
  }, d: (...e3) => l2.d(...e3), getDateTimeFormat: (e3) => l2.getDateTimeFormat(e3), setDateTimeFormat(e3, t2) {
    l2.setDateTimeFormat(e3, t2);
  }, mergeDateTimeFormat(e3, t2) {
    l2.mergeDateTimeFormat(e3, t2);
  }, n: (...e3) => l2.n(...e3), getNumberFormat: (e3) => l2.getNumberFormat(e3), setNumberFormat(e3, t2) {
    l2.setNumberFormat(e3, t2);
  }, mergeNumberFormat(e3, t2) {
    l2.mergeNumberFormat(e3, t2);
  }, getChoiceIndex: (e3, t2) => -1, __onComponentInstanceCreated(t2) {
    const { componentInstanceCreatedListener: a2 } = e2;
    a2 && a2(t2, s2);
  } };
  return s2;
}
const oe = { tag: { type: [String, Object] }, locale: { type: String }, scope: { type: String, validator: (e2) => "parent" === e2 || "global" === e2, default: "parent" }, i18n: { type: Object } }, ie = { name: "i18n-t", props: t({ keypath: { type: String, required: true }, plural: { type: [Number, String], validator: (e2) => a(e2) || !isNaN(e2) } }, oe), setup(e2, a2) {
  const { slots: r2, attrs: s2 } = a2, o2 = e2.i18n || de({ useScope: e2.scope, __useComponent: true }), i2 = Object.keys(r2).filter((e3) => "_" !== e3);
  return () => {
    const r3 = {};
    e2.locale && (r3.locale = e2.locale), void 0 !== e2.plural && (r3.plural = n(e2.plural) ? +e2.plural : e2.plural);
    const c2 = function({ slots: e3 }, t2) {
      return 1 === t2.length && "default" === t2[0] ? e3.default ? e3.default() : [] : t2.reduce((t3, a3) => {
        const n2 = e3[a3];
        return n2 && (t3[a3] = n2()), t3;
      }, {});
    }(a2, i2), u2 = o2[q](e2.keypath, c2, r3), m2 = t({}, s2);
    return n(e2.tag) || l(e2.tag) ? $(e2.tag, m2, u2) : $(D, m2, u2);
  };
} };
function ce(e2, a2, r2, s2) {
  const { slots: o2, attrs: i2 } = a2;
  return () => {
    const a3 = { part: true };
    let c2 = {};
    e2.locale && (a3.locale = e2.locale), n(e2.format) ? a3.key = e2.format : l(e2.format) && (n(e2.format.key) && (a3.key = e2.format.key), c2 = Object.keys(e2.format).reduce((a4, n2) => r2.includes(n2) ? t({}, a4, { [n2]: e2.format[n2] }) : a4, {}));
    const u2 = s2(e2.value, a3, c2);
    let m2 = [a3.key];
    _(u2) ? m2 = u2.map((e3, t2) => {
      const a4 = o2[e3.type];
      return a4 ? a4({ [e3.type]: e3.value, index: t2, parts: u2 }) : [e3.value];
    }) : n(u2) && (m2 = [u2]);
    const f2 = t({}, i2);
    return n(e2.tag) || l(e2.tag) ? $(e2.tag, f2, m2) : $(D, f2, m2);
  };
}
const ue = ["localeMatcher", "style", "unit", "unitDisplay", "currency", "currencyDisplay", "useGrouping", "numberingSystem", "minimumIntegerDigits", "minimumFractionDigits", "maximumFractionDigits", "minimumSignificantDigits", "maximumSignificantDigits", "notation", "formatMatcher"], me = { name: "i18n-n", props: t({ value: { type: Number, required: true }, format: { type: [String, Object] } }, oe), setup(e2, t2) {
  const a2 = e2.i18n || de({ useScope: "parent", __useComponent: true });
  return ce(e2, t2, ue, (...e3) => a2[Z](...e3));
} }, _e = ["dateStyle", "timeStyle", "fractionalSecondDigits", "calendar", "dayPeriod", "numberingSystem", "localeMatcher", "timeZone", "hour12", "hourCycle", "formatMatcher", "weekday", "era", "year", "month", "day", "hour", "minute", "second", "timeZoneName"], fe = { name: "i18n-d", props: t({ value: { type: [Number, Date], required: true }, format: { type: [String, Object] } }, oe), setup(e2, t2) {
  const a2 = e2.i18n || de({ useScope: "parent", __useComponent: true });
  return ce(e2, t2, _e, (...e3) => a2[J](...e3));
} };
function ge(e2) {
  const t2 = (t3, { instance: l2, value: r2, modifiers: s2 }) => {
    if (!l2 || !l2.$)
      throw z(22);
    const o2 = function(e3, t4) {
      const a2 = e3;
      if ("composition" === e3.mode)
        return a2.__getInstance(t4) || e3.global;
      {
        const n2 = a2.__getInstance(t4);
        return null != n2 ? n2.__composer : e3.global.__composer;
      }
    }(e2, l2.$), i2 = function(e3) {
      if (n(e3))
        return { path: e3 };
      if (m(e3)) {
        if (!("path" in e3))
          throw z(19);
        return e3;
      }
      throw z(20);
    }(r2);
    t3.textContent = o2.t(...function(e3) {
      const { path: t4, locale: l3, args: r3, choice: s3, plural: o3 } = e3, i3 = {}, c2 = r3 || {};
      n(l3) && (i3.locale = l3);
      a(s3) && (i3.plural = s3);
      a(o3) && (i3.plural = o3);
      return [t4, c2, i3];
    }(i2));
  };
  return { beforeMount: t2, beforeUpdate: t2 };
}
function pe(e2, t2) {
  e2.locale = t2.locale || e2.locale, e2.fallbackLocale = t2.fallbackLocale || e2.fallbackLocale, e2.missing = t2.missing || e2.missing, e2.silentTranslationWarn = t2.silentTranslationWarn || e2.silentFallbackWarn, e2.silentFallbackWarn = t2.silentFallbackWarn || e2.silentFallbackWarn, e2.formatFallbackMessages = t2.formatFallbackMessages || e2.formatFallbackMessages, e2.postTranslation = t2.postTranslation || e2.postTranslation, e2.warnHtmlInMessage = t2.warnHtmlInMessage || e2.warnHtmlInMessage, e2.escapeParameterHtml = t2.escapeParameterHtml || e2.escapeParameterHtml, e2.sync = t2.sync || e2.sync, e2.__composer[K](t2.pluralizationRules || e2.pluralizationRules);
  const a2 = te(e2.locale, { messages: t2.messages, __i18n: t2.__i18n });
  return Object.keys(a2).forEach((t3) => e2.mergeLocaleMessage(t3, a2[t3])), t2.datetimeFormats && Object.keys(t2.datetimeFormats).forEach((a3) => e2.mergeDateTimeFormat(a3, t2.datetimeFormats[a3])), t2.numberFormats && Object.keys(t2.numberFormats).forEach((a3) => e2.mergeNumberFormat(a3, t2.numberFormats[a3])), e2;
}
function be(t2 = {}) {
  const a2 = __VUE_I18N_LEGACY_API__ && r(t2.legacy) ? t2.legacy : __VUE_I18N_LEGACY_API__, n2 = !!t2.globalInjection, l2 = /* @__PURE__ */ new Map(), s2 = __VUE_I18N_LEGACY_API__ && a2 ? se(t2) : re(t2), o2 = e(""), i2 = { get mode() {
    return __VUE_I18N_LEGACY_API__ && a2 ? "legacy" : "composition";
  }, async install(e2, ...t3) {
    e2.__VUE_I18N_SYMBOL__ = o2, e2.provide(e2.__VUE_I18N_SYMBOL__, i2), !a2 && n2 && function(e3, t4) {
      const a3 = /* @__PURE__ */ Object.create(null);
      ve.forEach((e4) => {
        const n3 = Object.getOwnPropertyDescriptor(t4, e4);
        if (!n3)
          throw z(22);
        const l3 = V(n3.value) ? { get: () => n3.value.value, set(e5) {
          n3.value.value = e5;
        } } : { get: () => n3.get && n3.get() };
        Object.defineProperty(a3, e4, l3);
      }), e3.config.globalProperties.$i18n = a3, he.forEach((a4) => {
        const n3 = Object.getOwnPropertyDescriptor(t4, a4);
        if (!n3 || !n3.value)
          throw z(22);
        Object.defineProperty(e3.config.globalProperties, `$${a4}`, n3);
      });
    }(e2, i2.global), __VUE_I18N_FULL_INSTALL__ && function(e3, t4, ...a3) {
      const n3 = m(a3[0]) ? a3[0] : {}, l3 = !!n3.useI18nComponentName;
      (!r(n3.globalInstall) || n3.globalInstall) && (e3.component(l3 ? "i18n" : ie.name, ie), e3.component(me.name, me), e3.component(fe.name, fe)), e3.directive("t", ge(t4));
    }(e2, i2, ...t3), __VUE_I18N_LEGACY_API__ && a2 && e2.mixin(function(e3, t4, a3) {
      return { beforeCreate() {
        const n3 = S();
        if (!n3)
          throw z(22);
        const l3 = this.$options;
        if (l3.i18n) {
          const a4 = l3.i18n;
          l3.__i18n && (a4.__i18n = l3.__i18n), a4.__root = t4, this === this.$root ? this.$i18n = pe(e3, a4) : (a4.__injectWithOption = true, this.$i18n = se(a4));
        } else
          l3.__i18n ? this === this.$root ? this.$i18n = pe(e3, l3) : this.$i18n = se({ __i18n: l3.__i18n, __injectWithOption: true, __root: t4 }) : this.$i18n = e3;
        e3.__onComponentInstanceCreated(this.$i18n), a3.__setInstance(n3, this.$i18n), this.$t = (...e4) => this.$i18n.t(...e4), this.$rt = (...e4) => this.$i18n.rt(...e4), this.$tc = (...e4) => this.$i18n.tc(...e4), this.$te = (e4, t5) => this.$i18n.te(e4, t5), this.$d = (...e4) => this.$i18n.d(...e4), this.$n = (...e4) => this.$i18n.n(...e4), this.$tm = (e4) => this.$i18n.tm(e4);
      }, mounted() {
      }, beforeUnmount() {
        const e4 = S();
        if (!e4)
          throw z(22);
        delete this.$t, delete this.$rt, delete this.$tc, delete this.$te, delete this.$d, delete this.$n, delete this.$tm, a3.__deleteInstance(e4), delete this.$i18n;
      } };
    }(s2, s2.__composer, i2));
  }, get global() {
    return s2;
  }, __instances: l2, __getInstance: (e2) => l2.get(e2) || null, __setInstance(e2, t3) {
    l2.set(e2, t3);
  }, __deleteInstance(e2) {
    l2.delete(e2);
  } };
  return i2;
}
function de(e2 = {}) {
  const a2 = S();
  if (null == a2)
    throw z(16);
  if (!a2.appContext.app.__VUE_I18N_SYMBOL__)
    throw z(17);
  const n2 = H(a2.appContext.app.__VUE_I18N_SYMBOL__);
  if (!n2)
    throw z(22);
  const r2 = "composition" === n2.mode ? n2.global : n2.global.__composer, o2 = s(e2) ? "__i18n" in a2.type ? "local" : "global" : e2.useScope ? e2.useScope : "local";
  if ("global" === o2) {
    let t2 = l(e2.messages) ? e2.messages : {};
    "__i18nGlobal" in a2.type && (t2 = te(r2.locale.value, { messages: t2, __i18n: a2.type.__i18nGlobal }));
    const n3 = Object.keys(t2);
    if (n3.length && n3.forEach((e3) => {
      r2.mergeLocaleMessage(e3, t2[e3]);
    }), l(e2.datetimeFormats)) {
      const t3 = Object.keys(e2.datetimeFormats);
      t3.length && t3.forEach((t4) => {
        r2.mergeDateTimeFormat(t4, e2.datetimeFormats[t4]);
      });
    }
    if (l(e2.numberFormats)) {
      const t3 = Object.keys(e2.numberFormats);
      t3.length && t3.forEach((t4) => {
        r2.mergeNumberFormat(t4, e2.numberFormats[t4]);
      });
    }
    return r2;
  }
  if ("parent" === o2) {
    let t2 = function(e3, t3, a3 = false) {
      let n3 = null;
      const l2 = t3.root;
      let r3 = t3.parent;
      for (; null != r3; ) {
        const t4 = e3;
        if ("composition" === e3.mode)
          n3 = t4.__getInstance(r3);
        else {
          const e4 = t4.__getInstance(r3);
          null != e4 && (n3 = e4.__composer), a3 && n3 && !n3[Q] && (n3 = null);
        }
        if (null != n3)
          break;
        if (l2 === r3)
          break;
        r3 = r3.parent;
      }
      return n3;
    }(n2, a2, e2.__useComponent);
    return null == t2 && (t2 = r2), t2;
  }
  if ("legacy" === n2.mode)
    throw z(18);
  const i2 = n2;
  let c2 = i2.__getInstance(a2);
  if (null == c2) {
    const n3 = a2.type, l2 = t({}, e2);
    n3.__i18n && (l2.__i18n = n3.__i18n), r2 && (l2.__root = r2), c2 = re(l2), function(e3, t2, a3) {
      j(() => {
      }, t2), C(() => {
        e3.__deleteInstance(t2);
      }, t2);
    }(i2, a2), i2.__setInstance(a2, c2);
  }
  return c2;
}
const ve = ["locale", "fallbackLocale", "availableLocales"], he = ["t", "rt", "d", "n", "tm"];
if (o(v), "boolean" != typeof __VUE_I18N_FULL_INSTALL__ && (i().__VUE_I18N_FULL_INSTALL__ = true), "boolean" != typeof __VUE_I18N_LEGACY_API__ && (i().__VUE_I18N_LEGACY_API__ = true), "boolean" != typeof __INTLIFY_PROD_DEVTOOLS__ && (i().__INTLIFY_PROD_DEVTOOLS__ = false), __INTLIFY_PROD_DEVTOOLS__) {
  const e2 = i();
  e2.__INTLIFY__ = true, c(e2.__INTLIFY_DEVTOOLS_GLOBAL_HOOK__);
}
export { be as c, de as u };
