// @layui/layui-vue/es/index.js
import { i as e, c as o } from "./upload/index2.js";
export { c as LayUpload, l as layer } from "./upload/index2.js";
import { u as r, i as a, c as t } from "./datePicker/index2.js";
export { c as LayDatePicker, u as useI18n } from "./datePicker/index2.js";
import i from "./backTop/index.js";
export { default as LayBacktop } from "./backTop/index.js";
import s from "./avatar/index.js";
export { default as LayAvatar } from "./avatar/index.js";
import n from "./avatarList/index.js";
export { default as LayAvatarList } from "./avatarList/index.js";
import { c as d } from "./radio/index2.js";
export { c as LayRadio } from "./radio/index2.js";
import m from "./radioGroup/index.js";
export { default as LayRadioGroup } from "./radioGroup/index.js";
import { c as p } from "./button/index2.js";
export { c as LayButton } from "./button/index2.js";
import x from "./buttonContainer/index.js";
export { default as LayButtonContainer } from "./buttonContainer/index.js";
import f from "./buttonGroup/index.js";
export { default as LayButtonGroup } from "./buttonGroup/index.js";
import y from "./icon/index.js";
export { default as LayIcon } from "./icon/index.js";
import j from "./iconPicker/index.js";
export { default as LayIconPicker } from "./iconPicker/index.js";
import L from "./card/index.js";
export { default as LayCard } from "./card/index.js";
import b from "./checkcard/index.js";
export { default as LayCheckcard } from "./checkcard/index.js";
import h from "./checkcardGroup/index.js";
export { default as LayCheckcardGroup } from "./checkcardGroup/index.js";
import g from "./layout/index.js";
export { default as LayLayout } from "./layout/index.js";
import k from "./side/index.js";
export { default as LaySide } from "./side/index.js";
import I from "./body/index.js";
export { default as LayBody } from "./body/index.js";
import { c as S } from "./header/index2.js";
export { c as LayHeader } from "./header/index2.js";
import { c as C } from "./footer/index2.js";
export { c as LayFooter } from "./footer/index2.js";
import P from "./logo/index.js";
export { default as LayLogo } from "./logo/index.js";
import w from "./panel/index.js";
export { default as LayPanel } from "./panel/index.js";
import T from "./progress/index.js";
export { default as LayProgress } from "./progress/index.js";
import B from "./col/index.js";
export { default as LayCol } from "./col/index.js";
import v from "./row/index.js";
export { default as LayRow } from "./row/index.js";
import { c as M } from "./input/index2.js";
export { c as LayInput } from "./input/index2.js";
import G from "./badge/index.js";
export { default as LayBadge } from "./badge/index.js";
import R from "./quote/index.js";
export { default as LayQuote } from "./quote/index.js";
import _ from "./line/index.js";
export { default as LayLine } from "./line/index.js";
import F from "./timeline/index.js";
export { default as LayTimeline } from "./timeline/index.js";
import O from "./timelineItem/index.js";
export { default as LayTimelineItem } from "./timelineItem/index.js";
import D from "./textarea/index.js";
export { default as LayTextarea } from "./textarea/index.js";
import A from "./switch/index.js";
export { default as LaySwitch } from "./switch/index.js";
import N from "./collapse/index.js";
export { default as LayCollapse } from "./collapse/index.js";
import E from "./collapseItem/index.js";
export { default as LayCollapseItem } from "./collapseItem/index.js";
import H from "./container/index.js";
export { default as LayContainer } from "./container/index.js";
import U from "./countUp/index.js";
export { default as LayCountUp } from "./countUp/index.js";
import q from "./menu/index.js";
export { default as LayMenu } from "./menu/index.js";
import Q from "./menuItem/index.js";
export { default as LayMenuItem } from "./menuItem/index.js";
import { c as V } from "./checkbox/index2.js";
export { c as LayCheckbox } from "./checkbox/index2.js";
import z from "./checkboxGroup/index.js";
export { default as LayCheckboxGroup } from "./checkboxGroup/index.js";
import $ from "./form/index.js";
export { default as LayForm } from "./form/index.js";
import J from "./breadcrumb/index.js";
export { default as LayBreadcrumb } from "./breadcrumb/index.js";
import K from "./breadcrumbItem/index.js";
export { default as LayBreadcrumbItem } from "./breadcrumbItem/index.js";
import W from "./field/index.js";
export { default as LayField } from "./field/index.js";
import X from "./select/index.js";
export { default as LaySelect } from "./select/index.js";
import { c as Y } from "./selectOption/index2.js";
export { c as LaySelectOption } from "./selectOption/index2.js";
import { c as Z } from "./selectOptionGroup/index2.js";
export { c as LaySelectOptionGroup } from "./selectOptionGroup/index2.js";
import { c as ee } from "./scroll/index2.js";
export { c as LayScroll } from "./scroll/index2.js";
import { c as oe } from "./empty/index2.js";
export { c as LayEmpty } from "./empty/index2.js";
import re from "./formItem/index.js";
export { default as LayFormItem } from "./formItem/index.js";
import ae from "./rate/index.js";
export { default as LayRate } from "./rate/index.js";
import { c as te } from "./dropdown/index2.js";
export { c as LayDropdown } from "./dropdown/index2.js";
import { c as ie } from "./dropdownMenu/index2.js";
export { c as LayDropdownMenu } from "./dropdownMenu/index2.js";
import { c as se } from "./dropdownMenuItem/index2.js";
export { c as LayDropdownMenuItem } from "./dropdownMenuItem/index2.js";
import ne from "./dropdownSubMenu/index.js";
export { default as LayDropdownSubMenu } from "./dropdownSubMenu/index.js";
import de from "./tab/index.js";
export { default as LayTab } from "./tab/index.js";
import { c as me } from "./tabItem/index2.js";
export { c as LayTabItem } from "./tabItem/index2.js";
import { c as pe } from "./tree/index2.js";
export { c as LayTree } from "./tree/index2.js";
import le from "./treeSelect/index.js";
export { default as LayTreeSelect } from "./treeSelect/index.js";
import xe from "./table/index.js";
export { default as LayTable } from "./table/index.js";
import { c as fe } from "./page/index2.js";
export { c as LayPage } from "./page/index2.js";
import ce from "./transfer/index.js";
export { default as LayTransfer } from "./transfer/index.js";
import ue from "./slider/index.js";
export { default as LaySlider } from "./slider/index.js";
import { c as ye } from "./carousel/index2.js";
export { c as LayCarousel } from "./carousel/index2.js";
import je from "./carouselItem/index.js";
export { default as LayCarouselItem } from "./carouselItem/index.js";
import Le from "./colorPicker/index.js";
export { default as LayColorPicker } from "./colorPicker/index.js";
import { c as be } from "./tooltip/index2.js";
export { c as LayTooltip } from "./tooltip/index2.js";
import he from "./inputNumber/index.js";
export { default as LayInputNumber } from "./inputNumber/index.js";
import ge from "./skeleton/index.js";
export { default as LaySkeleton } from "./skeleton/index.js";
import { c as ke } from "./skeletonItem/index2.js";
export { c as LaySkeletonItem } from "./skeletonItem/index2.js";
import Ie from "./step/index.js";
export { default as LayStep } from "./step/index.js";
import Se from "./stepItem/index.js";
export { default as LayStepItem } from "./stepItem/index.js";
import Ce from "./subMenu/index.js";
export { default as LaySubMenu } from "./subMenu/index.js";
import Pe from "./splitPanel/index.js";
export { default as LaySplitPanel } from "./splitPanel/index.js";
import we from "./splitPanelItem/index.js";
export { default as LaySplitPanelItem } from "./splitPanelItem/index.js";
import Te from "./exception/index.js";
export { default as LayException } from "./exception/index.js";
import Be from "./result/index.js";
export { default as LayResult } from "./result/index.js";
import ve from "./fullscreen/index.js";
export { default as LayFullscreen } from "./fullscreen/index.js";
import { c as Me } from "./transition/index2.js";
export { c as LayTransition } from "./transition/index2.js";
import Ge from "./ripple/index.js";
export { default as LayRipple } from "./ripple/index.js";
import Re from "./noticeBar/index.js";
export { default as LayNoticeBar } from "./noticeBar/index.js";
import _e from "./pageHeader/index.js";
export { default as LayPageHeader } from "./pageHeader/index.js";
import Fe from "./cascader/index.js";
export { default as LayCascader } from "./cascader/index.js";
import Oe from "./affix/index.js";
export { default as LayAffix } from "./affix/index.js";
import De from "./space/index.js";
export { default as LaySpace } from "./space/index.js";
import { c as Ae } from "./tag/index2.js";
export { c as LayTag } from "./tag/index2.js";
import { c as Ne } from "./tagInput/index2.js";
export { c as LayTagInput } from "./tagInput/index2.js";
import Ee from "./qrcode/index.js";
export { default as LayQrcode } from "./qrcode/index.js";
import He from "./barcode/index.js";
export { default as LayBarcode } from "./barcode/index.js";
import Ue from "./autoComplete/index.js";
export { default as LayAutocomplete } from "./autoComplete/index.js";
import qe from "./radioButton/index.js";
export { default as LayRadioButton } from "./radioButton/index.js";
import Qe from "./popconfirm/index.js";
export { default as LayPopconfirm } from "./popconfirm/index.js";
import { w as Ve } from "./utils/index.js";
import { defineComponent as ze, watch as $e, renderSlot as Je } from "vue";
import { d as Ke } from "./_chunks/@umijs/index.js";
import "./_chunks/@vueuse/index.js";
import "./_chunks/cropperjs/index.js";
import "./_chunks/dayjs/index.js";
import "./_chunks/vue-i18n/index.js";
import "./_chunks/@intlify/index.js";
import "./useLevel/index.js";
import "./utils/index2.js";
import "./_chunks/@ctrl/index.js";
import "./_chunks/async-validator/index.js";
import "./_chunks/evtd/index.js";
import "./_chunks/qrcode/index.js";
import "./_chunks/encode-utf8/index.js";
import "./_chunks/dijkstrajs/index.js";
import "./_chunks/jsbarcode/index.js";
const We = Ve(ze({ name: "lay-config-provider", props: { locale: { default: "zh_CN" }, locales: null, theme: { default: "light" }, themeVariable: null, darkPartial: null }, setup(e2) {
  const o2 = e2, { locale: a2, setLocaleMessage: t2, mergeLocaleMessage: i2 } = r(), s2 = [".layui-colorpicker-trigger-span", "div.layui-color-picker *"];
  return $e(() => o2.locale, (e3) => {
    ((e4) => {
      a2.value = e4;
    })(e3);
  }, { immediate: true }), $e(() => o2.locales, (e3) => {
    e3 == null ? void 0 : e3.forEach((e4) => {
      var o3, r2;
      o3 = e4.name, r2 = e4.locale, e4.merge ? i2(o3, r2) : t2(o3, r2);
    });
  }, { immediate: true, deep: true }), $e(() => o2.theme, (e3) => {
    ((e4) => {
      const r2 = { mode: 1, brightness: 100, contrast: 90, sepia: 0, grayscale: 0 }, a3 = { css: "\n  .lay-progress-circle svg {\n    filter: invert(98%) hue-rotate(180deg) contrast(80%) !important;\n  }\n  .layui-tab-card > .layui-tab-head > .layui-tab-title.is-top .layui-this::after, \n  .layui-tab-title .layui-this::after {\n    border-bottom-color: rgb(34, 36, 37);\n  }\n  .layui-tab-card > .layui-tab-head > .layui-tab-title.is-bottom .layui-this::after,\n  .layui-tab-card > .layui-tab-head > .layui-tab-title.is-top .layui-this::after{\n    border-left-color: #3c4042;\n    border-right-color: #3c4042;\n  }\n  .layui-tab-card > .layui-tab-head > .layui-tab-title.is-bottom .layui-this::after {\n    border-top-color: rgb(34, 36, 37);\n  }\n  .layui-tab-card > .layui-tab-head > .layui-tab-title.is-left .layui-this::after,\n  .layui-tab-card > .layui-tab-head > .layui-tab-title.is-right .layui-this::after {\n    border-top-color: #3c4042;\n    border-bottom-color: #3c4042;\n  }\n  .layui-tab-title.is-left .layui-this::after {\n    border-right-color: rgb(34, 36, 37);\n    border-top-color: #3c4042;\n    border-bottom-color: #3c4042;\n    border-left-color: #3c4042;\n  }\n  .layui-tab-title.is-right .layui-this::after {\n    border-left-color: rgb(34, 36, 37);\n    border-top-color: #3c4042;\n    border-bottom-color: #3c4042;\n    border-right-color: #3c4042;\n  }\n", invert: [".lay-progress-circle svg", ".layui-layer-loading .layui-layer-loading1", ".layui-layer-loading .layui-layer-loading2", ".layui-layer-ico", ".layui-layer-min", ".layer-icon-ico16"], ignoreInlineStyle: s2, disableStyleSheetsProxy: false, ignoreImageAnalysis: ["*"] };
      Object.assign(r2, o2.darkPartial), "dark" === e4 ? (window && Ke.exports.setFetchMethod(window.fetch), Ke.exports.enable(r2, a3)) : "light" === e4 ? Ke.exports.disable() : "auto" === e4 && Ke.exports.auto(r2, a3);
    })(e3);
  }, { immediate: true }), $e(() => o2.themeVariable, (e3) => {
    ((e4) => {
      if (null != e4) {
        const o3 = Object.keys(e4);
        for (let r2 = 0; r2 < o3.length; r2++) {
          const a3 = o3[r2], t3 = e4[a3];
          document.documentElement.style.setProperty(a3, t3);
        }
      }
    })(e3);
  }, { immediate: true, deep: true }), (e3, o3) => Je(e3.$slots, "default");
} })), Xe = { LaySplitPanel: Pe, LaySplitPanelItem: we, LayRadio: d, LayRadioGroup: m, LayButton: p, LayIcon: y, LayBacktop: i, LayLayout: g, LaySide: k, LayHeader: S, LayBody: I, LayFooter: C, LayLogo: P, LayPanel: w, LayCard: L, LayCheckcard: b, LayCheckcardGroup: h, LayProgress: T, LayButtonGroup: f, LayButtonContainer: x, LayRow: v, LayCol: B, LayInput: M, LayBadge: G, LayQuote: R, LayLine: _, LayTimeline: F, LayTimelineItem: O, LayTextarea: D, LaySwitch: A, LayCollapse: N, LayCollapseItem: E, LayContainer: H, LayMenu: q, LayMenuItem: Q, LayCheckbox: V, LayForm: $, LayBreadcrumb: J, LayBreadcrumbItem: K, LayAvatar: s, LayAvatarList: n, LayField: W, LaySelect: X, LayScroll: ee, LaySelectOption: Y, LaySelectOptionGroup: Z, LayEmpty: oe, LayFormItem: re, LayRate: ae, LayDropdown: te, LayDropdownMenu: ie, LayDropdownMenuItem: se, LayDropdownSubMenu: ne, LayTab: de, LayTabItem: me, LayIconPicker: j, LayTree: pe, LayTable: xe, LayPage: fe, LayTransfer: ce, LayCheckboxGroup: z, LaySlider: ue, LayCarousel: ye, LayCarouselItem: je, LayColorPicker: Le, LayTooltip: be, LayInputNumber: he, LaySkeleton: ge, LaySkeletonItem: ke, LayCountUp: U, LayStep: Ie, LayStepItem: Se, LaySubMenu: Ce, LayException: Te, LayResult: Be, LayFullscreen: ve, LayConfigProvider: We, LayDatePicker: t, LayTransition: Me, LayUpload: o, LayRipple: Ge, LayNoticeBar: Re, LayPageHeader: _e, LayCascader: Fe, LayAffix: Oe, LaySpace: De, LayTag: Ae, LayTagInput: Ne, LayTreeSelect: le, LayQrcode: Ee, LayBarcode: He, LayAutocomplete: Ue, LayRadioButton: qe, LayPopconfirm: Qe }, Ye = (o2, r2) => {
  for (const e2 of Object.keys(Xe))
    o2.use(Xe[e2], r2);
  o2.use(a), o2.use(e, r2);
};
var Ze = { install: Ye };
export { We as LayConfigProvider, Ze as default, Ye as install };
