// @layui/layui-vue/es/slider/index.js
import { m as e, h as l, t as a, w as t } from "../utils/index.js";
import { defineComponent as i, ref as s, openBlock as n, createElementBlock as o, withModifiers as u, normalizeClass as d, createVNode as v, withCtx as r, createElementVNode as c, normalizeStyle as m, Fragment as f, renderList as p, unref as b, withDirectives as y, vShow as h, toRef as w } from "vue";
import { _ as k } from "../tooltip/index2.js";
import { o as x, a as g } from "../_chunks/evtd/index.js";
import "../_chunks/@vueuse/index.js";
const D = ["onMousedown"], M = c("div", { class: "layui-slider-line-v" }, null, -1), V = ["onClick"], B = i({ name: "StandardVue", props: { val: { default: 0 }, disabled: { type: Boolean, default: false }, step: { default: 0 }, min: { default: 0 }, max: { default: 100 }, showDots: { type: Boolean, default: false } }, emits: ["link-val-hook"], setup(t2, { emit: i2 }) {
  const w2 = t2, B2 = a(function(e2) {
    if (L2.value = false, !j2.value)
      return;
    let l2 = j2.value.getBoundingClientRect(), a2 = l2.left, t3 = e2.clientX - a2;
    if (t3 < w2.min)
      R2.value = w2.min;
    else {
      !function(e3, l3) {
        if (void 0 === e3)
          return false;
        if ("number" == typeof l3.value) {
          let a3 = e3 - l3.value;
          if (Math.abs(a3) < w2.step)
            return false;
          0 === w2.step && (l3.value = Math.floor(e3)), a3 < 0 && 0 !== w2.step ? l3.value -= w2.step : l3.value += w2.step;
        }
      }(t3 / l2.width * 100, R2), R2.value > w2.max && (R2.value = w2.max);
    }
    i2("link-val-hook", R2.value);
  });
  function _2() {
    g("selectstart", document, l), g("mouseup", window, _2), g("mousemove", window, B2), L2.value = true;
  }
  function C2() {
    x("selectstart", window, l, { once: true }), x("mouseup", window, _2), x("mousemove", window, B2);
  }
  const j2 = s(null);
  let R2 = s(w2.val);
  const L2 = s(true);
  const A2 = e(w2);
  return (e2, l2) => (n(), o("div", { ref_key: "tracker", ref: j2, onMousedown: u(C2, ["stop"]), class: d(["layui-slider-track-v", [t2.disabled ? "layui-slider-disabled" : ""]]) }, [v(k, { content: "" + t2.val, "is-can-hide": L2.value }, { default: r(() => [c("div", { style: m({ left: t2.val + "%" }), class: d(["layui-slider-btn-v", [t2.disabled ? "layui-slider-disabled disable-btn" : ""]]) }, null, 6)]), _: 1 }, 8, ["content", "is-can-hide"]), c("div", { style: m({ width: t2.val + "%" }), class: d(["layui-slider-rate-v", [t2.disabled ? "layui-slider-disabled disable-line" : ""]]) }, null, 6), M, (n(true), o(f, null, p(b(A2), (e3, l3) => y((n(), o("div", { onClick: (l4) => {
    i2("link-val-hook", e3);
  }, class: "layui-slider-dots", key: l3, style: m({ left: e3 + "%" }) }, null, 12, V)), [[h, t2.showDots]])), 128))], 42, D));
} }), _ = ["onMousedown"], C = c("div", { class: "layui-slider-line-v" }, null, -1), j = ["onClick"], R = i({ name: "StandardRange", props: { rangeValue: null, disabled: { type: Boolean, default: false }, step: { default: 0 }, min: { default: 0 }, max: { default: 100 }, showDots: { type: Boolean, default: true } }, emits: ["link-val-hook"], setup(l2, { emit: t2 }) {
  const i2 = l2;
  let D2 = w(i2, "rangeValue");
  const M2 = a(function(e2) {
    if (!A2.value)
      return;
    let l3 = A2.value.getBoundingClientRect(), a2 = l3.left, s2 = e2.clientX - a2;
    if (s2 < i2.min)
      D2.value[0] = i2.min;
    else {
      let e3 = s2 / l3.width * 100, a3 = -1;
      -1 === V2 ? (V2 = S2(Math.floor(e3), D2), a3 = V2) : a3 = V2, function(e4, l4, a4 = -1) {
        if (void 0 === e4)
          return false;
        if ("object" == typeof l4.value) {
          let t3 = e4 - l4.value[a4];
          if (Math.abs(t3) < i2.step)
            return false;
          0 === i2.step && (l4.value[a4] = Math.floor(e4)), Array.isArray(l4.value) && (t3 < 0 && 0 !== i2.step ? l4.value[a4] -= i2.step : l4.value[a4] += i2.step, function(e5) {
            if (e5.value[0] > e5.value[1]) {
              let l5 = e5.value[0];
              e5.value[0] = e5.value[1], e5.value[1] = l5, V2 = 0 === V2 ? 1 : 0;
            }
          }(l4));
        }
      }(e3, D2, a3), D2.value[1] > i2.max && (D2.value[1] = i2.max), D2.value[0] < i2.min && (D2.value[0] = i2.min);
    }
    t2("link-val-hook", D2.value);
  });
  let V2 = -1;
  function B2() {
    V2 = -1, H2.value = false, x("selectstart", window, L2, { once: true }), x("mouseup", window, R2), x("mousemove", window, M2);
  }
  function R2() {
    H2.value = true, g("selectstart", document, L2), g("mouseup", window, R2), g("mousemove", window, M2);
  }
  function L2(e2) {
    e2.preventDefault();
  }
  const A2 = s(null), H2 = s(true);
  function S2(e2, l3) {
    return Math.abs(e2 - l3.value[0]) > Math.abs(e2 - l3.value[1]) ? 1 : 0;
  }
  const X2 = e(i2);
  return (e2, a2) => (n(), o("div", { ref_key: "tracker", ref: A2, onMousedown: u(B2, ["stop"]), class: d(["layui-slider-srange", [l2.disabled ? "layui-slider-disabled" : ""]]) }, [v(k, { content: "" + b(D2)[0], "is-can-hide": H2.value }, { default: r(() => [c("div", { style: m({ left: b(D2)[0] + "%" }), class: d(["layui-slider-btn-v", [i2.disabled ? "layui-slider-disabled disable-btn" : ""]]) }, null, 6)]), _: 1 }, 8, ["content", "is-can-hide"]), v(k, { content: "" + b(D2)[1], "is-can-hide": H2.value }, { default: r(() => [c("div", { style: m({ left: b(D2)[1] + "%" }), class: d(["layui-slider-btn-v", [i2.disabled ? "layui-slider-disabled disable-btn" : ""]]) }, null, 6)]), _: 1 }, 8, ["content", "is-can-hide"]), C, c("div", { style: m({ width: b(D2)[1] - b(D2)[0] + "%", left: b(D2)[0] + "%" }), class: d(["layui-slider-rate-v", [i2.disabled ? "layui-slider-disabled disable-line" : ""]]) }, null, 6), (n(true), o(f, null, p(b(X2), (e3, a3) => y((n(), o("div", { onClick: (l3) => ((e4) => {
    let l4 = S2(e4, D2);
    D2.value[l4] = e4, t2("link-val-hook", D2.value);
  })(e3), class: "layui-slider-dots", key: a3, style: m({ left: e3 + "%" }) }, null, 12, j)), [[h, l2.showDots]])), 128))], 42, _));
} }), L = { class: "layui-slider-vertical" }, A = ["onMousedown"], H = c("div", { class: "layui-slider-vertical-line" }, null, -1), S = ["onClick"], X = i({ name: "Vertical", props: { val: { default: 0 }, disabled: { type: Boolean, default: true }, step: { default: 0 }, min: { default: 0 }, max: { default: 100 }, showDots: { type: Boolean, default: false } }, emits: ["link-val-hook"], setup(e2, { emit: l2 }) {
  const t2 = e2, i2 = a(function(e3) {
    if (_2.value = false, !V2.value)
      return;
    let a2 = V2.value.getBoundingClientRect(), i3 = a2.bottom, s2 = -1 * (e3.clientY - i3);
    if (s2 < t2.min)
      B2.value = t2.min;
    else {
      !function(e4, l3) {
        if (void 0 === e4)
          return false;
        if ("number" == typeof l3.value) {
          let a3 = e4 - l3.value;
          if (Math.abs(a3) < t2.step)
            return false;
          0 === t2.step && (l3.value = Math.floor(e4)), a3 < 0 && 0 !== t2.step ? l3.value -= t2.step : l3.value += t2.step;
        }
      }(s2 / a2.height * 100, B2), B2.value > t2.max && (B2.value = t2.max);
    }
    l2("link-val-hook", B2.value);
  });
  function w2() {
    g("selectstart", document, D2), g("mouseup", window, w2), g("mousemove", window, i2), _2.value = true;
  }
  function D2(e3) {
    e3.preventDefault();
  }
  function M2() {
    x("selectstart", window, D2, { once: true }), x("mouseup", window, w2), x("mousemove", window, i2);
  }
  const V2 = s(null);
  let B2 = s(t2.val);
  const _2 = s(true);
  const C2 = (() => {
    if (0 === t2.step)
      return [];
    let e3 = 0, l3 = [], a2 = Math.floor(100 / t2.step) - 1;
    for (let i3 = 0; i3 < a2; i3++)
      e3 += t2.step, l3.push(e3);
    return l3;
  })();
  return (a2, i3) => (n(), o("div", L, [c("div", { onMousedown: u(M2, ["stop"]), ref_key: "tracker", ref: V2, class: d([[e2.disabled ? "layui-slider-disabled" : ""], "layui-slider-vertical-track"]) }, [v(k, { content: "" + e2.val, "is-can-hide": _2.value }, { default: r(() => [c("div", { style: m({ bottom: e2.val + "%" }), class: d([[t2.disabled ? "layui-slider-disabled disable-btn" : ""], "layui-slider-vertical-btn"]) }, null, 6)]), _: 1 }, 8, ["content", "is-can-hide"]), c("div", { style: m({ height: e2.val + "%" }), class: d([[t2.disabled ? "layui-slider-disabled disable-line" : ""], "layui-slider-vertical-rate"]) }, null, 6), H, (n(true), o(f, null, p(b(C2), (a3, t3) => y((n(), o("div", { onClick: (e3) => {
    l2("link-val-hook", a3);
  }, class: "layui-slider-vertical-dots", key: t3, style: m({ bottom: a3 + "%" }) }, null, 12, S)), [[h, e2.showDots]])), 128))], 42, A)]));
} }), Y = { class: "layui-slider-vertical" }, q = ["onMousedown"], z = c("div", { class: "layui-slider-vertical-line" }, null, -1), E = ["onClick"], F = i({ name: "VerticalRange", props: { rangeValue: null, disabled: { type: Boolean, default: false }, step: { default: 0 }, min: { default: 0 }, max: { default: 100 }, showDots: { type: Boolean, default: false } }, emits: ["link-val-hook"], setup(l2, { emit: t2 }) {
  const i2 = l2;
  let D2 = w(i2, "rangeValue");
  const M2 = a(function(e2) {
    if (!j2.value)
      return;
    let l3 = j2.value.getBoundingClientRect(), a2 = l3.bottom, s2 = -1 * (e2.clientY - a2);
    if (s2 < i2.min)
      D2.value[0] = i2.min;
    else {
      let e3 = s2 / l3.height * 100, a3 = -1;
      -1 === V2 ? (V2 = L2(Math.floor(e3), D2), a3 = V2) : a3 = V2, function(e4, l4, a4 = -1) {
        if (void 0 === e4)
          return false;
        if ("object" == typeof l4.value) {
          let t3 = e4 - l4.value[a4];
          if (Math.abs(t3) < i2.step)
            return false;
          0 === i2.step && (l4.value[a4] = Math.floor(e4)), Array.isArray(l4.value) && (t3 < 0 && 0 !== i2.step ? l4.value[a4] -= i2.step : l4.value[a4] += i2.step, function(e5) {
            if (e5.value[0] > e5.value[1]) {
              let l5 = e5.value[0];
              e5.value[0] = e5.value[1], e5.value[1] = l5, V2 = 0 === V2 ? 1 : 0;
            }
          }(l4));
        }
      }(e3, D2, a3), D2.value[1] > i2.max && (D2.value[1] = i2.max), D2.value[0] < i2.min && (D2.value[0] = i2.min);
    }
    t2("link-val-hook", D2.value);
  });
  let V2 = -1;
  function B2() {
    V2 = -1, R2.value = false, x("selectstart", window, C2, { once: true }), x("mouseup", window, _2), x("mousemove", window, M2);
  }
  function _2() {
    R2.value = true, g("selectstart", document, C2), g("mouseup", window, _2), g("mousemove", window, M2);
  }
  function C2(e2) {
    e2.preventDefault();
  }
  const j2 = s(null), R2 = s(true);
  function L2(e2, l3) {
    return Math.abs(e2 - l3.value[0]) > Math.abs(e2 - l3.value[1]) ? 1 : 0;
  }
  const A2 = e(i2);
  return (e2, a2) => (n(), o("div", Y, [c("div", { ref_key: "tracker", ref: j2, onMousedown: u(B2, ["stop"]), class: d(["layui-slider-vrange", [l2.disabled ? "layui-slider-disabled" : ""]]) }, [v(k, { content: "" + b(D2)[1], "is-can-hide": R2.value }, { default: r(() => [c("div", { style: m({ bottom: b(D2)[1] + "%" }), class: d(["layui-slider-vertical-btn", [i2.disabled ? "layui-slider-disabled disable-btn" : ""]]) }, null, 6)]), _: 1 }, 8, ["content", "is-can-hide"]), v(k, { content: "" + b(D2)[0], "is-can-hide": R2.value }, { default: r(() => [c("div", { style: m({ bottom: b(D2)[0] + "%" }), class: d(["layui-slider-vertical-btn", [i2.disabled ? "layui-slider-disabled disable-btn" : ""]]) }, null, 6)]), _: 1 }, 8, ["content", "is-can-hide"]), z, c("div", { style: m({ height: b(D2)[1] - b(D2)[0] + "%", bottom: b(D2)[0] + "%" }), class: d(["layui-slider-vertical-rate", [i2.disabled ? "layui-slider-disabled disable-line" : ""]]) }, null, 6), (n(true), o(f, null, p(b(A2), (e3, a3) => y((n(), o("div", { onClick: (l3) => ((e4) => {
    let l4 = L2(e4, D2);
    D2.value[l4] = e4, t2("link-val-hook", D2.value);
  })(e3), class: "layui-slider-vertical-dots", key: a3, style: m({ bottom: e3 + "%" }) }, null, 12, E)), [[h, l2.showDots]])), 128))], 42, q)]));
} }), G = { key: 0 }, I = { key: 0 }, J = { key: 1 }, K = { key: 1 }, N = { key: 0 }, O = { key: 1 }, P = t(i({ name: "LaySlider", props: { vertical: { type: Boolean, default: false }, modelValue: { default: 0 }, min: { default: 0 }, max: { default: 100 }, step: { default: 0 }, disabled: { type: Boolean, default: false }, range: { type: Boolean }, rangeValue: null, showDots: { type: Boolean, default: false } }, emits: ["update:modelValue"], setup(e2, { emit: l2 }) {
  let a2 = w(e2, "rangeValue");
  function t2(e3) {
    l2("update:modelValue", e3);
  }
  return (l3, i2) => (n(), o("div", null, [e2.vertical ? (n(), o("div", G, [e2.range ? (n(), o("div", I, [v(F, { step: e2.step, onLinkValHook: t2, disabled: e2.disabled, rangeValue: b(a2), min: e2.min, max: e2.max, showDots: e2.showDots }, null, 8, ["step", "disabled", "rangeValue", "min", "max", "showDots"])])) : (n(), o("div", J, [v(X, { step: e2.step, onLinkValHook: t2, disabled: e2.disabled, val: e2.modelValue, min: e2.min, max: e2.max, showDots: e2.showDots }, null, 8, ["step", "disabled", "val", "min", "max", "showDots"])]))])) : (n(), o("div", K, [e2.range ? (n(), o("div", N, [v(R, { step: e2.step, onLinkValHook: t2, disabled: e2.disabled, rangeValue: b(a2), min: e2.min, max: e2.max, showDots: e2.showDots }, null, 8, ["step", "disabled", "rangeValue", "min", "max", "showDots"])])) : (n(), o("div", O, [v(B, { val: e2.modelValue, onLinkValHook: t2, disabled: e2.disabled, step: e2.step, min: e2.min, max: e2.max, showDots: e2.showDots }, null, 8, ["val", "disabled", "step", "min", "max", "showDots"])]))]))]));
} }));
export { P as default };
