// @layui/layui-vue/es/_chunks/@intlify/index.js
/*!
  * @intlify/shared v9.1.10
  * (c) 2022 kazuya kawaguchi
  * Released under the MIT License.
  */
const e = "function" == typeof Symbol && "symbol" == typeof Symbol.toStringTag, t = (t2) => e ? Symbol(t2) : t2, n = (e2, t2, n2) => r({ l: e2, k: t2, s: n2 }), r = (e2) => JSON.stringify(e2).replace(/\u2028/g, "\\u2028").replace(/\u2029/g, "\\u2029").replace(/\u0027/g, "\\u0027"), o = (e2) => "number" == typeof e2 && isFinite(e2), s = (e2) => "[object Date]" === L(e2), c = (e2) => "[object RegExp]" === L(e2), a = (e2) => x(e2) && 0 === Object.keys(e2).length;
function u(e2, t2) {
  "undefined" != typeof console && (console.warn("[intlify] " + e2), t2 && console.warn(t2.stack));
}
const l = Object.assign;
let i;
const f = () => i || (i = "undefined" != typeof globalThis ? globalThis : "undefined" != typeof self ? self : "undefined" != typeof window ? window : "undefined" != typeof global ? global : {});
function p(e2) {
  return e2.replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&apos;");
}
const m = Object.prototype.hasOwnProperty;
function d(e2, t2) {
  return m.call(e2, t2);
}
const k = Array.isArray, h = (e2) => "function" == typeof e2, b = (e2) => "string" == typeof e2, g = (e2) => "boolean" == typeof e2, y = (e2) => null !== e2 && "object" == typeof e2, _ = Object.prototype.toString, L = (e2) => _.call(e2), x = (e2) => "[object Object]" === L(e2), T = Object.prototype.hasOwnProperty;
function w(e2, t2) {
  return T.call(e2, t2);
}
const v = (e2) => null !== e2 && "object" == typeof e2, C = [];
C[0] = { w: [0], i: [3, 0], "[": [4], o: [7] }, C[1] = { w: [1], ".": [2], "[": [4], o: [7] }, C[2] = { w: [2], i: [3, 0], 0: [3, 0] }, C[3] = { i: [3, 0], 0: [3, 0], w: [1, 1], ".": [2, 1], "[": [4, 1], o: [7, 1] }, C[4] = { "'": [5, 0], '"': [6, 0], "[": [4, 2], "]": [1, 3], o: 8, l: [4, 0] }, C[5] = { "'": [4, 0], o: 8, l: [5, 0] }, C[6] = { '"': [4, 0], o: 8, l: [6, 0] };
const P = /^\s?(?:true|false|-?[\d.]+|'[^']*'|"[^"]*")\s?$/;
function O(e2) {
  if (null == e2)
    return "o";
  switch (e2.charCodeAt(0)) {
    case 91:
    case 93:
    case 46:
    case 34:
    case 39:
      return e2;
    case 95:
    case 36:
    case 45:
      return "i";
    case 9:
    case 10:
    case 13:
    case 160:
    case 65279:
    case 8232:
    case 8233:
      return "w";
  }
  return "i";
}
function S(e2) {
  const t2 = e2.trim();
  return ("0" !== e2.charAt(0) || !isNaN(parseInt(e2))) && (n2 = t2, P.test(n2) ? function(e3) {
    const t3 = e3.charCodeAt(0);
    return t3 !== e3.charCodeAt(e3.length - 1) || 34 !== t3 && 39 !== t3 ? e3 : e3.slice(1, -1);
  }(t2) : "*" + t2);
  var n2;
}
const F = /* @__PURE__ */ new Map();
function $(e2, t2) {
  if (!v(e2))
    return null;
  let n2 = F.get(t2);
  if (n2 || (n2 = function(e3) {
    const t3 = [];
    let n3, r3, o3, s3, c2, a2, u2, l2 = -1, i2 = 0, f2 = 0;
    const p2 = [];
    function m2() {
      const t4 = e3[l2 + 1];
      if (5 === i2 && "'" === t4 || 6 === i2 && '"' === t4)
        return l2++, o3 = "\\" + t4, p2[0](), true;
    }
    for (p2[0] = () => {
      void 0 === r3 ? r3 = o3 : r3 += o3;
    }, p2[1] = () => {
      void 0 !== r3 && (t3.push(r3), r3 = void 0);
    }, p2[2] = () => {
      p2[0](), f2++;
    }, p2[3] = () => {
      if (f2 > 0)
        f2--, i2 = 4, p2[0]();
      else {
        if (f2 = 0, void 0 === r3)
          return false;
        if (r3 = S(r3), false === r3)
          return false;
        p2[1]();
      }
    }; null !== i2; )
      if (l2++, n3 = e3[l2], "\\" !== n3 || !m2()) {
        if (s3 = O(n3), u2 = C[i2], c2 = u2[s3] || u2.l || 8, 8 === c2)
          return;
        if (i2 = c2[0], void 0 !== c2[1] && (a2 = p2[c2[1]], a2 && (o3 = n3, false === a2())))
          return;
        if (7 === i2)
          return t3;
      }
  }(t2), n2 && F.set(t2, n2)), !n2)
    return null;
  const r2 = n2.length;
  let o2 = e2, s2 = 0;
  for (; s2 < r2; ) {
    const e3 = o2[n2[s2]];
    if (void 0 === e3)
      return null;
    o2 = e3, s2++;
  }
  return o2;
}
function N(e2) {
  if (!v(e2))
    return e2;
  for (const t2 in e2)
    if (w(e2, t2))
      if (t2.includes(".")) {
        const n2 = t2.split("."), r2 = n2.length - 1;
        let o2 = e2;
        for (let e3 = 0; e3 < r2; e3++)
          n2[e3] in o2 || (o2[n2[e3]] = {}), o2 = o2[n2[e3]];
        o2[n2[r2]] = e2[t2], delete e2[t2], v(o2[n2[r2]]) && N(o2[n2[r2]]);
      } else
        v(e2[t2]) && N(e2[t2]);
  return e2;
}
/*!
  * @intlify/runtime v9.1.10
  * (c) 2022 kazuya kawaguchi
  * Released under the MIT License.
  */
const I = (e2) => e2, W = (e2) => "", j = "text", A = (e2) => 0 === e2.length ? "" : e2.join(""), E = (e2) => null == e2 ? "" : k(e2) || x(e2) && e2.toString === _ ? JSON.stringify(e2, null, 2) : String(e2);
function M(e2, t2) {
  return e2 = Math.abs(e2), 2 === t2 ? e2 ? e2 > 1 ? 1 : 0 : 1 : e2 ? Math.min(e2, 2) : 0;
}
function D(e2 = {}) {
  const t2 = e2.locale, n2 = function(e3) {
    const t3 = o(e3.pluralIndex) ? e3.pluralIndex : -1;
    return e3.named && (o(e3.named.count) || o(e3.named.n)) ? o(e3.named.count) ? e3.named.count : o(e3.named.n) ? e3.named.n : t3 : t3;
  }(e2), r2 = y(e2.pluralRules) && b(t2) && h(e2.pluralRules[t2]) ? e2.pluralRules[t2] : M, s2 = y(e2.pluralRules) && b(t2) && h(e2.pluralRules[t2]) ? M : void 0, c2 = e2.list || [], a2 = e2.named || {};
  o(e2.pluralIndex) && function(e3, t3) {
    t3.count || (t3.count = e3), t3.n || (t3.n = e3);
  }(n2, a2);
  function u2(t3) {
    const n3 = h(e2.messages) ? e2.messages(t3) : !!y(e2.messages) && e2.messages[t3];
    return n3 || (e2.parent ? e2.parent.message(t3) : W);
  }
  const l2 = x(e2.processor) && h(e2.processor.normalize) ? e2.processor.normalize : A, i2 = x(e2.processor) && h(e2.processor.interpolate) ? e2.processor.interpolate : E, f2 = { list: (e3) => c2[e3], named: (e3) => a2[e3], plural: (e3) => e3[r2(n2, e3.length, s2)], linked: (t3, n3) => {
    const r3 = u2(t3)(f2);
    return b(n3) ? (o2 = n3, e2.modifiers ? e2.modifiers[o2] : I)(r3) : r3;
    var o2;
  }, message: u2, type: x(e2.processor) && b(e2.processor.type) ? e2.processor.type : j, interpolate: i2, normalize: l2 };
  return f2;
}
/*!
  * @intlify/message-compiler v9.1.10
  * (c) 2022 kazuya kawaguchi
  * Released under the MIT License.
  */
function R(e2, t2, n2 = {}) {
  const { domain: r2, messages: o2, args: s2 } = n2, c2 = new SyntaxError(String(e2));
  return c2.code = e2, t2 && (c2.location = t2), c2.domain = r2, c2;
}
function z(e2) {
  throw e2;
}
function J(e2, t2, n2) {
  const r2 = { start: e2, end: t2 };
  return null != n2 && (r2.source = n2), r2;
}
const H = " ", U = "\r", V = "\n", Y = String.fromCharCode(8232), q = String.fromCharCode(8233);
function K(e2) {
  const t2 = e2;
  let n2 = 0, r2 = 1, o2 = 1, s2 = 0;
  const c2 = (e3) => t2[e3] === U && t2[e3 + 1] === V, a2 = (e3) => t2[e3] === q, u2 = (e3) => t2[e3] === Y, l2 = (e3) => c2(e3) || ((e4) => t2[e4] === V)(e3) || a2(e3) || u2(e3), i2 = (e3) => c2(e3) || a2(e3) || u2(e3) ? V : t2[e3];
  function f2() {
    return s2 = 0, l2(n2) && (r2++, o2 = 0), c2(n2) && n2++, n2++, o2++, t2[n2];
  }
  return { index: () => n2, line: () => r2, column: () => o2, peekOffset: () => s2, charAt: i2, currentChar: () => i2(n2), currentPeek: () => i2(n2 + s2), next: f2, peek: function() {
    return c2(n2 + s2) && s2++, s2++, t2[n2 + s2];
  }, reset: function() {
    n2 = 0, r2 = 1, o2 = 1, s2 = 0;
  }, resetPeek: function(e3 = 0) {
    s2 = e3;
  }, skipToPeek: function() {
    const e3 = n2 + s2;
    for (; e3 !== n2; )
      f2();
    s2 = 0;
  } };
}
const B = void 0, G = "'", Q = "tokenizer";
function X(e2, t2 = {}) {
  const n2 = false !== t2.location, r2 = K(e2), o2 = () => r2.index(), s2 = () => {
    return e3 = r2.line(), t3 = r2.column(), n3 = r2.index(), { line: e3, column: t3, offset: n3 };
    var e3, t3, n3;
  }, c2 = s2(), a2 = o2(), u2 = { currentType: 14, offset: a2, startLoc: c2, endLoc: c2, lastType: 14, lastOffset: a2, lastStartLoc: c2, lastEndLoc: c2, braceNest: 0, inLinked: false, text: "" }, l2 = () => u2, { onError: i2 } = t2;
  function f2(e3, t3, n3, ...r3) {
    const o3 = l2();
    if (t3.column += n3, t3.offset += n3, i2) {
      const n4 = R(e3, J(o3.startLoc, t3), { domain: Q, args: r3 });
      i2(n4);
    }
  }
  function p2(e3, t3, r3) {
    e3.endLoc = s2(), e3.currentType = t3;
    const o3 = { type: t3 };
    return n2 && (o3.loc = J(e3.startLoc, e3.endLoc)), null != r3 && (o3.value = r3), o3;
  }
  const m2 = (e3) => p2(e3, 14);
  function d2(e3, t3) {
    return e3.currentChar() === t3 ? (e3.next(), t3) : (f2(0, s2(), 0, t3), "");
  }
  function k2(e3) {
    let t3 = "";
    for (; e3.currentPeek() === H || e3.currentPeek() === V; )
      t3 += e3.currentPeek(), e3.peek();
    return t3;
  }
  function h2(e3) {
    const t3 = k2(e3);
    return e3.skipToPeek(), t3;
  }
  function b2(e3) {
    if (e3 === B)
      return false;
    const t3 = e3.charCodeAt(0);
    return t3 >= 97 && t3 <= 122 || t3 >= 65 && t3 <= 90 || 95 === t3;
  }
  function g2(e3, t3) {
    const { currentType: n3 } = t3;
    if (2 !== n3)
      return false;
    k2(e3);
    const r3 = function(e4) {
      if (e4 === B)
        return false;
      const t4 = e4.charCodeAt(0);
      return t4 >= 48 && t4 <= 57;
    }("-" === e3.currentPeek() ? e3.peek() : e3.currentPeek());
    return e3.resetPeek(), r3;
  }
  function y2(e3) {
    k2(e3);
    const t3 = "|" === e3.currentPeek();
    return e3.resetPeek(), t3;
  }
  function _2(e3, t3 = true) {
    const n3 = (t4 = false, r4 = "", o3 = false) => {
      const s3 = e3.currentPeek();
      return "{" === s3 ? "%" !== r4 && t4 : "@" !== s3 && s3 ? "%" === s3 ? (e3.peek(), n3(t4, "%", true)) : "|" === s3 ? !("%" !== r4 && !o3) || !(r4 === H || r4 === V) : s3 === H ? (e3.peek(), n3(true, H, o3)) : s3 !== V || (e3.peek(), n3(true, V, o3)) : "%" === r4 || t4;
    }, r3 = n3();
    return t3 && e3.resetPeek(), r3;
  }
  function L2(e3, t3) {
    const n3 = e3.currentChar();
    return n3 === B ? B : t3(n3) ? (e3.next(), n3) : null;
  }
  function x2(e3) {
    return L2(e3, (e4) => {
      const t3 = e4.charCodeAt(0);
      return t3 >= 97 && t3 <= 122 || t3 >= 65 && t3 <= 90 || t3 >= 48 && t3 <= 57 || 95 === t3 || 36 === t3;
    });
  }
  function T2(e3) {
    return L2(e3, (e4) => {
      const t3 = e4.charCodeAt(0);
      return t3 >= 48 && t3 <= 57;
    });
  }
  function w2(e3) {
    return L2(e3, (e4) => {
      const t3 = e4.charCodeAt(0);
      return t3 >= 48 && t3 <= 57 || t3 >= 65 && t3 <= 70 || t3 >= 97 && t3 <= 102;
    });
  }
  function v2(e3) {
    let t3 = "", n3 = "";
    for (; t3 = T2(e3); )
      n3 += t3;
    return n3;
  }
  function C2(e3) {
    const t3 = e3.currentChar();
    switch (t3) {
      case "\\":
      case "'":
        return e3.next(), `\\${t3}`;
      case "u":
        return P2(e3, t3, 4);
      case "U":
        return P2(e3, t3, 6);
      default:
        return f2(3, s2(), 0, t3), "";
    }
  }
  function P2(e3, t3, n3) {
    d2(e3, t3);
    let r3 = "";
    for (let o3 = 0; o3 < n3; o3++) {
      const n4 = w2(e3);
      if (!n4) {
        f2(4, s2(), 0, `\\${t3}${r3}${e3.currentChar()}`);
        break;
      }
      r3 += n4;
    }
    return `\\${t3}${r3}`;
  }
  function O2(e3) {
    h2(e3);
    const t3 = d2(e3, "|");
    return h2(e3), t3;
  }
  function S2(e3, t3) {
    let n3 = null;
    switch (e3.currentChar()) {
      case "{":
        return t3.braceNest >= 1 && f2(8, s2(), 0), e3.next(), n3 = p2(t3, 2, "{"), h2(e3), t3.braceNest++, n3;
      case "}":
        return t3.braceNest > 0 && 2 === t3.currentType && f2(7, s2(), 0), e3.next(), n3 = p2(t3, 3, "}"), t3.braceNest--, t3.braceNest > 0 && h2(e3), t3.inLinked && 0 === t3.braceNest && (t3.inLinked = false), n3;
      case "@":
        return t3.braceNest > 0 && f2(6, s2(), 0), n3 = F2(e3, t3) || m2(t3), t3.braceNest = 0, n3;
      default:
        let r3 = true, o3 = true, c3 = true;
        if (y2(e3))
          return t3.braceNest > 0 && f2(6, s2(), 0), n3 = p2(t3, 1, O2(e3)), t3.braceNest = 0, t3.inLinked = false, n3;
        if (t3.braceNest > 0 && (5 === t3.currentType || 6 === t3.currentType || 7 === t3.currentType))
          return f2(6, s2(), 0), t3.braceNest = 0, $2(e3, t3);
        if (r3 = function(e4, t4) {
          const { currentType: n4 } = t4;
          if (2 !== n4)
            return false;
          k2(e4);
          const r4 = b2(e4.currentPeek());
          return e4.resetPeek(), r4;
        }(e3, t3))
          return n3 = p2(t3, 5, function(e4) {
            h2(e4);
            let t4 = "", n4 = "";
            for (; t4 = x2(e4); )
              n4 += t4;
            return e4.currentChar() === B && f2(6, s2(), 0), n4;
          }(e3)), h2(e3), n3;
        if (o3 = g2(e3, t3))
          return n3 = p2(t3, 6, function(e4) {
            h2(e4);
            let t4 = "";
            return "-" === e4.currentChar() ? (e4.next(), t4 += `-${v2(e4)}`) : t4 += v2(e4), e4.currentChar() === B && f2(6, s2(), 0), t4;
          }(e3)), h2(e3), n3;
        if (c3 = function(e4, t4) {
          const { currentType: n4 } = t4;
          if (2 !== n4)
            return false;
          k2(e4);
          const r4 = e4.currentPeek() === G;
          return e4.resetPeek(), r4;
        }(e3, t3))
          return n3 = p2(t3, 7, function(e4) {
            h2(e4), d2(e4, "'");
            let t4 = "", n4 = "";
            const r4 = (e5) => e5 !== G && e5 !== V;
            for (; t4 = L2(e4, r4); )
              n4 += "\\" === t4 ? C2(e4) : t4;
            const o4 = e4.currentChar();
            return o4 === V || o4 === B ? (f2(2, s2(), 0), o4 === V && (e4.next(), d2(e4, "'")), n4) : (d2(e4, "'"), n4);
          }(e3)), h2(e3), n3;
        if (!r3 && !o3 && !c3)
          return n3 = p2(t3, 13, function(e4) {
            h2(e4);
            let t4 = "", n4 = "";
            const r4 = (e5) => "{" !== e5 && "}" !== e5 && e5 !== H && e5 !== V;
            for (; t4 = L2(e4, r4); )
              n4 += t4;
            return n4;
          }(e3)), f2(1, s2(), 0, n3.value), h2(e3), n3;
    }
    return n3;
  }
  function F2(e3, t3) {
    const { currentType: n3 } = t3;
    let r3 = null;
    const o3 = e3.currentChar();
    switch (8 !== n3 && 9 !== n3 && 12 !== n3 && 10 !== n3 || o3 !== V && o3 !== H || f2(9, s2(), 0), o3) {
      case "@":
        return e3.next(), r3 = p2(t3, 8, "@"), t3.inLinked = true, r3;
      case ".":
        return h2(e3), e3.next(), p2(t3, 9, ".");
      case ":":
        return h2(e3), e3.next(), p2(t3, 10, ":");
      default:
        return y2(e3) ? (r3 = p2(t3, 1, O2(e3)), t3.braceNest = 0, t3.inLinked = false, r3) : function(e4, t4) {
          const { currentType: n4 } = t4;
          if (8 !== n4)
            return false;
          k2(e4);
          const r4 = "." === e4.currentPeek();
          return e4.resetPeek(), r4;
        }(e3, t3) || function(e4, t4) {
          const { currentType: n4 } = t4;
          if (8 !== n4 && 12 !== n4)
            return false;
          k2(e4);
          const r4 = ":" === e4.currentPeek();
          return e4.resetPeek(), r4;
        }(e3, t3) ? (h2(e3), F2(e3, t3)) : function(e4, t4) {
          const { currentType: n4 } = t4;
          if (9 !== n4)
            return false;
          k2(e4);
          const r4 = b2(e4.currentPeek());
          return e4.resetPeek(), r4;
        }(e3, t3) ? (h2(e3), p2(t3, 12, function(e4) {
          let t4 = "", n4 = "";
          for (; t4 = x2(e4); )
            n4 += t4;
          return n4;
        }(e3))) : function(e4, t4) {
          const { currentType: n4 } = t4;
          if (10 !== n4)
            return false;
          const r4 = () => {
            const t5 = e4.currentPeek();
            return "{" === t5 ? b2(e4.peek()) : !("@" === t5 || "%" === t5 || "|" === t5 || ":" === t5 || "." === t5 || t5 === H || !t5) && (t5 === V ? (e4.peek(), r4()) : b2(t5));
          }, o4 = r4();
          return e4.resetPeek(), o4;
        }(e3, t3) ? (h2(e3), "{" === o3 ? S2(e3, t3) || r3 : p2(t3, 11, function(e4) {
          const t4 = (n4 = false, r4) => {
            const o4 = e4.currentChar();
            return "{" !== o4 && "%" !== o4 && "@" !== o4 && "|" !== o4 && o4 ? o4 === H ? r4 : o4 === V ? (r4 += o4, e4.next(), t4(n4, r4)) : (r4 += o4, e4.next(), t4(true, r4)) : r4;
          };
          return t4(false, "");
        }(e3))) : (8 === n3 && f2(9, s2(), 0), t3.braceNest = 0, t3.inLinked = false, $2(e3, t3));
    }
  }
  function $2(e3, t3) {
    let n3 = { type: 14 };
    if (t3.braceNest > 0)
      return S2(e3, t3) || m2(t3);
    if (t3.inLinked)
      return F2(e3, t3) || m2(t3);
    const r3 = e3.currentChar();
    switch (r3) {
      case "{":
        return S2(e3, t3) || m2(t3);
      case "}":
        return f2(5, s2(), 0), e3.next(), p2(t3, 3, "}");
      case "@":
        return F2(e3, t3) || m2(t3);
      default:
        if (y2(e3))
          return n3 = p2(t3, 1, O2(e3)), t3.braceNest = 0, t3.inLinked = false, n3;
        if (_2(e3))
          return p2(t3, 0, function(e4) {
            let t4 = "";
            for (; ; ) {
              const n4 = e4.currentChar();
              if ("{" === n4 || "}" === n4 || "@" === n4 || "|" === n4 || !n4)
                break;
              if ("%" === n4) {
                if (!_2(e4))
                  break;
                t4 += n4, e4.next();
              } else if (n4 === H || n4 === V)
                if (_2(e4))
                  t4 += n4, e4.next();
                else {
                  if (y2(e4))
                    break;
                  t4 += n4, e4.next();
                }
              else
                t4 += n4, e4.next();
            }
            return t4;
          }(e3));
        if ("%" === r3)
          return e3.next(), p2(t3, 4, "%");
    }
    return n3;
  }
  return { nextToken: function() {
    const { currentType: e3, offset: t3, startLoc: n3, endLoc: c3 } = u2;
    return u2.lastType = e3, u2.lastOffset = t3, u2.lastStartLoc = n3, u2.lastEndLoc = c3, u2.offset = o2(), u2.startLoc = s2(), r2.currentChar() === B ? p2(u2, 14) : $2(r2, u2);
  }, currentOffset: o2, currentPosition: s2, context: l2 };
}
const Z = "parser", ee = /(?:\\\\|\\'|\\u([0-9a-fA-F]{4})|\\U([0-9a-fA-F]{6}))/g;
function te(e2, t2, n2) {
  switch (e2) {
    case "\\\\":
      return "\\";
    case "\\'":
      return "'";
    default: {
      const e3 = parseInt(t2 || n2, 16);
      return e3 <= 55295 || e3 >= 57344 ? String.fromCodePoint(e3) : "\uFFFD";
    }
  }
}
function ne(e2 = {}) {
  const t2 = false !== e2.location, { onError: n2 } = e2;
  function r2(e3, t3, r3, o3, ...s3) {
    const c3 = e3.currentPosition();
    if (c3.offset += o3, c3.column += o3, n2) {
      const e4 = R(t3, J(r3, c3), { domain: Z, args: s3 });
      n2(e4);
    }
  }
  function o2(e3, n3, r3) {
    const o3 = { type: e3, start: n3, end: n3 };
    return t2 && (o3.loc = { start: r3, end: r3 }), o3;
  }
  function s2(e3, n3, r3, o3) {
    e3.end = n3, o3 && (e3.type = o3), t2 && e3.loc && (e3.loc.end = r3);
  }
  function c2(e3, t3) {
    const n3 = e3.context(), r3 = o2(3, n3.offset, n3.startLoc);
    return r3.value = t3, s2(r3, e3.currentOffset(), e3.currentPosition()), r3;
  }
  function a2(e3, t3) {
    const n3 = e3.context(), { lastOffset: r3, lastStartLoc: c3 } = n3, a3 = o2(5, r3, c3);
    return a3.index = parseInt(t3, 10), e3.nextToken(), s2(a3, e3.currentOffset(), e3.currentPosition()), a3;
  }
  function u2(e3, t3) {
    const n3 = e3.context(), { lastOffset: r3, lastStartLoc: c3 } = n3, a3 = o2(4, r3, c3);
    return a3.key = t3, e3.nextToken(), s2(a3, e3.currentOffset(), e3.currentPosition()), a3;
  }
  function i2(e3, t3) {
    const n3 = e3.context(), { lastOffset: r3, lastStartLoc: c3 } = n3, a3 = o2(9, r3, c3);
    return a3.value = t3.replace(ee, te), e3.nextToken(), s2(a3, e3.currentOffset(), e3.currentPosition()), a3;
  }
  function f2(e3) {
    const t3 = e3.context(), n3 = o2(6, t3.offset, t3.startLoc);
    let c3 = e3.nextToken();
    if (9 === c3.type) {
      const t4 = function(e4) {
        const t5 = e4.nextToken(), n4 = e4.context(), { lastOffset: c4, lastStartLoc: a3 } = n4, u3 = o2(8, c4, a3);
        return 12 !== t5.type ? (r2(e4, 11, n4.lastStartLoc, 0), u3.value = "", s2(u3, c4, a3), { nextConsumeToken: t5, node: u3 }) : (null == t5.value && r2(e4, 13, n4.lastStartLoc, 0, re(t5)), u3.value = t5.value || "", s2(u3, e4.currentOffset(), e4.currentPosition()), { node: u3 });
      }(e3);
      n3.modifier = t4.node, c3 = t4.nextConsumeToken || e3.nextToken();
    }
    switch (10 !== c3.type && r2(e3, 13, t3.lastStartLoc, 0, re(c3)), c3 = e3.nextToken(), 2 === c3.type && (c3 = e3.nextToken()), c3.type) {
      case 11:
        null == c3.value && r2(e3, 13, t3.lastStartLoc, 0, re(c3)), n3.key = function(e4, t4) {
          const n4 = e4.context(), r3 = o2(7, n4.offset, n4.startLoc);
          return r3.value = t4, s2(r3, e4.currentOffset(), e4.currentPosition()), r3;
        }(e3, c3.value || "");
        break;
      case 5:
        null == c3.value && r2(e3, 13, t3.lastStartLoc, 0, re(c3)), n3.key = u2(e3, c3.value || "");
        break;
      case 6:
        null == c3.value && r2(e3, 13, t3.lastStartLoc, 0, re(c3)), n3.key = a2(e3, c3.value || "");
        break;
      case 7:
        null == c3.value && r2(e3, 13, t3.lastStartLoc, 0, re(c3)), n3.key = i2(e3, c3.value || "");
        break;
      default:
        r2(e3, 12, t3.lastStartLoc, 0);
        const l2 = e3.context(), f3 = o2(7, l2.offset, l2.startLoc);
        return f3.value = "", s2(f3, l2.offset, l2.startLoc), n3.key = f3, s2(n3, l2.offset, l2.startLoc), { nextConsumeToken: c3, node: n3 };
    }
    return s2(n3, e3.currentOffset(), e3.currentPosition()), { node: n3 };
  }
  function p2(e3) {
    const t3 = e3.context(), n3 = o2(2, 1 === t3.currentType ? e3.currentOffset() : t3.offset, 1 === t3.currentType ? t3.endLoc : t3.startLoc);
    n3.items = [];
    let l2 = null;
    do {
      const o3 = l2 || e3.nextToken();
      switch (l2 = null, o3.type) {
        case 0:
          null == o3.value && r2(e3, 13, t3.lastStartLoc, 0, re(o3)), n3.items.push(c2(e3, o3.value || ""));
          break;
        case 6:
          null == o3.value && r2(e3, 13, t3.lastStartLoc, 0, re(o3)), n3.items.push(a2(e3, o3.value || ""));
          break;
        case 5:
          null == o3.value && r2(e3, 13, t3.lastStartLoc, 0, re(o3)), n3.items.push(u2(e3, o3.value || ""));
          break;
        case 7:
          null == o3.value && r2(e3, 13, t3.lastStartLoc, 0, re(o3)), n3.items.push(i2(e3, o3.value || ""));
          break;
        case 8:
          const s3 = f2(e3);
          n3.items.push(s3.node), l2 = s3.nextConsumeToken || null;
      }
    } while (14 !== t3.currentType && 1 !== t3.currentType);
    return s2(n3, 1 === t3.currentType ? t3.lastOffset : e3.currentOffset(), 1 === t3.currentType ? t3.lastEndLoc : e3.currentPosition()), n3;
  }
  function m2(e3) {
    const t3 = e3.context(), { offset: n3, startLoc: c3 } = t3, a3 = p2(e3);
    return 14 === t3.currentType ? a3 : function(e4, t4, n4, c4) {
      const a4 = e4.context();
      let u3 = 0 === c4.items.length;
      const l2 = o2(1, t4, n4);
      l2.cases = [], l2.cases.push(c4);
      do {
        const t5 = p2(e4);
        u3 || (u3 = 0 === t5.items.length), l2.cases.push(t5);
      } while (14 !== a4.currentType);
      return u3 && r2(e4, 10, n4, 0), s2(l2, e4.currentOffset(), e4.currentPosition()), l2;
    }(e3, n3, c3, a3);
  }
  return { parse: function(n3) {
    const c3 = X(n3, l({}, e2)), a3 = c3.context(), u3 = o2(0, a3.offset, a3.startLoc);
    return t2 && u3.loc && (u3.loc.source = n3), u3.body = m2(c3), 14 !== a3.currentType && r2(c3, 13, a3.lastStartLoc, 0, n3[a3.offset] || ""), s2(u3, c3.currentOffset(), c3.currentPosition()), u3;
  } };
}
function re(e2) {
  if (14 === e2.type)
    return "EOF";
  const t2 = (e2.value || "").replace(/\r?\n/gu, "\\n");
  return t2.length > 10 ? t2.slice(0, 9) + "\u2026" : t2;
}
function oe(e2, t2) {
  for (let n2 = 0; n2 < e2.length; n2++)
    se(e2[n2], t2);
}
function se(e2, t2) {
  switch (e2.type) {
    case 1:
      oe(e2.cases, t2), t2.helper("plural");
      break;
    case 2:
      oe(e2.items, t2);
      break;
    case 6:
      se(e2.key, t2), t2.helper("linked");
      break;
    case 5:
      t2.helper("interpolate"), t2.helper("list");
      break;
    case 4:
      t2.helper("interpolate"), t2.helper("named");
  }
}
function ce(e2, t2 = {}) {
  const n2 = function(e3, t3 = {}) {
    const n3 = { ast: e3, helpers: /* @__PURE__ */ new Set() };
    return { context: () => n3, helper: (e4) => (n3.helpers.add(e4), e4) };
  }(e2);
  n2.helper("normalize"), e2.body && se(e2.body, n2);
  const r2 = n2.context();
  e2.helpers = Array.from(r2.helpers);
}
function ae(e2, t2) {
  const { helper: n2 } = e2;
  switch (t2.type) {
    case 0:
      !function(e3, t3) {
        t3.body ? ae(e3, t3.body) : e3.push("null");
      }(e2, t2);
      break;
    case 1:
      !function(e3, t3) {
        const { helper: n3, needIndent: r2 } = e3;
        if (t3.cases.length > 1) {
          e3.push(`${n3("plural")}([`), e3.indent(r2());
          const o2 = t3.cases.length;
          for (let n4 = 0; n4 < o2 && (ae(e3, t3.cases[n4]), n4 !== o2 - 1); n4++)
            e3.push(", ");
          e3.deindent(r2()), e3.push("])");
        }
      }(e2, t2);
      break;
    case 2:
      !function(e3, t3) {
        const { helper: n3, needIndent: r2 } = e3;
        e3.push(`${n3("normalize")}([`), e3.indent(r2());
        const o2 = t3.items.length;
        for (let n4 = 0; n4 < o2 && (ae(e3, t3.items[n4]), n4 !== o2 - 1); n4++)
          e3.push(", ");
        e3.deindent(r2()), e3.push("])");
      }(e2, t2);
      break;
    case 6:
      !function(e3, t3) {
        const { helper: n3 } = e3;
        e3.push(`${n3("linked")}(`), ae(e3, t3.key), t3.modifier && (e3.push(", "), ae(e3, t3.modifier)), e3.push(")");
      }(e2, t2);
      break;
    case 8:
    case 7:
    case 9:
    case 3:
      e2.push(JSON.stringify(t2.value), t2);
      break;
    case 5:
      e2.push(`${n2("interpolate")}(${n2("list")}(${t2.index}))`, t2);
      break;
    case 4:
      e2.push(`${n2("interpolate")}(${n2("named")}(${JSON.stringify(t2.key)}))`, t2);
  }
}
const ue = (e2, t2 = {}) => {
  const n2 = b(t2.mode) ? t2.mode : "normal", r2 = b(t2.filename) ? t2.filename : "message.intl", o2 = !!t2.sourceMap, s2 = null != t2.breakLineCode ? t2.breakLineCode : "arrow" === n2 ? ";" : "\n", c2 = t2.needIndent ? t2.needIndent : "arrow" !== n2, a2 = e2.helpers || [], u2 = function(e3, t3) {
    const { sourceMap: n3, filename: r3, breakLineCode: o3, needIndent: s3 } = t3, c3 = { source: e3.loc.source, filename: r3, code: "", column: 1, line: 1, offset: 0, map: void 0, breakLineCode: o3, needIndent: s3, indentLevel: 0 };
    function a3(e4, t4) {
      c3.code += e4;
    }
    function u3(e4, t4 = true) {
      const n4 = t4 ? o3 : "";
      a3(s3 ? n4 + "  ".repeat(e4) : n4);
    }
    return { context: () => c3, push: a3, indent: function(e4 = true) {
      const t4 = ++c3.indentLevel;
      e4 && u3(t4);
    }, deindent: function(e4 = true) {
      const t4 = --c3.indentLevel;
      e4 && u3(t4);
    }, newline: function() {
      u3(c3.indentLevel);
    }, helper: (e4) => `_${e4}`, needIndent: () => c3.needIndent };
  }(e2, { mode: n2, filename: r2, sourceMap: o2, breakLineCode: s2, needIndent: c2 });
  u2.push("normal" === n2 ? "function __msg__ (ctx) {" : "(ctx) => {"), u2.indent(c2), a2.length > 0 && (u2.push(`const { ${a2.map((e3) => `${e3}: _${e3}`).join(", ")} } = ctx`), u2.newline()), u2.push("return "), ae(u2, e2), u2.deindent(c2), u2.push("}");
  const { code: l2, map: i2 } = u2.context();
  return { ast: e2, code: l2, map: i2 ? i2.toJSON() : void 0 };
};
/*!
  * @intlify/devtools-if v9.1.10
  * (c) 2022 kazuya kawaguchi
  * Released under the MIT License.
  */
const le = { I18nInit: "i18n:init", FunctionTranslate: "function:translate" };
/*!
  * @intlify/core-base v9.1.10
  * (c) 2022 kazuya kawaguchi
  * Released under the MIT License.
  */
let ie = null;
function fe(e2) {
  ie = e2;
}
const pe = me(le.FunctionTranslate);
function me(e2) {
  return (t2) => ie && ie.emit(e2, t2);
}
const de = -1, ke = "";
let he;
function be(e2) {
  he = e2;
}
let ge = null;
const ye = (e2) => {
  ge = e2;
};
let _e = 0;
function Le(e2 = {}) {
  const t2 = b(e2.version) ? e2.version : "9.1.10", n2 = b(e2.locale) ? e2.locale : "en-US", r2 = k(e2.fallbackLocale) || x(e2.fallbackLocale) || b(e2.fallbackLocale) || false === e2.fallbackLocale ? e2.fallbackLocale : n2, o2 = x(e2.messages) ? e2.messages : { [n2]: {} }, s2 = x(e2.datetimeFormats) ? e2.datetimeFormats : { [n2]: {} }, a2 = x(e2.numberFormats) ? e2.numberFormats : { [n2]: {} }, i2 = l({}, e2.modifiers || {}, { upper: (e3) => b(e3) ? e3.toUpperCase() : e3, lower: (e3) => b(e3) ? e3.toLowerCase() : e3, capitalize: (e3) => b(e3) ? `${e3.charAt(0).toLocaleUpperCase()}${e3.substr(1)}` : e3 }), f2 = e2.pluralRules || {}, p2 = h(e2.missing) ? e2.missing : null, m2 = !g(e2.missingWarn) && !c(e2.missingWarn) || e2.missingWarn, d2 = !g(e2.fallbackWarn) && !c(e2.fallbackWarn) || e2.fallbackWarn, _2 = !!e2.fallbackFormat, L2 = !!e2.unresolving, T2 = h(e2.postTranslation) ? e2.postTranslation : null, w2 = x(e2.processor) ? e2.processor : null, v2 = !g(e2.warnHtmlMessage) || e2.warnHtmlMessage, C2 = !!e2.escapeParameter, P2 = h(e2.messageCompiler) ? e2.messageCompiler : he, O2 = h(e2.onWarn) ? e2.onWarn : u, S2 = e2, F2 = y(S2.__datetimeFormatters) ? S2.__datetimeFormatters : /* @__PURE__ */ new Map(), $2 = y(S2.__numberFormatters) ? S2.__numberFormatters : /* @__PURE__ */ new Map(), N2 = y(S2.__meta) ? S2.__meta : {};
  _e++;
  const I2 = { version: t2, cid: _e, locale: n2, fallbackLocale: r2, messages: o2, datetimeFormats: s2, numberFormats: a2, modifiers: i2, pluralRules: f2, missing: p2, missingWarn: m2, fallbackWarn: d2, fallbackFormat: _2, unresolving: L2, postTranslation: T2, processor: w2, warnHtmlMessage: v2, escapeParameter: C2, messageCompiler: P2, onWarn: O2, __datetimeFormatters: F2, __numberFormatters: $2, __meta: N2 };
  return __INTLIFY_PROD_DEVTOOLS__ && function(e3, t3, n3) {
    ie && ie.emit(le.I18nInit, { timestamp: Date.now(), i18n: e3, version: t3, meta: n3 });
  }(I2, t2, N2), I2;
}
function xe(e2, t2, n2, r2, o2) {
  const { missing: s2, onWarn: c2 } = e2;
  if (null !== s2) {
    const r3 = s2(e2, n2, t2, o2);
    return b(r3) ? r3 : t2;
  }
  return t2;
}
function Te(e2, t2, n2) {
  const r2 = e2;
  r2.__localeChainCache || (r2.__localeChainCache = /* @__PURE__ */ new Map());
  let o2 = r2.__localeChainCache.get(n2);
  if (!o2) {
    o2 = [];
    let e3 = [n2];
    for (; k(e3); )
      e3 = we(o2, e3, t2);
    const s2 = k(t2) ? t2 : x(t2) ? t2.default ? t2.default : null : t2;
    e3 = b(s2) ? [s2] : s2, k(e3) && we(o2, e3, false), r2.__localeChainCache.set(n2, o2);
  }
  return o2;
}
function we(e2, t2, n2) {
  let r2 = true;
  for (let o2 = 0; o2 < t2.length && g(r2); o2++) {
    const s2 = t2[o2];
    b(s2) && (r2 = ve(e2, t2[o2], n2));
  }
  return r2;
}
function ve(e2, t2, n2) {
  let r2;
  const o2 = t2.split("-");
  do {
    r2 = Ce(e2, o2.join("-"), n2), o2.splice(-1, 1);
  } while (o2.length && true === r2);
  return r2;
}
function Ce(e2, t2, n2) {
  let r2 = false;
  if (!e2.includes(t2) && (r2 = true, t2)) {
    r2 = "!" !== t2[t2.length - 1];
    const o2 = t2.replace(/!/g, "");
    e2.push(o2), (k(n2) || x(n2)) && n2[o2] && (r2 = n2[o2]);
  }
  return r2;
}
function Pe(e2, t2, n2) {
  e2.__localeChainCache = /* @__PURE__ */ new Map(), Te(e2, n2, t2);
}
const Oe = (e2) => e2;
let Se = /* @__PURE__ */ Object.create(null);
function Fe(e2, t2 = {}) {
  {
    const n2 = (t2.onCacheKey || Oe)(e2), r2 = Se[n2];
    if (r2)
      return r2;
    let o2 = false;
    const s2 = t2.onError || z;
    t2.onError = (e3) => {
      o2 = true, s2(e3);
    };
    const { code: c2 } = function(e3, t3 = {}) {
      const n3 = l({}, t3), r3 = ne(n3).parse(e3);
      return ce(r3, n3), ue(r3, n3);
    }(e2, t2), a2 = new Function(`return ${c2}`)();
    return o2 ? a2 : Se[n2] = a2;
  }
}
function $e(e2) {
  return R(e2, null, void 0);
}
const Ne = () => "", Ie = (e2) => h(e2);
function We(e2, ...t2) {
  const { fallbackFormat: n2, postTranslation: r2, unresolving: s2, fallbackLocale: c2, messages: a2 } = e2, [u2, i2] = Ae(...t2), f2 = (g(i2.missingWarn) ? i2.missingWarn : e2.missingWarn, g(i2.fallbackWarn) ? i2.fallbackWarn : e2.fallbackWarn, g(i2.escapeParameter) ? i2.escapeParameter : e2.escapeParameter), m2 = !!i2.resolvedMessage, d2 = b(i2.default) || g(i2.default) ? g(i2.default) ? u2 : i2.default : n2 ? u2 : "", _2 = n2 || "" !== d2, L2 = b(i2.locale) ? i2.locale : e2.locale;
  f2 && function(e3) {
    k(e3.list) ? e3.list = e3.list.map((e4) => b(e4) ? p(e4) : e4) : y(e3.named) && Object.keys(e3.named).forEach((t3) => {
      b(e3.named[t3]) && (e3.named[t3] = p(e3.named[t3]));
    });
  }(i2);
  let [x2, T2, w2] = m2 ? [u2, L2, a2[L2] || {}] : function(e3, t3, n3, r3, o2, s3) {
    const { messages: c3, onWarn: a3 } = e3, u3 = Te(e3, r3, n3);
    let l2, i3 = {}, f3 = null;
    const p2 = "translate";
    for (let n4 = 0; n4 < u3.length && (l2 = u3[n4], i3 = c3[l2] || {}, null === (f3 = $(i3, t3)) && (f3 = i3[t3]), !b(f3) && !h(f3)); n4++) {
      const n5 = xe(e3, t3, l2, 0, p2);
      n5 !== t3 && (f3 = n5);
    }
    return [f3, l2, i3];
  }(e2, u2, L2, c2), v2 = u2;
  if (m2 || b(x2) || Ie(x2) || _2 && (x2 = d2, v2 = x2), !(m2 || (b(x2) || Ie(x2)) && b(T2)))
    return s2 ? -1 : u2;
  let C2 = false;
  const P2 = Ie(x2) ? x2 : je(e2, u2, T2, x2, v2, () => {
    C2 = true;
  });
  if (C2)
    return x2;
  const O2 = function(e3, t3, n3, r3) {
    const { modifiers: s3, pluralRules: c3 } = e3, a3 = (r4) => {
      const o2 = $(n3, r4);
      if (b(o2)) {
        let n4 = false;
        const s4 = je(e3, r4, t3, o2, r4, () => {
          n4 = true;
        });
        return n4 ? Ne : s4;
      }
      return Ie(o2) ? o2 : Ne;
    }, u3 = { locale: t3, modifiers: s3, pluralRules: c3, messages: a3 };
    e3.processor && (u3.processor = e3.processor);
    r3.list && (u3.list = r3.list);
    r3.named && (u3.named = r3.named);
    o(r3.plural) && (u3.pluralIndex = r3.plural);
    return u3;
  }(e2, T2, w2, i2), S2 = function(e3, t3, n3) {
    const r3 = t3(n3);
    return r3;
  }(0, P2, D(O2)), F2 = r2 ? r2(S2) : S2;
  if (__INTLIFY_PROD_DEVTOOLS__) {
    const t3 = { timestamp: Date.now(), key: b(u2) ? u2 : Ie(x2) ? x2.key : "", locale: T2 || (Ie(x2) ? x2.locale : ""), format: b(x2) ? x2 : Ie(x2) ? x2.source : "", message: F2 };
    t3.meta = l({}, e2.__meta, ge || {}), pe(t3);
  }
  return F2;
}
function je(e2, t2, r2, o2, s2, c2) {
  const { messageCompiler: a2, warnHtmlMessage: u2 } = e2;
  if (Ie(o2)) {
    const e3 = o2;
    return e3.locale = e3.locale || r2, e3.key = e3.key || t2, e3;
  }
  const l2 = a2(o2, function(e3, t3, r3, o3, s3, c3) {
    return { warnHtmlMessage: s3, onError: (e4) => {
      throw c3 && c3(e4), e4;
    }, onCacheKey: (e4) => n(t3, r3, e4) };
  }(0, r2, s2, 0, u2, c2));
  return l2.locale = r2, l2.key = t2, l2.source = o2, l2;
}
function Ae(...e2) {
  const [t2, n2, r2] = e2, s2 = {};
  if (!b(t2) && !o(t2) && !Ie(t2))
    throw $e(14);
  const c2 = o(t2) ? String(t2) : (Ie(t2), t2);
  return o(n2) ? s2.plural = n2 : b(n2) ? s2.default = n2 : x(n2) && !a(n2) ? s2.named = n2 : k(n2) && (s2.list = n2), o(r2) ? s2.plural = r2 : b(r2) ? s2.default = r2 : x(r2) && l(s2, r2), [c2, s2];
}
function Ee(e2, ...t2) {
  const { datetimeFormats: n2, unresolving: r2, fallbackLocale: o2, onWarn: s2 } = e2, { __datetimeFormatters: c2 } = e2, [u2, i2, f2, p2] = Me(...t2);
  g(f2.missingWarn) ? f2.missingWarn : e2.missingWarn;
  g(f2.fallbackWarn) ? f2.fallbackWarn : e2.fallbackWarn;
  const m2 = !!f2.part, d2 = b(f2.locale) ? f2.locale : e2.locale, k2 = Te(e2, o2, d2);
  if (!b(u2) || "" === u2)
    return new Intl.DateTimeFormat(d2).format(i2);
  let h2, y2 = {}, _2 = null;
  for (let t3 = 0; t3 < k2.length && (h2 = k2[t3], y2 = n2[h2] || {}, _2 = y2[u2], !x(_2)); t3++)
    xe(e2, u2, h2, 0, "datetime format");
  if (!x(_2) || !b(h2))
    return r2 ? -1 : u2;
  let L2 = `${h2}__${u2}`;
  a(p2) || (L2 = `${L2}__${JSON.stringify(p2)}`);
  let T2 = c2.get(L2);
  return T2 || (T2 = new Intl.DateTimeFormat(h2, l({}, _2, p2)), c2.set(L2, T2)), m2 ? T2.formatToParts(i2) : T2.format(i2);
}
function Me(...e2) {
  const [t2, n2, r2, c2] = e2;
  let a2, u2 = {}, l2 = {};
  if (b(t2)) {
    if (!/\d{4}-\d{2}-\d{2}(T.*)?/.test(t2))
      throw $e(16);
    a2 = new Date(t2);
    try {
      a2.toISOString();
    } catch (e3) {
      throw $e(16);
    }
  } else if (s(t2)) {
    if (isNaN(t2.getTime()))
      throw $e(15);
    a2 = t2;
  } else {
    if (!o(t2))
      throw $e(14);
    a2 = t2;
  }
  return b(n2) ? u2.key = n2 : x(n2) && (u2 = n2), b(r2) ? u2.locale = r2 : x(r2) && (l2 = r2), x(c2) && (l2 = c2), [u2.key || "", a2, u2, l2];
}
function De(e2, t2, n2) {
  const r2 = e2;
  for (const e3 in n2) {
    const n3 = `${t2}__${e3}`;
    r2.__datetimeFormatters.has(n3) && r2.__datetimeFormatters.delete(n3);
  }
}
function Re(e2, ...t2) {
  const { numberFormats: n2, unresolving: r2, fallbackLocale: o2, onWarn: s2 } = e2, { __numberFormatters: c2 } = e2, [u2, i2, f2, p2] = ze(...t2);
  g(f2.missingWarn) ? f2.missingWarn : e2.missingWarn;
  g(f2.fallbackWarn) ? f2.fallbackWarn : e2.fallbackWarn;
  const m2 = !!f2.part, d2 = b(f2.locale) ? f2.locale : e2.locale, k2 = Te(e2, o2, d2);
  if (!b(u2) || "" === u2)
    return new Intl.NumberFormat(d2).format(i2);
  let h2, y2 = {}, _2 = null;
  for (let t3 = 0; t3 < k2.length && (h2 = k2[t3], y2 = n2[h2] || {}, _2 = y2[u2], !x(_2)); t3++)
    xe(e2, u2, h2, 0, "number format");
  if (!x(_2) || !b(h2))
    return r2 ? -1 : u2;
  let L2 = `${h2}__${u2}`;
  a(p2) || (L2 = `${L2}__${JSON.stringify(p2)}`);
  let T2 = c2.get(L2);
  return T2 || (T2 = new Intl.NumberFormat(h2, l({}, _2, p2)), c2.set(L2, T2)), m2 ? T2.formatToParts(i2) : T2.format(i2);
}
function ze(...e2) {
  const [t2, n2, r2, s2] = e2;
  let c2 = {}, a2 = {};
  if (!o(t2))
    throw $e(14);
  const u2 = t2;
  return b(n2) ? c2.key = n2 : x(n2) && (c2 = n2), b(r2) ? c2.locale = r2 : x(r2) && (a2 = r2), x(s2) && (a2 = s2), [c2.key || "", u2, c2, a2];
}
function Je(e2, t2, n2) {
  const r2 = e2;
  for (const e3 in n2) {
    const n3 = `${t2}__${e3}`;
    r2.__numberFormatters.has(n3) && r2.__numberFormatters.delete(n3);
  }
}
"boolean" != typeof __INTLIFY_PROD_DEVTOOLS__ && (f().__INTLIFY_PROD_DEVTOOLS__ = false);
export { Me as A, Ee as B, ze as C, Re as D, Te as E, ke as M, de as N, l as a, b, y as c, g as d, a as e, R as f, f as g, x as h, o as i, k as j, d as k, N as l, t as m, c as n, h as o, Fe as p, Le as q, be as r, fe as s, $ as t, Pe as u, De as v, Je as w, ye as x, Ae as y, We as z };
