// @layui/layui-vue/es/checkbox/index2.js
import { w as o } from "../utils/index.js";
import { defineComponent as n, computed as l, openBlock as e, createElementBlock as r, normalizeClass as i, normalizeStyle as c, unref as u, createBlock as s, inject as t, useSlots as a, withModifiers as p, createElementVNode as z, renderSlot as y, createTextVNode as f, toDisplayString as m, createCommentVNode as v, createVNode as d } from "vue";
var I = [{ name: "\u5B9E\u5FC3", class: "layui-icon-heart-fill" }, { name: "\u7A7A\u5FC3", class: "layui-icon-heart" }, { name: "\u4EAE\u5EA6/\u6674", class: "layui-icon-light" }, { name: "\u65F6\u95F4/\u5386\u53F2", class: "layui-icon-time" }, { name: "\u84DD\u7259", class: "layui-icon-bluetooth" }, { name: "@\u827E\u7279", class: "layui-icon-at" }, { name: "\u9759\u97F3", class: "layui-icon-mute" }, { name: "\u5F55\u97F3/\u9EA6\u514B\u98CE", class: "layui-icon-mike" }, { name: "\u5BC6\u94A5/\u94A5\u5319", class: "layui-icon-key" }, { name: "\u793C\u7269/\u6D3B\u52A8", class: "layui-icon-gift" }, { name: "\u90AE\u7BB1", class: "layui-icon-email" }, { name: "RSS", class: "layui-icon-rss" }, { name: "WiFi", class: "layui-icon-wifi" }, { name: "\u9000\u51FA/\u6CE8\u9500", class: "layui-icon-logout" }, { name: "Android \u5B89\u5353", class: "layui-icon-android" }, { name: "Apple IOS \u82F9\u679C", class: "layui-icon-ios" }, { name: "Windows", class: "layui-icon-windows" }, { name: "\u7A7F\u68AD\u6846", class: "layui-icon-transfer" }, { name: "\u5BA2\u670D", class: "layui-icon-service" }, { name: "\u51CF", class: "layui-icon-subtraction" }, { name: "\u52A0", class: "layui-icon-addition" }, { name: "\u6ED1\u5757", class: "layui-icon-slider" }, { name: "\u6253\u5370", class: "layui-icon-print" }, { name: "\u5BFC\u51FA", class: "layui-icon-export" }, { name: "\u5217", class: "layui-icon-cols" }, { name: "\u9000\u51FA\u5168\u5C4F", class: "layui-icon-screen-restore" }, { name: "\u5168\u5C4F", class: "layui-icon-screen-full" }, { name: "\u534A\u661F", class: "layui-icon-rate-half" }, { name: "\u661F\u661F-\u7A7A\u5FC3", class: "layui-icon-rate" }, { name: "\u661F\u661F-\u5B9E\u5FC3", class: "layui-icon-rate-solid" }, { name: "\u624B\u673A", class: "layui-icon-cellphone" }, { name: "\u9A8C\u8BC1\u7801", class: "layui-icon-vercode" }, { name: "\u5FAE\u4FE1", class: "layui-icon-login-wechat" }, { name: "QQ", class: "layui-icon-login-qq" }, { name: "\u5FAE\u535A", class: "layui-icon-login-weibo" }, { name: "\u5BC6\u7801", class: "layui-icon-password" }, { name: "\u7528\u6237\u540D", class: "layui-icon-username" }, { name: "\u5237\u65B0-\u7C97", class: "layui-icon-refresh-three" }, { name: "\u6388\u6743", class: "layui-icon-auz" }, { name: "\u5DE6\u5411\u53F3\u4F38\u7F29\u83DC\u5355", class: "layui-icon-spread-left" }, { name: "\u53F3\u5411\u5DE6\u4F38\u7F29\u83DC\u5355", class: "layui-icon-shrink-right" }, { name: "\u96EA\u82B1", class: "layui-icon-snowflake" }, { name: "\u63D0\u793A\u8BF4\u660E", class: "layui-icon-tips" }, { name: "\u4FBF\u7B7E", class: "layui-icon-note" }, { name: "\u4E3B\u9875", class: "layui-icon-home" }, { name: "\u9AD8\u7EA7", class: "layui-icon-senior" }, { name: "\u5237\u65B0", class: "layui-icon-refresh" }, { name: "\u5237\u65B0", class: "layui-icon-refresh-one" }, { name: "\u65D7\u5E1C", class: "layui-icon-flag" }, { name: "\u4E3B\u9898", class: "layui-icon-theme" }, { name: "\u6D88\u606F-\u901A\u77E5", class: "layui-icon-notice" }, { name: "\u7F51\u7AD9", class: "layui-icon-website" }, { name: "\u63A7\u5236\u53F0", class: "layui-icon-console" }, { name: "\u8868\u60C5-\u60CA\u8BB6", class: "layui-icon-face-surprised" }, { name: "\u8BBE\u7F6E-\u7A7A\u5FC3", class: "layui-icon-set" }, { name: "\u6A21\u677F", class: "layui-icon-template-one" }, { name: "\u5E94\u7528", class: "layui-icon-app" }, { name: "\u6A21\u677F", class: "layui-icon-template" }, { name: "\u8D5E", class: "layui-icon-praise" }, { name: "\u8E29", class: "layui-icon-tread" }, { name: "\u7537", class: "layui-icon-male" }, { name: "\u5973", class: "layui-icon-female" }, { name: "\u76F8\u673A-\u7A7A\u5FC3", class: "layui-icon-camera" }, { name: "\u76F8\u673A-\u5B9E\u5FC3", class: "layui-icon-camera-fill" }, { name: "\u83DC\u5355-\u6C34\u5E73", class: "layui-icon-more" }, { name: "\u83DC\u5355-\u5782\u76F4", class: "layui-icon-more-vertical" }, { name: "\u91D1\u989D-\u4EBA\u6C11\u5E01", class: "layui-icon-rmb" }, { name: "\u91D1\u989D-\u7F8E\u5143", class: "layui-icon-dollar" }, { name: "\u94BB\u77F3-\u7B49\u7EA7", class: "layui-icon-diamond" }, { name: "\u706B", class: "layui-icon-fire" }, { name: "\u8FD4\u56DE", class: "layui-icon-return" }, { name: "\u4F4D\u7F6E-\u5730\u56FE", class: "layui-icon-location" }, { name: "\u529E\u516C-\u9605\u8BFB", class: "layui-icon-read" }, { name: "\u8C03\u67E5", class: "layui-icon-survey" }, { name: "\u8868\u60C5-\u5FAE\u7B11", class: "layui-icon-face-smile" }, { name: "\u8868\u60C5-\u54ED\u6CE3", class: "layui-icon-face-cry" }, { name: "\u8D2D\u7269\u8F66", class: "layui-icon-cart-simple" }, { name: "\u8D2D\u7269\u8F66", class: "layui-icon-cart" }, { name: "\u4E0B\u4E00\u9875", class: "layui-icon-next" }, { name: "\u4E0A\u4E00\u9875", class: "layui-icon-prev" }, { name: "\u4E0A\u4F20-\u7A7A\u5FC3-\u62D6\u62FD", class: "layui-icon-upload-drag" }, { name: "\u4E0A\u4F20-\u5B9E\u5FC3", class: "layui-icon-upload" }, { name: "\u4E0B\u8F7D-\u5706\u5708", class: "layui-icon-download-circle" }, { name: "\u7EC4\u4EF6", class: "layui-icon-component" }, { name: "\u6587\u4EF6-\u7C97", class: "layui-icon-file-b" }, { name: "\u7528\u6237", class: "layui-icon-user" }, { name: "\u53D1\u73B0-\u5B9E\u5FC3", class: "layui-icon-find-fill" }, { name: "loading", class: "layui-icon-loading" }, { name: "loading", class: "layui-icon-loading-one" }, { name: "\u6DFB\u52A0", class: "layui-icon-add-one" }, { name: "\u64AD\u653E", class: "layui-icon-play" }, { name: "\u6682\u505C", class: "layui-icon-pause" }, { name: "\u97F3\u9891-\u8033\u673A", class: "layui-icon-headset" }, { name: "\u89C6\u9891", class: "layui-icon-video" }, { name: "\u8BED\u97F3-\u58F0\u97F3", class: "layui-icon-voice" }, { name: "\u6D88\u606F-\u901A\u77E5-\u5587\u53ED", class: "layui-icon-speaker" }, { name: "\u5220\u9664\u7EBF", class: "layui-icon-fonts-del" }, { name: "\u4EE3\u7801", class: "layui-icon-fonts-code" }, { name: "HTML", class: "layui-icon-fonts-html" }, { name: "\u5B57\u4F53\u52A0\u7C97", class: "layui-icon-fonts-strong" }, { name: "\u5220\u9664\u94FE\u63A5", class: "layui-icon-unlink" }, { name: "\u56FE\u7247", class: "layui-icon-picture" }, { name: "\u94FE\u63A5", class: "layui-icon-link" }, { name: "\u8868\u60C5-\u7B11-\u7C97", class: "layui-icon-face-smile-b" }, { name: "\u5DE6\u5BF9\u9F50", class: "layui-icon-align-left" }, { name: "\u53F3\u5BF9\u9F50", class: "layui-icon-align-right" }, { name: "\u5C45\u4E2D\u5BF9\u9F50", class: "layui-icon-align-center" }, { name: "\u5B57\u4F53-\u4E0B\u5212\u7EBF", class: "layui-icon-fonts-u" }, { name: "\u5B57\u4F53-\u659C\u4F53", class: "layui-icon-fonts-i" }, { name: "Tabs \u9009\u9879\u5361", class: "layui-icon-tabs" }, { name: "\u5355\u9009\u6846-\u9009\u4E2D", class: "layui-icon-radio" }, { name: "\u5355\u9009\u6846-\u5019\u9009", class: "layui-icon-circle" }, { name: "\u7F16\u8F91", class: "layui-icon-edit" }, { name: "\u5206\u4EAB", class: "layui-icon-share" }, { name: "\u5220\u9664", class: "layui-icon-delete" }, { name: "\u8868\u5355", class: "layui-icon-form" }, { name: "\u624B\u673A-\u7EC6\u4F53", class: "layui-icon-cellphone-fine" }, { name: "\u804A\u5929 \u5BF9\u8BDD \u6C9F\u901A", class: "layui-icon-dialogue" }, { name: "\u6587\u5B57\u683C\u5F0F\u5316", class: "layui-icon-fonts-clear" }, { name: "\u7A97\u53E3", class: "layui-icon-layer" }, { name: "\u65E5\u671F", class: "layui-icon-date" }, { name: "\u6C34 \u4E0B\u96E8", class: "layui-icon-water" }, { name: "\u4EE3\u7801-\u5706\u5708", class: "layui-icon-code-circle" }, { name: "\u8F6E\u64AD\u7EC4\u56FE", class: "layui-icon-carousel" }, { name: "\u7FFB\u9875", class: "layui-icon-prev-circle" }, { name: "\u5E03\u5C40", class: "layui-icon-layouts" }, { name: "\u5DE5\u5177", class: "layui-icon-util" }, { name: "\u9009\u62E9\u6A21\u677F", class: "layui-icon-templeate-one" }, { name: "\u4E0A\u4F20-\u5706\u5708", class: "layui-icon-upload-circle" }, { name: "\u6811", class: "layui-icon-tree" }, { name: "\u8868\u683C", class: "layui-icon-table" }, { name: "\u56FE\u8868", class: "layui-icon-chart" }, { name: "\u56FE\u6807 \u62A5\u8868 \u5C4F\u5E55", class: "layui-icon-chart-screen" }, { name: "\u5F15\u64CE", class: "layui-icon-engine" }, { name: "\u4E0B\u4E09\u89D2", class: "layui-icon-triangle-d" }, { name: "\u53F3\u4E09\u89D2", class: "layui-icon-triangle-r" }, { name: "\u6587\u4EF6", class: "layui-icon-file" }, { name: "\u8BBE\u7F6E-\u5C0F\u578B", class: "layui-icon-set-sm" }, { name: "\u51CF\u5C11-\u5706\u5708", class: "layui-icon-reduce-circle" }, { name: "\u6DFB\u52A0-\u5706\u5708", class: "layui-icon-add-circle" }, { name: "404", class: "layui-icon-not-found" }, { name: "\u5173\u4E8E", class: "layui-icon-about" }, { name: "\u7BAD\u5934 \u5411\u4E0A", class: "layui-icon-up" }, { name: "\u7BAD\u5934 \u5411\u4E0B", class: "layui-icon-down" }, { name: "\u7BAD\u5934 \u5411\u5DE6", class: "layui-icon-left" }, { name: "\u7BAD\u5934 \u5411\u53F3", class: "layui-icon-right" }, { name: "\u5706\u70B9", class: "layui-icon-circle-dot" }, { name: "\u641C\u7D22", class: "layui-icon-search" }, { name: "\u8BBE\u7F6E-\u5B9E\u5FC3", class: "layui-icon-set-fill" }, { name: "\u7FA4\u7EC4", class: "layui-icon-group" }, { name: "\u597D\u53CB", class: "layui-icon-friends" }, { name: "\u56DE\u590D \u8BC4\u8BBA \u5B9E\u5FC3", class: "layui-icon-reply-fill" }, { name: "\u83DC\u5355 \u9690\u8EAB \u5B9E\u5FC3", class: "layui-icon-menu-fill" }, { name: "\u8BB0\u5F55", class: "layui-icon-log" }, { name: "\u56FE\u7247-\u7EC6\u4F53", class: "layui-icon-picture-fine" }, { name: "\u8868\u60C5-\u7B11-\u7EC6\u4F53", class: "layui-icon-face-smile-fine" }, { name: "\u5217\u8868", class: "layui-icon-list" }, { name: "\u53D1\u5E03 \u7EB8\u98DE\u673A", class: "layui-icon-release" }, { name: "\u5BF9 OK", class: "layui-icon-ok" }, { name: "\u5E2E\u52A9", class: "layui-icon-help" }, { name: "\u5BA2\u670D", class: "layui-icon-chat" }, { name: "top \u7F6E\u9876", class: "layui-icon-top" }, { name: "\u6536\u85CF-\u7A7A\u5FC3", class: "layui-icon-star" }, { name: "\u6536\u85CF-\u5B9E\u5FC3", class: "layui-icon-star-fill" }, { name: "\u5173\u95ED-\u5B9E\u5FC3", class: "layui-icon-close-fill" }, { name: "\u5173\u95ED-\u7A7A\u5FC3", class: "layui-icon-close" }, { name: "\u6B63\u786E", class: "layui-icon-ok-circle" }, { name: "\u6DFB\u52A0-\u5706\u5708-\u7EC6\u4F53", class: "layui-icon-add-circle-fine" }, { name: "\u5E2E\u52A9", class: "layui-icon-help-circle" }, { name: "\u611F\u53F9\u53F7", class: "layui-icon-tips-fill" }, { name: "\u6D4B\u8BD5", class: "layui-icon-test" }, { name: "\u6E05\u7A7A\u5220\u9664", class: "layui-icon-clear" }, { name: "\u952E\u76D8", class: "layui-icon-keyboard" }, { name: "\u9000\u683C", class: "layui-icon-backspace" }, { name: "\u663E\u793A\u5BC6\u7801", class: "layui-icon-show" }, { name: "\u9690\u85CF\u5BC6\u7801", class: "layui-icon-hide" }, { name: "\u9519\u8BEF", class: "layui-icon-error" }, { name: "\u6210\u529F", class: "layui-icon-success" }, { name: "\u7591\u95EE", class: "layui-icon-question" }, { name: "\u9501\u5B9A", class: "layui-icon-lock" }, { name: "\u6708\u4EAE", class: "layui-icon-moon" }, { name: "Github", class: "layui-icon-github" }, { name: "\u7981\u7528", class: "layui-icon-disabled" }, { name: "Gitee", class: "layui-icon-gitee" }, { name: "\u95ED\u773C", class: "layui-icon-eye-invisible" }, { name: "\u773C\u775B", class: "layui-icon-eye" }];
function h(o2, n2) {
  var l2 = Object.keys(o2);
  if (Object.getOwnPropertySymbols) {
    var e2 = Object.getOwnPropertySymbols(o2);
    n2 && (e2 = e2.filter(function(n3) {
      return Object.getOwnPropertyDescriptor(o2, n3).enumerable;
    })), l2.push.apply(l2, e2);
  }
  return l2;
}
function g(o2) {
  for (var n2 = 1; n2 < arguments.length; n2++) {
    var l2 = null != arguments[n2] ? arguments[n2] : {};
    n2 % 2 ? h(Object(l2), true).forEach(function(n3) {
      b(o2, n3, l2[n3]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(o2, Object.getOwnPropertyDescriptors(l2)) : h(Object(l2)).forEach(function(n3) {
      Object.defineProperty(o2, n3, Object.getOwnPropertyDescriptor(l2, n3));
    });
  }
  return o2;
}
function b(o2, n2, l2) {
  return (n2 = function(o3) {
    var n3 = function(o4, n4) {
      if ("object" != typeof o4 || null === o4)
        return o4;
      var l3 = o4[Symbol.toPrimitive];
      if (void 0 !== l3) {
        var e2 = l3.call(o4, n4 || "default");
        if ("object" != typeof e2)
          return e2;
        throw new TypeError("@@toPrimitive must return a primitive value.");
      }
      return ("string" === n4 ? String : Number)(o4);
    }(o3, "string");
    return "symbol" == typeof n3 ? n3 : String(n3);
  }(n2)) in o2 ? Object.defineProperty(o2, n2, { value: l2, enumerable: true, configurable: true, writable: true }) : o2[n2] = l2, o2;
}
var k = n(g(g({}, { name: "LayIcon" }), {}, { props: { size: null, type: null, color: null, prefix: { default: "layui-icon" } }, setup: function(o2) {
  var n2 = o2, s2 = l(function() {
    return { color: n2.color, fontSize: n2.size };
  });
  return l(function() {
    return { type: n2.type, prefix: n2.prefix };
  }), function(n3, l2) {
    return e(), r("i", { class: i([o2.prefix, o2.type]), style: c(u(s2)) }, null, 6);
  };
} }));
g(g({}, { name: "HeartFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-heart-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "HeartIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-heart" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LightIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-light" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TimeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-time" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "BluetoothIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-bluetooth" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AtIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-at" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "MuteIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-mute" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "MikeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-mike" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "KeyIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-key" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "GiftIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-gift" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "EmailIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-email" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RssIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-rss" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "WifiIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-wifi" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LogoutIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-logout" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AndroidIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-android" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "IosIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-ios" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "WindowsIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-windows" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TransferIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-transfer" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ServiceIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-service" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SubtractionIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-subtraction" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AdditionIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-addition" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SliderIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-slider" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PrintIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-print" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ExportIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-export" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ColsIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-cols" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ScreenRestoreIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-screen-restore" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ScreenFullIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-screen-full" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RateHalfIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-rate-half" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RateIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-rate" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RateSolidIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-rate-solid" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CellphoneIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-cellphone" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "VercodeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-vercode" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LoginWechatIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-login-wechat" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LoginQqIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-login-qq" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LoginWeiboIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-login-weibo" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PasswordIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-password" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UsernameIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-username" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RefreshThreeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-refresh-three" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AuzIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-auz" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SpreadLeftIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-spread-left" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ShrinkRightIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-shrink-right" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SnowflakeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-snowflake" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TipsIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-tips" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "NoteIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-note" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "HomeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-home" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SeniorIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-senior" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RefreshIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-refresh" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RefreshOneIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-refresh-one" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FlagIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-flag" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ThemeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-theme" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "NoticeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-notice" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "WebsiteIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-website" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ConsoleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-console" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FaceSurprisedIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-face-surprised" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SetIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-set" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TemplateOneIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-template-one" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AppIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-app" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TemplateIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-template" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PraiseIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-praise" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TreadIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-tread" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "MaleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-male" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FemaleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-female" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CameraIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-camera" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CameraFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-camera-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "MoreIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-more" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "MoreVerticalIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-more-vertical" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RmbIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-rmb" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DollarIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-dollar" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DiamondIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-diamond" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FireIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fire" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ReturnIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-return" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LocationIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-location" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ReadIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-read" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SurveyIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-survey" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FaceSmileIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-face-smile" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FaceCryIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-face-cry" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CartSimpleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-cart-simple" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CartIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-cart" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "NextIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-next" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PrevIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-prev" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UploadDragIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-upload-drag" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UploadIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-upload" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DownloadCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-download-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ComponentIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-component" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FileBIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-file-b" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UserIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-user" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FindFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-find-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LoadingIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-loading" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LoadingOneIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-loading-one" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AddOneIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-add-one" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PlayIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-play" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PauseIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-pause" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "HeadsetIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-headset" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "VideoIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-video" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "VoiceIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-voice" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SpeakerIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-speaker" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FontsDelIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fonts-del" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FontsCodeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fonts-code" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FontsHtmlIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fonts-html" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FontsStrongIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fonts-strong" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UnlinkIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-unlink" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PictureIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-picture" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LinkIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-link" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FaceSmileBIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-face-smile-b" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AlignLeftIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-align-left" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AlignRightIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-align-right" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AlignCenterIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-align-center" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FontsUIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fonts-u" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FontsIIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fonts-i" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TabsIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-tabs" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RadioIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-radio" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "EditIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-edit" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ShareIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-share" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DeleteIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-delete" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FormIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-form" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CellphoneFineIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-cellphone-fine" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DialogueIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-dialogue" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FontsClearIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-fonts-clear" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LayerIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-layer" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DateIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-date" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "WaterIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-water" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CodeCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-code-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CarouselIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-carousel" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PrevCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-prev-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LayoutsIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-layouts" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UtilIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-util" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TempleateOneIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-templeate-one" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UploadCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-upload-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TreeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-tree" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TableIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-table" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ChartIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-chart" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ChartScreenIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-chart-screen" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "EngineIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-engine" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TriangleDIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-triangle-d" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TriangleRIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-triangle-r" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FileIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-file" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SetSmIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-set-sm" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ReduceCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-reduce-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AddCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-add-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "NotFoundIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-not-found" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AboutIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-about" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "UpIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-up" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DownIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-down" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LeftIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-left" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "RightIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-right" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CircleDotIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-circle-dot" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SearchIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-search" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SetFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-set-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "GroupIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-group" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FriendsIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-friends" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ReplyFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-reply-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "MenuFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-menu-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LogIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-log" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "PictureFineIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-picture-fine" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "FaceSmileFineIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-face-smile-fine" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ListIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-list" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ReleaseIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-release" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "OkIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-ok" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "HelpIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-help" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ChatIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-chat" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TopIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-top" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "StarIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-star" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "StarFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-star-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CloseFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-close-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "CloseIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-close" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "OkCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-ok-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "AddCircleFineIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-add-circle-fine" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "HelpCircleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-help-circle" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TipsFillIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-tips-fill" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "TestIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-test" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ClearIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-clear" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "KeyboardIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-keyboard" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "BackspaceIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-backspace" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ShowIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-show" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "HideIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-hide" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "ErrorIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-error" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "SuccessIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-success" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "QuestionIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-question" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "LockIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-lock" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "MoonIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-moon" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "GithubIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-github" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "DisabledIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-disabled" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "GiteeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-gitee" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "EyeInvisibleIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-eye-invisible" }, null, 8, ["color", "size"]);
  };
} });
g(g({}, { name: "EyeIcon" }), {}, { props: { color: null, size: null }, setup: function(o2) {
  var n2 = o2;
  return function(o3, l2) {
    return e(), s(u(k), { color: n2.color, size: n2.size, type: "layui-icon-eye" }, null, 8, ["color", "size"]);
  };
} });
const w = ["onClick", "size"], S = ["name", "value"], F = ["lay-skin"], C = { class: "layui-checkbox-label" }, O = n({ name: "LayCheckbox", props: { name: null, skin: null, label: { default: "" }, value: null, modelValue: { type: [Boolean, Array], default: false }, isIndeterminate: { type: Boolean, default: false }, size: null, disabled: { type: Boolean, default: false } }, emits: ["update:modelValue", "change"], setup(o2, { expose: n2, emit: c2 }) {
  const s2 = o2, { size: I2 } = function(o3) {
    return { size: l(() => {
      const n3 = t("LayForm", {});
      return o3.size || n3.size || "md";
    }) };
  }(s2), h2 = t("checkboxGroup", {}), g2 = l(() => null != h2 && "LayCheckboxGroup" === (h2 == null ? void 0 : h2.name)), b2 = a(), O2 = l({ get: () => g2.value ? h2.modelValue.value.includes(s2.value) : Array.isArray(s2.modelValue) ? s2.modelValue.includes(s2.value) : s2.modelValue, set(o3) {
    g2.value ? L(o3) : Array.isArray(s2.modelValue) ? A(o3) : (c2("update:modelValue", o3), c2("change", o3));
  } }), x2 = l(() => Array.isArray(s2.modelValue) ? [...s2.modelValue] : []), L = function(o3) {
    let n3 = [...h2.modelValue.value];
    o3 ? n3.push(s2.value) : n3.splice(n3.indexOf(s2.value), 1), h2.modelValue.value = n3;
  }, A = function(o3) {
    let n3 = [...x2.value];
    o3 ? n3.push(s2.value) : n3.splice(n3.indexOf(s2.value), 1), c2("update:modelValue", n3), c2("change", n3);
  }, P = function() {
    R.value || (O2.value = !O2.value);
  }, R = l(() => !!s2.disabled || !(!h2.hasOwnProperty("disabled") || !h2.disabled.value));
  return n2({ toggle: P }), (n3, l2) => (e(), r("span", { onClick: p(P, ["stop"]), class: "layui-checkbox", size: u(I2) }, [z("input", { type: "checkbox", name: o2.name, value: o2.value }, null, 8, S), z("div", { class: i(["layui-form-checkbox", { "layui-form-checked": u(O2), "layui-checkbox-disabled layui-disabled": u(R) }]), "lay-skin": o2.skin }, [z("span", C, [u(b2).default || o2.label ? y(n3.$slots, "default", { key: 0 }, () => [f(m(o2.label), 1)]) : v("", true)]), d(u(k), { type: s2.isIndeterminate && u(O2) ? "layui-icon-subtraction" : u(O2) ? "layui-icon-ok" : "" }, null, 8, ["type"])], 10, F)], 8, w));
} }), x = o(O);
export { k as _, O as a, x as c, I as i };
