// @layui/layui-vue/es/colorPicker/index.js
import { w as e } from "../utils/index.js";
import { computed as l, inject as t, defineComponent as a, ref as n, onMounted as u, watch as o, openBlock as r, createBlock as i, withCtx as s, createElementVNode as v, normalizeStyle as g, unref as d, createElementBlock as c, createVNode as p, createCommentVNode as f, Fragment as h, renderList as b, normalizeClass as m } from "vue";
import { _ as y } from "../dropdown/index2.js";
import { d as $ } from "../_chunks/@vueuse/index.js";
import { _ as w } from "../checkbox/index2.js";
const x = ["size"], k = [v("i", { class: "layui-icon layui-colorpicker-trigger-i layui-icon-down" }, null, -1)], S = { class: "layui-color-picker" }, I = v("div", { class: "saturation-value-2" }, null, -1), M = v("div", { class: "saturation-value-3" }, null, -1), A = { class: "layui-color-picker-middle" }, F = { style: { flex: "auto" } }, L = { class: "color-diamond" }, C = { class: "color-value" }, E = { class: "hex" }, B = ["value"], z = { class: "rgba-r" }, V = ["value"], _ = { class: "rgba-g" }, R = ["value"], j = { class: "rgba-b" }, W = ["value"], X = { class: "rgba-a" }, D = ["value"], H = { class: "preset" }, P = ["onClick"], G = e(a({ name: "LayColorPicker", props: { modelValue: { default: { r: 255, g: 255, b: 255, a: 1 } }, preset: { default: ["#16BAAA", "#1e9fff", "#ffb800", "#ff5722", "#16B777"] }, disabled: { type: Boolean, default: false }, eyeDropper: { type: Boolean }, contentClass: null, contentStyle: null, size: null }, emits: ["update:modelValue"], setup(e2, { emit: a2 }) {
  const G2 = e2, { size: U } = function(e3) {
    return { size: l(() => {
      const l2 = t("LayForm", {});
      return e3.size || l2.size || "md";
    }) };
  }(G2), Y = n(null), q = n(null), J = n(null), { isSupported: K, open: N, sRGBHex: O } = $();
  let Q = n("top: 25%;left: 80%;"), T = n("left: 0;"), Z = n("left: calc(100% - 6px);"), ee = n(0), le = n(1), te = n(1), ae = n(255), ne = n(0), ue = n(0), oe = n(1);
  const re = function() {
    K ? N() : console.warn("LayColorPicker: Eye dropper not supported by your browser!");
  };
  u(() => {
    let { r: e3, g: l2, b: t2, a: a3 } = ke(G2.modelValue);
    ae.value = e3, ne.value = l2, ue.value = t2, oe.value = a3;
  }), o(() => G2.modelValue, () => {
    let { r: e3, g: l2, b: t2, a: a3 } = ke(G2.modelValue);
    ae.value = e3, ne.value = l2, ue.value = t2, oe.value = a3;
  }), o(O, (e3) => {
    let { r: l2, g: t2, b: a3, a: n2 } = Me(e3);
    ae.value = l2, ne.value = t2, ue.value = a3, oe.value = n2;
  }), o([ae, ne, ue], (e3) => {
    a2("update:modelValue", Ie(ae.value, ne.value, ue.value, oe.value));
    let { h: l2, s: t2, v: n2 } = function(e4, l3, t3) {
      let a3, n3, u2, o2 = e4 / 255, r2 = l3 / 255, i2 = t3 / 255, s2 = Math.max(o2, r2, i2), v2 = Math.min(o2, r2, i2), g2 = s2 - v2;
      0 === g2 ? a3 = 0 : s2 === o2 ? a3 = (60 * (r2 - i2) / g2 + 360) % 360 : s2 === r2 ? a3 = 60 * ((i2 - o2) / g2 + 2) : s2 === i2 && (a3 = 60 * ((o2 - r2) / g2 + 4));
      n3 = 0 === s2 ? 0 : g2 / s2;
      return u2 = s2, a3 = Math.floor(a3 + 0.5), n3 = Math.floor(100 * n3 + 0.5) / 100, u2 = Math.floor(100 * u2 + 0.5) / 100, { h: a3, s: n3, v: u2 };
    }(ae.value, ne.value, ue.value);
    ee.value = l2, le.value = t2, te.value = n2, Q.value = `top: ${100 - 100 * n2}%;left: ${100 * t2}%;`, T.value = `left: ${ee.value / 360 * 100}%;`;
  }), o(oe, () => {
    a2("update:modelValue", Ie(ae.value, ne.value, ue.value, oe.value)), Z.value = `left: ${oe.value >= 1 ? "calc(100% - 6px)" : 100 * oe.value + "%"};`;
  });
  let ie = l(() => {
    let e3 = ae.value, l2 = ne.value, t2 = ue.value, a3 = oe.value, n2 = ee.value, u2 = le.value, o2 = te.value;
    return { rgb: `rgba(${e3},${l2},${t2})`, rgba: `rgba(${e3},${l2},${t2},${a3})`, hex6: Ie(e3, l2, t2), hex8: Ie(e3, l2, t2, a3), hsv: `hsv(${n2},${u2},${o2})`, hsl: "" };
  });
  function se(e3) {
    let l2 = e3.target.value;
    if (/^#?([0-9a-fA-F]{6}|[0-9a-fA-F]{8})$/.test(l2)) {
      let { r: e4, g: t2, b: a3, a: n2 } = Me(l2);
      ae.value = e4, ne.value = t2, ue.value = a3, oe.value = n2;
    }
  }
  function ve(e3) {
    let l2 = e3.target.value;
    "" !== l2 && (l2 > 255 && (ae.value = 255), l2 < 0 && (ae.value = 0), l2 >= 0 && l2 <= 255 && (ae.value = parseInt(l2)));
  }
  function ge(e3) {
    let l2 = e3.target.value;
    "" !== l2 && (l2 > 255 && (ne.value = 255), l2 < 0 && (ne.value = 0), l2 >= 0 && l2 <= 255 && (ne.value = parseInt(l2)));
  }
  function de(e3) {
    let l2 = e3.target.value;
    "" !== l2 && (l2 > 255 && (ue.value = 255), l2 < 0 && (ue.value = 0), l2 >= 0 && l2 <= 255 && (ue.value = parseInt(l2)));
  }
  function ce(e3) {
    let l2 = e3.target.value;
    "" !== l2 && (l2 = parseFloat(l2), oe.value = l2, l2 > 1 && (oe.value = 1), l2 < 0 && (oe.value = 0), l2 >= 0 && l2 <= 1 && (oe.value = l2));
  }
  function pe(e3) {
    let l2 = Y.value.clientWidth, t2 = Y.value.clientHeight, a3 = e3.pageX - Y.value.getBoundingClientRect().left, n2 = e3.pageY - Y.value.getBoundingClientRect().top;
    a3 = a3 < l2 && a3 > 0 ? a3 : a3 > l2 ? l2 : 0, n2 = n2 < t2 && n2 > 0 ? n2 : n2 > t2 ? t2 : 0, le.value = Math.floor(a3 / l2 * 100 + 0.5) / 100, te.value = Math.floor(100 * (1 - n2 / t2) + 0.5) / 100;
    let { r: u2, g: o2, b: r2 } = Se(ee.value, le.value, te.value);
    ae.value = u2, ne.value = o2, ue.value = r2, Q.value = `top: ${n2}px;left: ${a3}px;`;
  }
  function fe(e3) {
    pe(e3), window.addEventListener("mousemove", pe), window.addEventListener("mouseup", he);
  }
  function he(e3) {
    window.removeEventListener("mousemove", pe), window.removeEventListener("mouseup", he);
  }
  function be(e3) {
    let l2 = q.value.clientWidth, t2 = e3.pageX - Y.value.getBoundingClientRect().left;
    t2 = t2 < l2 && t2 > 0 ? t2 : t2 > l2 ? l2 : 0, ee.value = Math.floor(t2 / l2 * 360 + 0.5);
    let { r: a3, g: n2, b: u2 } = Se(ee.value, le.value, te.value);
    ae.value = a3, ne.value = n2, ue.value = u2, T.value = `left: ${t2 >= l2 - 6 ? l2 - 6 : t2}px;`;
  }
  function me(e3) {
    be(e3), window.addEventListener("mousemove", be), window.addEventListener("mouseup", ye);
  }
  function ye(e3) {
    window.removeEventListener("mousemove", be), window.removeEventListener("mouseup", ye);
  }
  function $e(e3) {
    let l2 = J.value.clientWidth, t2 = e3.pageX - Y.value.getBoundingClientRect().left;
    t2 = t2 < l2 && t2 > 0 ? t2 : t2 > l2 ? l2 : 0, oe.value = Math.floor(t2 / l2 * 100 + 0.5) / 100, Z.value = `left: ${t2 >= l2 - 6 ? l2 - 6 : t2}px;`;
  }
  function we(e3) {
    $e(e3), window.addEventListener("mousemove", $e), window.addEventListener("mouseup", xe);
  }
  function xe(e3) {
    window.removeEventListener("mousemove", $e), window.removeEventListener("mouseup", xe);
  }
  function ke(e3) {
    if (!e3)
      return null;
    {
      let l2, t2, a3, n2;
      if ("string" != typeof e3)
        return l2 = e3.r > 255 ? 255 : e3.r < 0 ? 0 : e3.r, t2 = e3.g > 255 ? 255 : e3.g < 0 ? 0 : e3.g, a3 = e3.b > 255 ? 255 : e3.b < 0 ? 0 : e3.b, n2 = e3.a > 1 ? 1 : e3.a < 0 ? 0 : e3.a, { r: l2, g: t2, b: a3, a: n2 };
      if (/^#?([0-9a-fA-F]{6}|[0-9a-fA-F]{8}|[0-9a-fA-F]{3}|[0-9a-fA-F]{4})$/.test(e3))
        return Me(e3);
    }
  }
  function Se(e3, l2, t2) {
    360 === e3 && (e3 = 0);
    let a3, n2, u2, o2 = Math.floor(e3 / 60) % 6, r2 = e3 / 60 - o2, i2 = t2 * (1 - l2), s2 = t2 * (1 - l2 * r2), v2 = t2 * (1 - l2 * (1 - r2));
    return 0 === o2 ? (a3 = t2, n2 = v2, u2 = i2) : 1 === o2 ? (a3 = s2, n2 = t2, u2 = i2) : 2 === o2 ? (a3 = i2, n2 = t2, u2 = v2) : 3 === o2 ? (a3 = i2, n2 = s2, u2 = t2) : 4 === o2 ? (a3 = v2, n2 = i2, u2 = t2) : 5 === o2 && (a3 = t2, n2 = i2, u2 = s2), a3 = Math.floor(255 * a3 + 0.5), n2 = Math.floor(255 * n2 + 0.5), u2 = Math.floor(255 * u2 + 0.5), { r: a3, g: n2, b: u2 };
  }
  function Ie(e3, l2, t2, a3 = 1) {
    let n2 = 2 !== (e3 = parseInt(e3)).toString(16).length ? "0" + e3.toString(16) : e3.toString(16), u2 = 2 !== (l2 = parseInt(l2)).toString(16).length ? "0" + l2.toString(16) : l2.toString(16), o2 = 2 !== (t2 = parseInt(t2)).toString(16).length ? "0" + t2.toString(16) : t2.toString(16), r2 = "";
    if (1 !== (a3 = parseFloat(a3))) {
      let e4 = Math.floor(256 * a3);
      r2 = 2 !== e4.toString(16).length ? "0" + e4.toString(16) : e4.toString(16);
    }
    return `#${n2}${u2}${o2}${r2}`.toUpperCase();
  }
  function Me(e3) {
    if (/^#?[0-9a-fA-F]{3}$/.test(e3)) {
      let l2 = e3.substring(e3.length - 1, e3.length), t2 = e3.substring(e3.length - 2, e3.length - 1), a3 = e3.substring(e3.length - 3, e3.length - 2);
      return Me(`${a3 + a3}${t2 + t2}${l2 + l2}`);
    }
    if (/^#?[0-9a-fA-F]{4}$/.test(e3)) {
      let l2 = e3.substring(e3.length - 1, e3.length), t2 = e3.substring(e3.length - 2, e3.length - 1), a3 = e3.substring(e3.length - 3, e3.length - 2), n2 = e3.substring(e3.length - 4, e3.length - 3);
      return Me(`${n2 + n2}${a3 + a3}${t2 + t2}${l2 + l2}`);
    }
    if (/^#?[0-9a-fA-F]{6}$/.test(e3)) {
      let l2 = parseInt("0x" + e3.substring(e3.length - 2, e3.length)), t2 = parseInt("0x" + e3.substring(e3.length - 4, e3.length - 2));
      return { r: parseInt("0x" + e3.substring(e3.length - 6, e3.length - 4)), g: t2, b: l2, a: 1 };
    }
    if (/^#?[0-9a-fA-F]{8}$/.test(e3)) {
      let l2 = parseInt("0x" + e3.substring(e3.length - 2, e3.length));
      l2 /= 255;
      let t2 = parseInt("0x" + e3.substring(e3.length - 4, e3.length - 2)), a3 = parseInt("0x" + e3.substring(e3.length - 6, e3.length - 4));
      return { r: parseInt("0x" + e3.substring(e3.length - 8, e3.length - 6)), g: a3, b: t2, a: l2 };
    }
  }
  return (l2, t2) => (r(), i(y, { disabled: e2.disabled, contentClass: e2.contentClass, contentStyle: e2.contentStyle, updateAtScroll: "" }, { content: s(() => [v("div", S, [v("div", { class: "saturation-value", ref_key: "saturationValue", ref: Y, onMousedown: fe }, [v("div", { style: g(`background-color: hsl(${d(ee)}, 100%, 50%);`) }, [v("div", { class: "point", style: g(d(Q)) }, null, 4)], 4), I, M], 544), v("div", A, [v("div", F, [v("div", { class: "hue-slider", ref_key: "hueSlider", ref: q, onMousedown: me }, [v("div", { class: "slider", style: g(d(T)) }, null, 4)], 544), v("div", { class: "alpha-slider", ref_key: "alphaSlider", ref: J, onMousedown: we }, [v("div", { class: "slider", style: g(d(Z)) }, null, 4), v("div", { style: g(`background: linear-gradient(to right, rgba(0,0,0,0), ${d(ie).rgb});width: 100%;height: 100%`) }, null, 4)], 544)]), e2.eyeDropper ? (r(), c("div", { key: 0, onClick: re, style: { "margin-left": "5px" } }, [p(d(w), { type: "layui-icon-edit", size: "22px" })])) : f("", true), v("div", L, [v("div", { style: g(`background-color: ${d(ie).rgba};width: 100%;height: 100%;box-shadow: inset 0 0 0 1px rgba(0, 0, 0, .15), inset 0 0 4px rgba(0, 0, 0, .25);`) }, null, 4)])]), v("div", C, [v("div", E, [v("label", null, [v("input", { value: d(ie).hex8, onInput: se, spellcheck: "false" }, null, 40, B)])]), v("div", z, [v("label", null, [v("input", { value: d(ae), onInput: ve }, null, 40, V)])]), v("div", _, [v("label", null, [v("input", { value: d(ne), onInput: ge }, null, 40, R)])]), v("div", j, [v("label", null, [v("input", { value: d(ue), onInput: de }, null, 40, W)])]), v("div", X, [v("label", null, [v("input", { value: d(oe), onInput: ce }, null, 40, D)])])]), v("ul", H, [(r(true), c(h, null, b(e2.preset, (e3) => (r(), c("li", { key: e3, style: g(`background-color: ${e3}`), onClick: (l3) => function(e4) {
    if (/^#?([0-9a-fA-F]{6}|[0-9a-fA-F]{8})$/.test(e4)) {
      let { r: l4, g: t3, b: a3, a: n2 } = Me(e4);
      ae.value = l4, ne.value = t3, ue.value = a3, oe.value = n2;
    }
  }(e3) }, null, 12, P))), 128))])])]), default: s(() => [v("div", { class: m(["layui-unselect layui-colorpicker", [{ "layui-colorpicker-disabled": e2.disabled }]]), size: d(U) }, [v("span", null, [v("span", { class: "layui-colorpicker-trigger-span", "lay-type": "", style: g(`background-color: ${d(ie).rgba}`) }, k, 4)])], 10, x)]), _: 1 }, 8, ["disabled", "contentClass", "contentStyle"]));
} }));
export { G as default };
